<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Alina G. Mikhailova" />

<meta name="date" content="2025-06-18" />

<title>Analysis of cytochrome b mutational spectra in fishes</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">TemperatureEffectsOnMtDNAspectra</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="I_Polymorphic_Data_Analysis.html">Analysis of the polymorphic data in Fishes obtained by NeMu pipeline</a>
    </li>
  </ul>
</li>
<li>
  <a href="methods.html">Methods</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="license.html">License</a>
</li>
<li>
  <a href="https://github.com/mitoclub/TemperatureEffectsOnMtDNAspectra">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Analysis of cytochrome b mutational spectra
in fishes</h1>
<h4 class="author">Alina G. Mikhailova</h4>
<h4 class="date">2025-06-18</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-06-18
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong>
<code>TemperatureEffectsOnMtDNAspectra/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongobjectspresent">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>Environment:</strong> objects present
</a>
</p>
</div>
<div id="strongEnvironmentstrongobjectspresent"
class="panel-collapse collapse">
<div class="panel-body">
<p>
The global environment had objects present when the code in the R
Markdown file was run. These objects can affect the analysis in your R
Markdown file in unknown ways. For reproduciblity it’s best to always
run the code in an empty environment. Use <code>wflow_publish</code> or
<code>wflow_build</code> to ensure that the code is always run in an
empty environment.
</p>
<p>
The following objects were defined in the global environment when these
results were created:
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
Name
</th>
<th style="text-align:left;">
Class
</th>
<th style="text-align:left;">
Size
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
allEcology
</td>
<td style="text-align:left;">
data.frame
</td>
<td style="text-align:left;">
393.7 Kb
</td>
</tr>
<tr>
<td style="text-align:left;">
allEcologyNOCLASS
</td>
<td style="text-align:left;">
data.frame
</td>
<td style="text-align:left;">
383.1 Kb
</td>
</tr>
<tr>
<td style="text-align:left;">
blue_red_colors
</td>
<td style="text-align:left;">
character
</td>
<td style="text-align:left;">
376 bytes
</td>
</tr>
<tr>
<td style="text-align:left;">
both_freqs
</td>
<td style="text-align:left;">
data.frame
</td>
<td style="text-align:left;">
1.7 Kb
</td>
</tr>
<tr>
<td style="text-align:left;">
Classes
</td>
<td style="text-align:left;">
character
</td>
<td style="text-align:left;">
408 bytes
</td>
</tr>
<tr>
<td style="text-align:left;">
cold_fish_data
</td>
<td style="text-align:left;">
tbl_df;tbl;data.frame
</td>
<td style="text-align:left;">
1.3 Mb
</td>
</tr>
<tr>
<td style="text-align:left;">
ColdFishes
</td>
<td style="text-align:left;">
tbl_df;tbl;data.frame
</td>
<td style="text-align:left;">
2.3 Kb
</td>
</tr>
<tr>
<td style="text-align:left;">
ColdFishesNuc
</td>
<td style="text-align:left;">
tbl_df;tbl;data.frame
</td>
<td style="text-align:left;">
1.2 Kb
</td>
</tr>
<tr>
<td style="text-align:left;">
ColdSubset
</td>
<td style="text-align:left;">
tbl_df;tbl;data.frame
</td>
<td style="text-align:left;">
13.6 Kb
</td>
</tr>
<tr>
<td style="text-align:left;">
ColorA
</td>
<td style="text-align:left;">
character
</td>
<td style="text-align:left;">
112 bytes
</td>
</tr>
<tr>
<td style="text-align:left;">
ColorC
</td>
<td style="text-align:left;">
character
</td>
<td style="text-align:left;">
120 bytes
</td>
</tr>
<tr>
<td style="text-align:left;">
ColorG
</td>
<td style="text-align:left;">
character
</td>
<td style="text-align:left;">
120 bytes
</td>
</tr>
<tr>
<td style="text-align:left;">
ColorT
</td>
<td style="text-align:left;">
character
</td>
<td style="text-align:left;">
112 bytes
</td>
</tr>
<tr>
<td style="text-align:left;">
CompColdFishes
</td>
<td style="text-align:left;">
tbl_df;tbl;data.frame
</td>
<td style="text-align:left;">
2.5 Kb
</td>
</tr>
<tr>
<td style="text-align:left;">
CompMutspec
</td>
<td style="text-align:left;">
tbl_df;tbl;data.frame
</td>
<td style="text-align:left;">
2.6 Kb
</td>
</tr>
<tr>
<td style="text-align:left;">
CompWarmFishes
</td>
<td style="text-align:left;">
tbl_df;tbl;data.frame
</td>
<td style="text-align:left;">
2.5 Kb
</td>
</tr>
<tr>
<td style="text-align:left;">
current_iteration_manual
</td>
<td style="text-align:left;">
numeric
</td>
<td style="text-align:left;">
56 bytes
</td>
</tr>
<tr>
<td style="text-align:left;">
data_dir
</td>
<td style="text-align:left;">
character
</td>
<td style="text-align:left;">
232 bytes
</td>
</tr>
<tr>
<td style="text-align:left;">
docname
</td>
<td style="text-align:left;">
character
</td>
<td style="text-align:left;">
136 bytes
</td>
</tr>
<tr>
<td style="text-align:left;">
exp_both
</td>
<td style="text-align:left;">
data.frame
</td>
<td style="text-align:left;">
1.5 Kb
</td>
</tr>
<tr>
<td style="text-align:left;">
exp_cold
</td>
<td style="text-align:left;">
data.frame
</td>
<td style="text-align:left;">
1.3 Kb
</td>
</tr>
<tr>
<td style="text-align:left;">
exp_warm
</td>
<td style="text-align:left;">
data.frame
</td>
<td style="text-align:left;">
1.3 Kb
</td>
</tr>
<tr>
<td style="text-align:left;">
f2
</td>
<td style="text-align:left;">
gg;ggplot
</td>
<td style="text-align:left;">
86.4 Kb
</td>
</tr>
<tr>
<td style="text-align:left;">
fishEcology
</td>
<td style="text-align:left;">
data.frame
</td>
<td style="text-align:left;">
746.4 Kb
</td>
</tr>
<tr>
<td style="text-align:left;">
fishEcologyTemp
</td>
<td style="text-align:left;">
data.frame
</td>
<td style="text-align:left;">
335.9 Kb
</td>
</tr>
<tr>
<td style="text-align:left;">
GenomeLength
</td>
<td style="text-align:left;">
numeric
</td>
<td style="text-align:left;">
56 bytes
</td>
</tr>
<tr>
<td style="text-align:left;">
hlines_cold
</td>
<td style="text-align:left;">
data.frame
</td>
<td style="text-align:left;">
1.5 Kb
</td>
</tr>
<tr>
<td style="text-align:left;">
hlines_warm
</td>
<td style="text-align:left;">
data.frame
</td>
<td style="text-align:left;">
1.5 Kb
</td>
</tr>
<tr>
<td style="text-align:left;">
merging
</td>
<td style="text-align:left;">
list
</td>
<td style="text-align:left;">
699 Kb
</td>
</tr>
<tr>
<td style="text-align:left;">
mergingtest
</td>
<td style="text-align:left;">
list
</td>
<td style="text-align:left;">
1.3 Mb
</td>
</tr>
<tr>
<td style="text-align:left;">
mtDNAcodonUsage
</td>
<td style="text-align:left;">
data.frame
</td>
<td style="text-align:left;">
1.5 Mb
</td>
</tr>
<tr>
<td style="text-align:left;">
MutAndEcoCb
</td>
<td style="text-align:left;">
tbl_df;tbl;data.frame
</td>
<td style="text-align:left;">
225.9 Kb
</td>
</tr>
<tr>
<td style="text-align:left;">
MutNucAndEcoCb
</td>
<td style="text-align:left;">
tbl_df;tbl;data.frame
</td>
<td style="text-align:left;">
266.3 Kb
</td>
</tr>
<tr>
<td style="text-align:left;">
mutSpec12
</td>
<td style="text-align:left;">
spec_tbl_df;tbl_df;tbl;data.frame
</td>
<td style="text-align:left;">
1 Mb
</td>
</tr>
<tr>
<td style="text-align:left;">
mutSpec12all
</td>
<td style="text-align:left;">
spec_tbl_df;tbl_df;tbl;data.frame
</td>
<td style="text-align:left;">
2 Mb
</td>
</tr>
<tr>
<td style="text-align:left;">
mutSpec12allCb
</td>
<td style="text-align:left;">
spec_tbl_df;tbl_df;tbl;data.frame
</td>
<td style="text-align:left;">
1.2 Mb
</td>
</tr>
<tr>
<td style="text-align:left;">
mutSpec12Cb
</td>
<td style="text-align:left;">
spec_tbl_df;tbl_df;tbl;data.frame
</td>
<td style="text-align:left;">
780.9 Kb
</td>
</tr>
<tr>
<td style="text-align:left;">
MutSpecForAnalysisCb
</td>
<td style="text-align:left;">
tbl_df;tbl;data.frame
</td>
<td style="text-align:left;">
158.2 Kb
</td>
</tr>
<tr>
<td style="text-align:left;">
MutSpecForAnalysisCbAll
</td>
<td style="text-align:left;">
tbl_df;tbl;data.frame
</td>
<td style="text-align:left;">
323 Kb
</td>
</tr>
<tr>
<td style="text-align:left;">
n_cores
</td>
<td style="text-align:left;">
numeric
</td>
<td style="text-align:left;">
56 bytes
</td>
</tr>
<tr>
<td style="text-align:left;">
Ncold
</td>
<td style="text-align:left;">
integer
</td>
<td style="text-align:left;">
56 bytes
</td>
</tr>
<tr>
<td style="text-align:left;">
NcoldNuc
</td>
<td style="text-align:left;">
integer
</td>
<td style="text-align:left;">
56 bytes
</td>
</tr>
<tr>
<td style="text-align:left;">
now
</td>
<td style="text-align:left;">
POSIXct;POSIXt
</td>
<td style="text-align:left;">
344 bytes
</td>
</tr>
<tr>
<td style="text-align:left;">
NucAndEco
</td>
<td style="text-align:left;">
data.frame
</td>
<td style="text-align:left;">
615.3 Kb
</td>
</tr>
<tr>
<td style="text-align:left;">
nucleotide_colors
</td>
<td style="text-align:left;">
character
</td>
<td style="text-align:left;">
736 bytes
</td>
</tr>
<tr>
<td style="text-align:left;">
Nwarm
</td>
<td style="text-align:left;">
integer
</td>
<td style="text-align:left;">
56 bytes
</td>
</tr>
<tr>
<td style="text-align:left;">
NwarmNuc
</td>
<td style="text-align:left;">
integer
</td>
<td style="text-align:left;">
56 bytes
</td>
</tr>
<tr>
<td style="text-align:left;">
obs_both
</td>
<td style="text-align:left;">
data.frame
</td>
<td style="text-align:left;">
1.5 Kb
</td>
</tr>
<tr>
<td style="text-align:left;">
obs_cold
</td>
<td style="text-align:left;">
data.frame
</td>
<td style="text-align:left;">
1.3 Kb
</td>
</tr>
<tr>
<td style="text-align:left;">
obs_warm
</td>
<td style="text-align:left;">
data.frame
</td>
<td style="text-align:left;">
1.3 Kb
</td>
</tr>
<tr>
<td style="text-align:left;">
ObservedCold
</td>
<td style="text-align:left;">
tbl_df;tbl;data.frame
</td>
<td style="text-align:left;">
1.2 Kb
</td>
</tr>
<tr>
<td style="text-align:left;">
ObservedWarm
</td>
<td style="text-align:left;">
tbl_df;tbl;data.frame
</td>
<td style="text-align:left;">
1.2 Kb
</td>
</tr>
<tr>
<td style="text-align:left;">
output_dir
</td>
<td style="text-align:left;">
character
</td>
<td style="text-align:left;">
232 bytes
</td>
</tr>
<tr>
<td style="text-align:left;">
plot_cold_water
</td>
<td style="text-align:left;">
gg;ggplot
</td>
<td style="text-align:left;">
1.4 Mb
</td>
</tr>
<tr>
<td style="text-align:left;">
plot_warm_water
</td>
<td style="text-align:left;">
gg;ggplot
</td>
<td style="text-align:left;">
1.4 Mb
</td>
</tr>
<tr>
<td style="text-align:left;">
plots_dir
</td>
<td style="text-align:left;">
character
</td>
<td style="text-align:left;">
232 bytes
</td>
</tr>
<tr>
<td style="text-align:left;">
plt
</td>
<td style="text-align:left;">
gg;ggplot
</td>
<td style="text-align:left;">
276.9 Kb
</td>
</tr>
<tr>
<td style="text-align:left;">
plt2
</td>
<td style="text-align:left;">
gg;ggplot
</td>
<td style="text-align:left;">
208.2 Kb
</td>
</tr>
<tr>
<td style="text-align:left;">
progress_bar_width
</td>
<td style="text-align:left;">
numeric
</td>
<td style="text-align:left;">
56 bytes
</td>
</tr>
<tr>
<td style="text-align:left;">
reseed
</td>
<td style="text-align:left;">
numeric
</td>
<td style="text-align:left;">
56 bytes
</td>
</tr>
<tr>
<td style="text-align:left;">
sfn
</td>
<td style="text-align:left;">
gg;ggplot;ggarrange
</td>
<td style="text-align:left;">
39.9 Kb
</td>
</tr>
<tr>
<td style="text-align:left;">
simulation_data_long
</td>
<td style="text-align:left;">
tbl_df;tbl;data.frame
</td>
<td style="text-align:left;">
2.6 Mb
</td>
</tr>
<tr>
<td style="text-align:left;">
SimulationData
</td>
<td style="text-align:left;">
data.frame
</td>
<td style="text-align:left;">
799.1 Kb
</td>
</tr>
<tr>
<td style="text-align:left;">
SimulationLengthNumberOfGenerations
</td>
<td style="text-align:left;">
numeric
</td>
<td style="text-align:left;">
56 bytes
</td>
</tr>
<tr>
<td style="text-align:left;">
src_dir
</td>
<td style="text-align:left;">
character
</td>
<td style="text-align:left;">
232 bytes
</td>
</tr>
<tr>
<td style="text-align:left;">
tables_dir
</td>
<td style="text-align:left;">
character
</td>
<td style="text-align:left;">
232 bytes
</td>
</tr>
<tr>
<td style="text-align:left;">
temperaturestat
</td>
<td style="text-align:left;">
summaryDefault;table
</td>
<td style="text-align:left;">
1 Kb
</td>
</tr>
<tr>
<td style="text-align:left;">
todelete
</td>
<td style="text-align:left;">
character
</td>
<td style="text-align:left;">
192 bytes
</td>
</tr>
<tr>
<td style="text-align:left;">
ToMerge
</td>
<td style="text-align:left;">
data.frame
</td>
<td style="text-align:left;">
1.5 Mb
</td>
</tr>
<tr>
<td style="text-align:left;">
total_iterations_manual
</td>
<td style="text-align:left;">
numeric
</td>
<td style="text-align:left;">
56 bytes
</td>
</tr>
<tr>
<td style="text-align:left;">
warm_fish_data
</td>
<td style="text-align:left;">
tbl_df;tbl;data.frame
</td>
<td style="text-align:left;">
1.3 Mb
</td>
</tr>
<tr>
<td style="text-align:left;">
WarmFishes
</td>
<td style="text-align:left;">
tbl_df;tbl;data.frame
</td>
<td style="text-align:left;">
2.3 Kb
</td>
</tr>
<tr>
<td style="text-align:left;">
WarmFishesNuc
</td>
<td style="text-align:left;">
tbl_df;tbl;data.frame
</td>
<td style="text-align:left;">
1.2 Kb
</td>
</tr>
<tr>
<td style="text-align:left;">
WarmSubset
</td>
<td style="text-align:left;">
tbl_df;tbl;data.frame
</td>
<td style="text-align:left;">
17.6 Kb
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240719code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240719)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240719code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240719)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcommitoclubTemperatureEffectsOnMtDNAspectratree77db01ef622b2f1f41cab6356064e0562a9a41e8targetblank77db01ea">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/mitoclub/TemperatureEffectsOnMtDNAspectra/tree/77db01ef622b2f1f41cab6356064e0562a9a41e8" target="_blank">77db01e</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcommitoclubTemperatureEffectsOnMtDNAspectratree77db01ef622b2f1f41cab6356064e0562a9a41e8targetblank77db01ea"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/mitoclub/TemperatureEffectsOnMtDNAspectra/tree/77db01ef622b2f1f41cab6356064e0562a9a41e8" target="_blank">77db01e</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/.RData
    Ignored:    analysis/.Rhistory
    Ignored:    code/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    data/Codons_of_CytB_gene_Chordata.txt
    Ignored:    data/ColdAndWarmFishesSpectra.txt
    Ignored:    data/Dataset_S1.csv
    Ignored:    data/ExpectedFractions.txt
    Ignored:    data/Extra_climate_data_2025.csv
    Ignored:    data/Extra_temperature_grow_data_2025.csv
    Ignored:    data/Extra_temperature_popQB_data_2025.csv
    Ignored:    data/Full_fish_ecology_table_2025.csv
    Ignored:    data/MIDORI2_LONGEST_NUC_GB265_Cytb_BLAST.fasta
    Ignored:    data/MIDORI2_LONGEST_NUC_GB265_Cytb_BLAST.fasta.fai
    Ignored:    data/MIDORI_WG_CYTB_FISHES.csv
    Ignored:    data/MolEvolRatesProxy.txt
    Ignored:    data/SimulationData.csv
    Ignored:    data/SimulationData2.csv
    Ignored:    data/Species_by_climate_zones.txt
    Ignored:    output/.DS_Store
    Ignored:    output/figures/.DS_Store
    Ignored:    output/tables/.DS_Store

Untracked files:
    Untracked:  output/figures/Observed_vs_Expected_Frequencies.png
    Untracked:  output/figures/Observed_vs_Expected_Frequencies_subset.png
    Untracked:  output/figures/Simulation_expected_observed.pdf
    Untracked:  output/figures/Simulation_expected_observed_subset.pdf

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/I_CYTB_polymorphic_mutspec_analysis.Rmd</code>) and HTML
(<code>docs/I_CYTB_polymorphic_mutspec_analysis.html</code>) files. If
you’ve configured a remote Git repository (see
<code>?wflow_git_remote</code>), click on the hyperlinks in the table
below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mitoclub/TemperatureEffectsOnMtDNAspectra/blob/77db01ef622b2f1f41cab6356064e0562a9a41e8/analysis/I_CYTB_polymorphic_mutspec_analysis.Rmd" target="_blank">77db01e</a>
</td>
<td>
Alya Mikhailova
</td>
<td>
2025-06-18
</td>
<td>
First chapter all done
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mitoclub/TemperatureEffectsOnMtDNAspectra/77db01ef622b2f1f41cab6356064e0562a9a41e8/docs/I_CYTB_polymorphic_mutspec_analysis.html" target="_blank">77db01e</a>
</td>
<td>
Alya Mikhailova
</td>
<td>
2025-06-18
</td>
<td>
First chapter all done
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div
id="reading-mutation-spectra-database-l-strand-and-ecological-databases"
class="section level1">
<h1>Reading mutation spectra database (L-strand) and ecological
databases</h1>
<pre class="r"><code>mutSpec12 &lt;- read_csv(here(data_dir, &quot;MutSpecVertebrates12fish.csv&quot;))
mutSpec12Cb &lt;- dplyr::filter(mutSpec12, Gene == &quot;Cytb&quot;)
cat(&quot;N species =&quot;, length(unique(mutSpec12Cb$Species)))</code></pre>
<pre><code>N species = 840</code></pre>
<pre class="r"><code>head(mutSpec12Cb)</code></pre>
<pre><code># A tibble: 6 × 9
  Gene  Class       Species            Mut   Observed TreeLeaves MSAsize Expected MutSpec
  &lt;chr&gt; &lt;chr&gt;       &lt;chr&gt;              &lt;chr&gt;    &lt;dbl&gt;      &lt;dbl&gt;   &lt;dbl&gt;    &lt;dbl&gt;   &lt;dbl&gt;
1 Cytb  Actinopteri Scombrops_gilberti A&gt;C        4           68    1140     71.0  0.0326
2 Cytb  Actinopteri Scombrops_gilberti A&gt;G       21.6         68    1140    111.   0.113 
3 Cytb  Actinopteri Scombrops_gilberti A&gt;T        2           68    1140     71.0  0.0163
4 Cytb  Actinopteri Scombrops_gilberti C&gt;A        3           68    1140     86.3  0.0201
5 Cytb  Actinopteri Scombrops_gilberti C&gt;G        0           68    1140     86.3  0     
6 Cytb  Actinopteri Scombrops_gilberti C&gt;T       10.7         68    1140    182.   0.0340</code></pre>
<pre class="r"><code>MutSpecForAnalysisCb &lt;- mutSpec12Cb %&gt;% dplyr::select(Class, Species, Mut, MutSpec)
MutSpecForAnalysisCb$Mut &lt;- gsub(&quot;&gt;&quot;, &quot;_&quot;, MutSpecForAnalysisCb$Mut)
MutSpecForAnalysisCb &lt;- MutSpecForAnalysisCb %&gt;% pivot_wider(names_from = Mut, values_from = MutSpec, , values_fn = median) 
colnames(MutSpecForAnalysisCb) &lt;- c(&quot;Class&quot;,     # Only CytB
                                  &quot;Species&quot;, 
                                  &quot;T_G.H&quot;, 
                                  &quot;T_C.H&quot;, 
                                  &quot;T_A.H&quot;, 
                                  &quot;G_T.H&quot;, 
                                  &quot;G_C.H&quot;, 
                                  &quot;G_A.H&quot;, 
                                  &quot;C_T.H&quot;, 
                                  &quot;C_G.H&quot;, 
                                  &quot;C_A.H&quot;, 
                                  &quot;A_T.H&quot;, 
                                  &quot;A_G.H&quot;, 
                                  &quot;A_C.H&quot;)</code></pre>
<pre class="r"><code>MutSpecForAnalysisBoth &lt;- mutSpec12 %&gt;% dplyr::select(Class, Species, Mut, MutSpec)
MutSpecForAnalysisBoth$Mut &lt;- gsub(&quot;&gt;&quot;, &quot;_&quot;, MutSpecForAnalysisBoth$Mut)
MutSpecForAnalysisBoth &lt;- MutSpecForAnalysisBoth %&gt;% pivot_wider(names_from = Mut, values_from = MutSpec, , values_fn = median) 
cat(&quot;N species =&quot;, length(unique(mutSpec12$Species)))</code></pre>
<pre><code>N species = 1035</code></pre>
<pre class="r"><code>colnames(MutSpecForAnalysisBoth) &lt;- c(&quot;Class&quot;,      # Both genes
                                  &quot;Species&quot;, 
                                  &quot;T_G.H&quot;, 
                                  &quot;T_C.H&quot;, 
                                  &quot;T_A.H&quot;, 
                                  &quot;G_T.H&quot;, 
                                  &quot;G_C.H&quot;, 
                                  &quot;G_A.H&quot;, 
                                  &quot;C_T.H&quot;, 
                                  &quot;C_G.H&quot;, 
                                  &quot;C_A.H&quot;, 
                                  &quot;A_T.H&quot;, 
                                  &quot;A_G.H&quot;, 
                                  &quot;A_C.H&quot;)</code></pre>
<pre class="r"><code>fishEcology &lt;- read.csv(here(data_dir, &quot;Full_fish_ecology_table_2025.csv&quot;))
head(fishEcology)</code></pre>
<pre><code>                  Species     Wmax      Lmax     tmax                   FeedingType FoodTroph    tm       Lm Temp_min Temp_max
1     Abbottina_rivularis       NA   9.10000  2.00000                          &lt;NA&gt;        NA 1.000  4.50000       NA       NA
2           Abramis_brama 4112.167  46.91111 14.41667 hunting macrofauna (predator)      3.15 4.125 28.75000       10       24
3    Abudefduf_vaigiensis       NA  18.96667       NA     grazing on aquatic plants      3.39    NA 12.00000       NA       NA
4 Acanthemblemaria_aspera       NA   3.00000  4.00000 hunting macrofauna (predator)      3.12    NA       NA       NA       NA
5  Acanthocybium_solandri 7200.000 177.88000  9.30000 hunting macrofauna (predator)      4.45 2.000 99.33333       NA       NA
6     Acanthopagrus_berda  660.000  56.00000 12.50000 hunting macrofauna (predator)      3.50    NA 21.00000       NA       NA
  Climate_zone Temp_avg
1  subtropical       NA
2    temperate       17
3     tropical       NA
4         &lt;NA&gt;       NA
5  subtropical       NA
6     tropical       NA</code></pre>
<pre class="r"><code>cor.test(fishEcology$tm, fishEcology$Temp_avg, method = &quot;spearman&quot;)</code></pre>
<pre><code>
    Spearman&#39;s rank correlation rho

data:  fishEcology$tm and fishEcology$Temp_avg
S = 232808, p-value = 0.0002596
alternative hypothesis: true rho is not equal to 0
sample estimates:
       rho 
-0.3558985 </code></pre>
</div>
<div id="calculating-of-average-mutspec-for-fisches"
class="section level1">
<h1>Calculating of average MutSpec for Fisches</h1>
<div id="cytb" class="section level2">
<h2>Cytb</h2>
<pre class="r"><code>AverageMutSpecCb &lt;- mutSpec12Cb %&gt;% group_by(Mut) %&gt;% summarise(meanmutspec = mean(MutSpec, na.rm = TRUE))
AverageMutSpecCb$Mut.H &lt;- c(&quot;T_G.H&quot;, 
                          &quot;T_C.H&quot;, 
                          &quot;T_A.H&quot;, 
                          &quot;G_T.H&quot;, 
                          &quot;G_C.H&quot;, 
                          &quot;G_A.H&quot;, 
                          &quot;C_T.H&quot;, 
                          &quot;C_G.H&quot;, 
                          &quot;C_A.H&quot;, 
                          &quot;A_T.H&quot;, 
                          &quot;A_G.H&quot;, 
                          &quot;A_C.H&quot;)

f1 = ggplot(AverageMutSpecCb,
  aes(
    x = AverageMutSpecCb$Mut.H,
    y = AverageMutSpecCb$meanmutspec,
    fill = AverageMutSpecCb$Mut.H
    )
  ) +
  geom_bar(stat = &quot;identity&quot;) +
  scale_fill_manual(
    values = c(
      &quot;#bdbdbd&quot;,
      &quot;#00a426&quot;,
      &quot;#bdbdbd&quot;,
      &quot;#bdbdbd&quot;,
      &quot;#bdbdbd&quot;,
      &quot;red&quot;,
      &quot;#900000&quot;,
      &quot;#bdbdbd&quot;,
      &quot;#bdbdbd&quot;,
      &quot;#bdbdbd&quot;,
      &quot;darkgreen&quot;,
      &quot;#bdbdbd&quot;
    )
  ) +
  labs(title = &quot;Average mitochondrial mutational spectrum in Fishes (cytB, N=840)&quot;, x = &quot;Substitution types&quot;, y = &quot;Normalised Frequencies&quot;) + theme(
    legend.position = &quot;none&quot;,
    panel.background = element_blank(),
    axis.line = element_line(color = &quot;black&quot;)
  )

ggsave(here(plots_dir,&quot;Average_mutspec_in_Fishes_CytB.pdf&quot;),
  plot = f1,
  width = 6,
  height = 6
)
f1</code></pre>
<p><img src="figure/I_CYTB_polymorphic_mutspec_analysis.Rmd/average_mutspec_plot_cytb-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-average_mutspec_plot_cytb-1">
Past versions of average_mutspec_plot_cytb-1.png
</button>
</p>
<div id="fig-average_mutspec_plot_cytb-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mitoclub/TemperatureEffectsOnMtDNAspectra/blob/77db01ef622b2f1f41cab6356064e0562a9a41e8/docs/figure/I_CYTB_polymorphic_mutspec_analysis.Rmd/average_mutspec_plot_cytb-1.png" target="_blank">77db01e</a>
</td>
<td>
Alya Mikhailova
</td>
<td>
2025-06-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="both-genes" class="section level2">
<h2>Both genes</h2>
<pre class="r"><code>AverageMutSpecBoth &lt;- mutSpec12 %&gt;% group_by(Mut) %&gt;% summarise(meanmutspec = mean(MutSpec, na.rm = TRUE))
AverageMutSpecBoth$Mut.H &lt;- c(&quot;T_G.H&quot;, 
                          &quot;T_C.H&quot;, 
                          &quot;T_A.H&quot;, 
                          &quot;G_T.H&quot;, 
                          &quot;G_C.H&quot;, 
                          &quot;G_A.H&quot;, 
                          &quot;C_T.H&quot;, 
                          &quot;C_G.H&quot;, 
                          &quot;C_A.H&quot;, 
                          &quot;A_T.H&quot;, 
                          &quot;A_G.H&quot;, 
                          &quot;A_C.H&quot;)

f1 = ggplot(AverageMutSpecBoth,
  aes(
    x = AverageMutSpecBoth$Mut.H,
    y = AverageMutSpecBoth$meanmutspec,
    fill = AverageMutSpecBoth$Mut.H
    )
  ) +
  geom_bar(stat = &quot;identity&quot;) +
  scale_fill_manual(
    values = c(
      &quot;#bdbdbd&quot;,
      &quot;#00a426&quot;,
      &quot;#bdbdbd&quot;,
      &quot;#bdbdbd&quot;,
      &quot;#bdbdbd&quot;,
      &quot;#ff0000&quot;,
      &quot;#900000&quot;,
      &quot;#bdbdbd&quot;,
      &quot;#bdbdbd&quot;,
      &quot;#bdbdbd&quot;,
      &quot;#005213&quot;,
      &quot;#bdbdbd&quot;
    )
  ) +
  labs(title = &quot;Average mitochondrial mutational spectrum in Fishes (cytB+COX1, N=1035)&quot;, x = &quot;Substitution types&quot;, y = &quot;Normalised Frequencies&quot;) + theme(
    legend.position = &quot;none&quot;,
    # Remove legend
    panel.background = element_blank(),
    axis.line = element_line(color = &quot;black&quot;)# Remove background
  )

f1</code></pre>
<p><img src="figure/I_CYTB_polymorphic_mutspec_analysis.Rmd/average_mutspec_plot_both-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-average_mutspec_plot_both-1">
Past versions of average_mutspec_plot_both-1.png
</button>
</p>
<div id="fig-average_mutspec_plot_both-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mitoclub/TemperatureEffectsOnMtDNAspectra/blob/77db01ef622b2f1f41cab6356064e0562a9a41e8/docs/figure/I_CYTB_polymorphic_mutspec_analysis.Rmd/average_mutspec_plot_both-1.png" target="_blank">77db01e</a>
</td>
<td>
Alya Mikhailova
</td>
<td>
2025-06-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="comparison-of-average-mutspec-between-cold-and-warm-fisches"
class="section level1">
<h1>Comparison of average MutSpec between cold and warm fisches</h1>
<pre class="r"><code>merging &lt;- list(MutSpecForAnalysisCb, fishEcology)
MutAndEcoCb &lt;- merging %&gt;% reduce(left_join, by= &quot;Species&quot;)

#write.table(MutAndEcoCb, file = here(tables_dir, &quot;Supplementary_Material_1_Mitochondrial_mutational_spectra_cytb_and_ecology_Fishes.txt&quot;), row.names = FALSE)

temperaturestat &lt;- summary(MutAndEcoCb$Temp_avg)

WarmFishesCb &lt;- filter(MutAndEcoCb, Temp_avg &gt; as.numeric(temperaturestat[&quot;3rd Qu.&quot;]))
WarmFishesCb &lt;- WarmFishesCb %&gt;% dplyr::select(Species, T_G.H, 
                                        T_C.H, 
                                        T_A.H, 
                                        G_T.H, 
                                        G_C.H, 
                                        G_A.H, 
                                        C_T.H, 
                                        C_G.H, 
                                        C_A.H, 
                                        A_T.H, 
                                        A_G.H, 
                                        A_C.H)
WarmFishesCb &lt;- pivot_longer(WarmFishesCb, cols = c(&quot;T_G.H&quot;, 
                                  &quot;T_C.H&quot;, 
                                  &quot;T_A.H&quot;, 
                                  &quot;G_T.H&quot;, 
                                  &quot;G_C.H&quot;, 
                                  &quot;G_A.H&quot;, 
                                  &quot;C_T.H&quot;, 
                                  &quot;C_G.H&quot;, 
                                  &quot;C_A.H&quot;, 
                                  &quot;A_T.H&quot;, 
                                  &quot;A_G.H&quot;, 
                                  &quot;A_C.H&quot;), names_to = &quot;Mut&quot;, values_to = &quot;MutSpec&quot; )
NwarmCb &lt;- length(unique(WarmFishesCb$Species))

ColdFishesCb &lt;- filter(MutAndEcoCb, Temp_avg &lt; as.numeric(temperaturestat[&quot;1st Qu.&quot;]))
ColdFishesCb &lt;- ColdFishesCb %&gt;% dplyr::select(Species, T_G.H, 
                                        T_C.H, 
                                        T_A.H, 
                                        G_T.H, 
                                        G_C.H, 
                                        G_A.H, 
                                        C_T.H, 
                                        C_G.H, 
                                        C_A.H, 
                                        A_T.H, 
                                        A_G.H, 
                                        A_C.H)
ColdFishesCb &lt;- pivot_longer(ColdFishesCb, cols = c(&quot;T_G.H&quot;, 
                                  &quot;T_C.H&quot;, 
                                  &quot;T_A.H&quot;, 
                                  &quot;G_T.H&quot;, 
                                  &quot;G_C.H&quot;, 
                                  &quot;G_A.H&quot;, 
                                  &quot;C_T.H&quot;, 
                                  &quot;C_G.H&quot;, 
                                  &quot;C_A.H&quot;, 
                                  &quot;A_T.H&quot;, 
                                  &quot;A_G.H&quot;, 
                                  &quot;A_C.H&quot;), names_to = &quot;Mut&quot;, values_to = &quot;MutSpec&quot; )
NcoldCb &lt;- length(unique(ColdFishesCb$Species))

f1a &lt;- ggplot(
  WarmFishesCb,
  aes(
    x = Mut,
    y = MutSpec
  )
) +
  geom_jitter(aes(colour = Mut), width = 0.1) +
  scale_color_manual(
    values = c(
      &quot;#bdbdbd&quot;,
      &quot;#00a426&quot;,
      &quot;#bdbdbd&quot;,
      &quot;#bdbdbd&quot;,
      &quot;#bdbdbd&quot;,
      &quot;red&quot;,
      &quot;#900000&quot;,
      &quot;#bdbdbd&quot;,
      &quot;#bdbdbd&quot;,
      &quot;#bdbdbd&quot;,
      &quot;darkgreen&quot;,
      &quot;#bdbdbd&quot;
    )
  ) +
  labs(
    title = &quot;Warm Fishes, &gt;23C&quot;,
    x = &quot;Substitution types&quot;,
    y = &quot;Normalised Frequencies&quot;
  ) +
  theme(
    legend.position = &quot;none&quot;,
    panel.background = element_blank(),
    axis.line = element_line(color = &quot;black&quot;)
  ) +
  geom_text(
    x = 2.5,
    y = 0.63,
    label = paste(&quot;N warm =&quot;, NwarmCb),
    size = 7
  ) +
  stat_summary(
    fun = &quot;median&quot;,
    geom = &quot;line&quot;,        
    aes(group = 2),      
    color = &quot;black&quot;,     
    size = 0.5             
  )


f1b &lt;- ggplot(
 ColdFishesCb,
  aes(
    x = Mut,
    y = MutSpec
  )
) +
  geom_jitter(aes(colour = Mut), width = 0.1) +
  scale_color_manual(
    values = c(
      &quot;#bdbdbd&quot;,
      &quot;#00a426&quot;,
      &quot;#bdbdbd&quot;,
      &quot;#bdbdbd&quot;,
      &quot;#bdbdbd&quot;,
      &quot;red&quot;,
      &quot;#900000&quot;,
      &quot;#bdbdbd&quot;,
      &quot;#bdbdbd&quot;,
      &quot;#bdbdbd&quot;,
      &quot;darkgreen&quot;,
      &quot;#bdbdbd&quot;
    )
  ) +
  labs(
    title = &quot;Cold Fishes, &lt;13C&quot;,
    x = &quot;Substitution types&quot;,
    y = &quot;Normalised Frequencies&quot;
  ) +
  theme(
    legend.position = &quot;none&quot;,
    panel.background = element_blank(),
    axis.line = element_line(color = &quot;black&quot;)
  ) +
  geom_text(
    x = 2.5,
    y = 0.56,
    label = paste(&quot;N cold =&quot;, NcoldCb),
    size = 7
  ) +
  stat_summary(
    fun = &quot;median&quot;,
    geom = &quot;line&quot;,        
    aes(group = 2),      
    color = &quot;black&quot;,     
    size = 0.5             
  )

f1 &lt;- ggarrange(f1b, f1a, ncol = 2)

ggsave(here(plots_dir,&quot;Average mitochondrial mutational spectrum in Fishes divided by groups (cytB).pdf&quot;),
  plot = f1,
  width = 12,
  height = 6
)
f1</code></pre>
<p><img src="figure/I_CYTB_polymorphic_mutspec_analysis.Rmd/average_mutspec_cytb_groups-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-average_mutspec_cytb_groups-1">
Past versions of average_mutspec_cytb_groups-1.png
</button>
</p>
<div id="fig-average_mutspec_cytb_groups-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mitoclub/TemperatureEffectsOnMtDNAspectra/blob/77db01ef622b2f1f41cab6356064e0562a9a41e8/docs/figure/I_CYTB_polymorphic_mutspec_analysis.Rmd/average_mutspec_cytb_groups-1.png" target="_blank">77db01e</a>
</td>
<td>
Alya Mikhailova
</td>
<td>
2025-06-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="studies-of-a_g-spectra-with-temperature-and-other-parameters-in-fishes"
class="section level1">
<h1>Studies of A_G spectra with temperature and other parameters in
fishes</h1>
<div id="a_g-freqs-dependence-on-temperature" class="section level2">
<h2>A_G freqs dependence on temperature</h2>
<pre class="r"><code>correlation_results &lt;- MutAndEcoCb |&gt;
  dplyr::select(
    &quot;A_G.H&quot;,
    &quot;A_T.H&quot;,
    &quot;A_C.H&quot;,
    &quot;T_A.H&quot;,
    &quot;T_G.H&quot;,
    &quot;T_C.H&quot;,
    &quot;C_A.H&quot;,
    &quot;C_T.H&quot;,
    &quot;C_G.H&quot;,
    &quot;G_T.H&quot;,
    &quot;G_A.H&quot;,
    &quot;G_C.H&quot;
  ) |&gt;
  map_dfr( ~ tidy(cor.test(.x, MutAndEcoCb$Temp_avg), method = &#39;spearman&#39;), .id = &quot;variable&quot;) |&gt;
  arrange(desc(abs(estimate)))

correlation_results</code></pre>
<pre><code># A tibble: 12 × 9
   variable estimate statistic   p.value parameter conf.low conf.high method                               alternative
   &lt;chr&gt;       &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;     &lt;int&gt;    &lt;dbl&gt;     &lt;dbl&gt; &lt;chr&gt;                                &lt;chr&gt;      
 1 A_G.H      0.317      4.16  0.0000527       155   0.169     0.451  Pearson&#39;s product-moment correlation two.sided  
 2 T_C.H     -0.168     -2.13  0.0351          155  -0.317    -0.0120 Pearson&#39;s product-moment correlation two.sided  
 3 G_A.H      0.123      1.55  0.124           155  -0.0339    0.275  Pearson&#39;s product-moment correlation two.sided  
 4 A_T.H     -0.119     -1.49  0.137           155  -0.271     0.0382 Pearson&#39;s product-moment correlation two.sided  
 5 C_G.H     -0.115     -1.45  0.150           155  -0.267     0.0419 Pearson&#39;s product-moment correlation two.sided  
 6 T_A.H      0.0912     1.14  0.256           155  -0.0664    0.244  Pearson&#39;s product-moment correlation two.sided  
 7 T_G.H      0.0798     0.997 0.320           155  -0.0778    0.234  Pearson&#39;s product-moment correlation two.sided  
 8 A_C.H     -0.0498    -0.621 0.536           155  -0.205     0.108  Pearson&#39;s product-moment correlation two.sided  
 9 C_A.H     -0.0489    -0.609 0.543           155  -0.204     0.109  Pearson&#39;s product-moment correlation two.sided  
10 C_T.H     -0.0413    -0.515 0.607           155  -0.197     0.116  Pearson&#39;s product-moment correlation two.sided  
11 G_C.H     -0.0364    -0.454 0.651           155  -0.192     0.121  Pearson&#39;s product-moment correlation two.sided  
12 G_T.H     -0.0325    -0.405 0.686           155  -0.188     0.125  Pearson&#39;s product-moment correlation two.sided  </code></pre>
<pre class="r"><code>#write.table(correlation_results, file = here(tables_dir, &quot;Table1.txt&quot;), row.names = FALSE)

SubsetClass&lt;- unique(MutAndEcoCb$Class)
spN &lt;- nrow(MutAndEcoCb[!is.na(MutAndEcoCb$Temp_avg),])
cat(&quot;N species =&quot;, spN, &quot;Taxa =&quot;, SubsetClass)</code></pre>
<pre><code>N species = 157 Taxa = Actinopteri Chondrichthyes</code></pre>
<pre class="r"><code>f1 &lt;- ggplot(MutAndEcoCb, aes(x = Temp_avg, y = A_G.H)) +
  geom_point(color = &quot;#73514f&quot;) +                  # Scatter plot
  geom_smooth(
    method = &quot;lm&quot;,
    color = &quot;black&quot;,
    # Regression line + CI
    fill = &quot;lightgray&quot;,
    se = TRUE
  ) +     # Set se=TRUE for confidence interval
  stat_cor(
    method = &quot;spearman&quot;,
    # Spearman correlation stats
    aes(label = paste(..r.label.., ..p.label.., &quot;N&quot;, spN, sep = &quot;~&quot;))
  ) +  # Include n
  labs(title = &quot;Positive assosiation of A&gt;G.H with temperature&quot;, x = &quot;Median annual water temperature, ?C&quot;, y = &quot;AH&gt;GH&quot;) +
  theme(
    legend.position = &quot;none&quot;,
    panel.background = element_blank(),
    axis.line = element_line(color = &quot;black&quot;)
  )
 
ggsave(here(plots_dir,&quot;Assosiation_of_A_G_with_temperature.pdf&quot;),
  plot = f1,
  width = 6,
  height = 6
) 
f1</code></pre>
<p><img src="figure/I_CYTB_polymorphic_mutspec_analysis.Rmd/statistics_with_temp_cytb-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-statistics_with_temp_cytb-1">
Past versions of statistics_with_temp_cytb-1.png
</button>
</p>
<div id="fig-statistics_with_temp_cytb-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mitoclub/TemperatureEffectsOnMtDNAspectra/blob/77db01ef622b2f1f41cab6356064e0562a9a41e8/docs/figure/I_CYTB_polymorphic_mutspec_analysis.Rmd/statistics_with_temp_cytb-1.png" target="_blank">77db01e</a>
</td>
<td>
Alya Mikhailova
</td>
<td>
2025-06-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="a_g-freqs-dependence-on-climate-zones" class="section level2">
<h2>A_G freqs dependence on climate zones</h2>
<pre class="r"><code>ClimateZonetest &lt;- na.omit(data.frame(subset(MutAndEcoCb, select = c(Species ,A_G.H, Climate_zone))))

ClimateZonetest$ClimateGroup &lt;- ifelse(ClimateZonetest$Climate_zone %in% c(&quot;tropical&quot;, &quot;subtropical&quot;), &quot;Warm&quot;, &quot;Cold&quot;)
ClimateZonetest$Climate_zone &lt;- as.factor(ClimateZonetest$Climate_zone)
AGmedian &lt;- median(ClimateZonetest$A_G.H)
ClimateZonetest$A_G.Freq &lt;- &quot;low_A_G&quot;
ClimateZonetest[ClimateZonetest$A_G.H &gt;= AGmedian,]$A_G.Freq &lt;- &quot;high_A_G&quot;
ClimateZonetest$A_G.Freq &lt;- as.factor(ClimateZonetest$A_G.Freq)
ClimateZonePie &lt;- ClimateZonetest[,2-3]
s1 &lt;- ggpiestats(data = ClimateZonePie, x = Climate_zone, y = A_G.Freq) +
  scale_fill_manual(values = c(&quot;#F2AF4AFF&quot;, &quot;#C36377FF&quot;, &quot;#EB7F54FF&quot;, &quot;#61599DFF&quot;, &quot;#1D457FFF&quot;))
s1</code></pre>
<p><img src="figure/I_CYTB_polymorphic_mutspec_analysis.Rmd/comparison_climate_A_G-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-comparison_climate_A_G-1">
Past versions of comparison_climate_A_G-1.png
</button>
</p>
<div id="fig-comparison_climate_A_G-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mitoclub/TemperatureEffectsOnMtDNAspectra/blob/77db01ef622b2f1f41cab6356064e0562a9a41e8/docs/figure/I_CYTB_polymorphic_mutspec_analysis.Rmd/comparison_climate_A_G-1.png" target="_blank">77db01e</a>
</td>
<td>
Alya Mikhailova
</td>
<td>
2025-06-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>tidy(wilcox.test(ClimateZonetest[ClimateZonetest$ClimateGroup == &quot;Warm&quot;,]$A_G.H, ClimateZonetest[ClimateZonetest$ClimateGroup == &quot;Cold&quot;,]$A_G.H))</code></pre>
<pre><code># A tibble: 1 × 4
  statistic    p.value method                                            alternative
      &lt;dbl&gt;      &lt;dbl&gt; &lt;chr&gt;                                             &lt;chr&gt;      
1    17568. 0.00000467 Wilcoxon rank sum test with continuity correction two.sided  </code></pre>
</div>
<div id="ag-freqs-dependence-on-time-of-maturation"
class="section level2">
<h2>A&gt;G freqs dependence on time of maturation</h2>
<pre class="r"><code>correlation_results1 &lt;- MutAndEcoCb |&gt;
  dplyr::select(
    &quot;A_G.H&quot;,
    &quot;A_T.H&quot;,
    &quot;A_C.H&quot;,
    &quot;T_A.H&quot;,
    &quot;T_G.H&quot;,
    &quot;T_C.H&quot;,
    &quot;C_A.H&quot;,
    &quot;C_T.H&quot;,
    &quot;C_G.H&quot;,
    &quot;G_T.H&quot;,
    &quot;G_A.H&quot;,
    &quot;G_C.H&quot;
  ) |&gt;
  map_dfr( ~ tidy(cor.test(.x, MutAndEcoCb$tm), method = &#39;spearman&#39;), .id = &quot;variable&quot;) |&gt;
  arrange(desc(abs(estimate)))
correlation_results1</code></pre>
<pre><code># A tibble: 12 × 9
   variable estimate statistic p.value parameter conf.low conf.high method                               alternative
   &lt;chr&gt;       &lt;dbl&gt;     &lt;dbl&gt;   &lt;dbl&gt;     &lt;int&gt;    &lt;dbl&gt;     &lt;dbl&gt; &lt;chr&gt;                                &lt;chr&gt;      
 1 G_A.H     0.140      1.90    0.0596       181 -0.00562    0.279  Pearson&#39;s product-moment correlation two.sided  
 2 T_C.H    -0.124     -1.68    0.0951       181 -0.264      0.0217 Pearson&#39;s product-moment correlation two.sided  
 3 T_G.H    -0.0766    -1.03    0.302        181 -0.219      0.0692 Pearson&#39;s product-moment correlation two.sided  
 4 C_A.H    -0.0740    -0.999   0.319        181 -0.217      0.0718 Pearson&#39;s product-moment correlation two.sided  
 5 T_A.H    -0.0648    -0.873   0.384        181 -0.208      0.0810 Pearson&#39;s product-moment correlation two.sided  
 6 A_G.H     0.0644     0.868   0.386        181 -0.0814     0.208  Pearson&#39;s product-moment correlation two.sided  
 7 C_G.H    -0.0565    -0.762   0.447        181 -0.200      0.0893 Pearson&#39;s product-moment correlation two.sided  
 8 C_T.H     0.0557     0.750   0.454        181 -0.0901     0.199  Pearson&#39;s product-moment correlation two.sided  
 9 G_T.H     0.0375     0.505   0.614        181 -0.108      0.182  Pearson&#39;s product-moment correlation two.sided  
10 G_C.H    -0.0298    -0.401   0.689        181 -0.174      0.116  Pearson&#39;s product-moment correlation two.sided  
11 A_T.H    -0.0237    -0.319   0.750        181 -0.168      0.122  Pearson&#39;s product-moment correlation two.sided  
12 A_C.H    -0.00107   -0.0144  0.989        181 -0.146      0.144  Pearson&#39;s product-moment correlation two.sided  </code></pre>
</div>
<div id="pgls" class="section level2">
<h2>PGLS</h2>
<pre class="r"><code>tree &lt;- read.tree(file = here(data_dir, &quot;chordates_species.nwk&quot;))
tree &lt;- ape::drop.tip(tree, &quot;Tenebrosternarchus_preto&quot;)
MutAndEcoPGLS &lt;- MutAndEcoCb %&gt;% dplyr::select(Species, A_G.H, C_T.H, G_A.H, T_C.H, Temp_avg)

row.names(MutAndEcoPGLS) = MutAndEcoPGLS$Species
tree_pruned = treedata(tree, MutAndEcoPGLS, sort=T, warnings=T)$phy
data&lt;-as.data.frame(treedata(tree_pruned, MutAndEcoPGLS, sort=T, warnings=T)$data)
data$Species = as.character(data$Species)
data$A_G.H = as.numeric(data$A_G.H)
data$C_T.H = as.numeric(data$C_T.H)
data$G_A.H = as.numeric(data$G_A.H)
data$T_C.H = as.numeric(data$T_C.H)
data$Temp_avg = as.numeric(data$Temp_avg)
data$Tv &lt;- 1-(data$A_G.H + data$G_A.H + data$T_C.H + data$C_T.H)
data_comp &lt;- comparative.data(tree_pruned, data, Species, vcv=TRUE)

summary(pgls(A_G.H ~ Temp_avg, data_comp, lambda=&quot;ML&quot;))</code></pre>
<pre><code>
Call:
pgls(formula = A_G.H ~ Temp_avg, data = data_comp, lambda = &quot;ML&quot;)

Residuals:
       Min         1Q     Median         3Q        Max 
-0.0081580 -0.0024766 -0.0007092  0.0012258  0.0165195 

Branch length transformations:

kappa  [Fix]  : 1.000
lambda [ ML]  : 0.000
   lower bound : 0.000, p = 1    
   upper bound : 1.000, p = &lt; 2.22e-16
   95.0% CI   : (NA, 0.119)
delta  [Fix]  : 1.000

Coefficients:
             Estimate Std. Error t value  Pr(&gt;|t|)    
(Intercept) 0.0682919  0.0243949  2.7994  0.005802 ** 
Temp_avg    0.0053567  0.0012733  4.2069 4.467e-05 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 0.004066 on 148 degrees of freedom
Multiple R-squared: 0.1068, Adjusted R-squared: 0.1008 
F-statistic:  17.7 on 1 and 148 DF,  p-value: 4.467e-05 </code></pre>
<pre class="r"><code>#la.A_G &lt;- phylosig(tree_pruned, data$A_G.H, method = &quot;lambda&quot;, test = TRUE) #Phylogenetic signal lambda : 0.280122 
#la.C_T &lt;- phylosig(tree_pruned, data$C_T.H, method = &quot;lambda&quot;, test = TRUE) #Phylogenetic signal lambda : 0.119451 
#la.G_A &lt;- phylosig(tree_pruned, data$G_A.H, method = &quot;lambda&quot;, test = TRUE) #Phylogenetic signal lambda : 0.719915 
#la.T_C &lt;- phylosig(tree_pruned, data$T_C.H, method = &quot;lambda&quot;, test = TRUE) #Phylogenetic signal lambda : 0.813846 
#la.Tv &lt;- phylosig(tree_pruned, data$Tv, method = &quot;lambda&quot;, test = TRUE) #Phylogenetic signal lambda : 0.113722
#la.temp &lt;- phylosig(tree_pruned, data$Temp_avg, method = &quot;lambda&quot;, test = TRUE) #Phylogenetic signal lambda : 0.92032</code></pre>
<pre class="r"><code>tree &lt;- read.tree(file = here(data_dir, &quot;chordates_species.nwk&quot;))
tree &lt;- ape::drop.tip(tree, &quot;Tenebrosternarchus_preto&quot;)

row.names(ClimateZonetest) = ClimateZonetest$Species
tree_pruned = treedata(tree, ClimateZonetest, sort=T, warnings=T)$phy
data&lt;-as.data.frame(treedata(tree_pruned, ClimateZonetest, sort=T, warnings=T)$data)
data$Species = as.character(data$Species)
data$A_G.H = as.numeric(data$A_G.H)
data$ClimateGroup = as.factor(data$ClimateGroup)
data_comp &lt;- comparative.data(tree_pruned, data, Species, vcv=TRUE)

summary(pgls(A_G.H ~ ClimateGroup, data_comp, lambda=&quot;ML&quot;))</code></pre>
<pre><code>
Call:
pgls(formula = A_G.H ~ ClimateGroup, data = data_comp, lambda = &quot;ML&quot;)

Residuals:
       Min         1Q     Median         3Q        Max 
-0.0080064 -0.0026449 -0.0007283  0.0013327  0.0172939 

Branch length transformations:

kappa  [Fix]  : 1.000
lambda [ ML]  : 0.000
   lower bound : 0.000, p = 1    
   upper bound : 1.000, p = &lt; 2.22e-16
   95.0% CI   : (NA, 0.368)
delta  [Fix]  : 1.000

Coefficients:
                  Estimate Std. Error t value  Pr(&gt;|t|)    
(Intercept)      0.1375144  0.0077634 17.7133 &lt; 2.2e-16 ***
ClimateGroupWarm 0.0346512  0.0103527  3.3471  0.000917 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 0.004226 on 311 degrees of freedom
Multiple R-squared: 0.03477,    Adjusted R-squared: 0.03167 
F-statistic:  11.2 on 1 and 311 DF,  p-value: 0.000917 </code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.5.0 (2025-04-11)
Platform: aarch64-apple-darwin20
Running under: macOS Sequoia 15.5

Matrix products: default
BLAS:   /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.5-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.1

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Vienna
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] RColorBrewer_1.1-3 skimr_2.1.5        ggrepel_0.9.6      rfishbase_5.0.1    ggstatsplot_0.13.1 broom_1.0.8       
 [7] ggpubr_0.6.0       ggExtra_0.10.1     geiger_2.0.11      phytools_2.4-4     maps_3.4.3         caper_1.0.3       
[13] mvtnorm_1.3-3      MASS_7.3-65        ape_5.8-1          magrittr_2.0.3     lubridate_1.9.4    forcats_1.0.0     
[19] stringr_1.5.1      dplyr_1.1.4        purrr_1.0.4        readr_2.1.5        tidyr_1.3.1        tibble_3.3.0      
[25] ggplot2_3.5.2      tidyverse_2.0.0    knitr_1.50         here_1.0.1         workflowr_1.7.1   

loaded via a namespace (and not attached):
  [1] rstudioapi_0.17.1       jsonlite_2.0.0          datawizard_1.1.0        correlation_0.8.7       subplex_1.9            
  [6] SuppDists_1.1-9.9       estimability_1.5.1      farver_2.1.2            rmarkdown_2.29          ragg_1.4.0             
 [11] fs_1.6.6                vctrs_0.6.5             memoise_2.0.1           paletteer_1.6.0         base64enc_0.1-3        
 [16] effectsize_1.0.1        rstatix_0.7.2           htmltools_0.5.8.1       DEoptim_2.2-8           deSolve_1.40           
 [21] BWStest_0.2.3           Formula_1.2-5           sass_0.4.10             bslib_0.9.0             fontawesome_0.5.3      
 [26] emmeans_1.11.1          cachem_1.1.0            whisker_0.4.1           igraph_2.1.4            mime_0.13              
 [31] lifecycle_1.0.4         iterators_1.0.14        pkgconfig_2.0.3         Matrix_1.7-3            R6_2.6.1               
 [36] fastmap_1.2.0           PMCMRplus_1.9.12        shiny_1.10.0            BayesFactor_0.9.12-4.7  digest_0.6.37          
 [41] numDeriv_2016.8-1.1     rematch2_2.1.2          patchwork_1.3.0         ps_1.9.1                rprojroot_2.0.4        
 [46] textshaping_1.0.1       labeling_0.4.3          clusterGeneration_1.3.8 timechange_0.3.0        mgcv_1.9-3             
 [51] httr_1.4.7              abind_1.4-8             compiler_4.5.0          bit64_4.6.0-1           withr_3.0.2            
 [56] doParallel_1.0.17       backports_1.5.0         optimParallel_1.0-2     carData_3.0-5           performance_0.14.0     
 [61] ggsignif_0.6.4          scatterplot3d_0.3-44    tools_4.5.0             httpuv_1.6.16           statsExpressions_1.7.0 
 [66] glue_1.8.0              quadprog_1.5-8          callr_3.7.6             nlme_3.1-168            promises_1.3.3         
 [71] grid_4.5.0              rsconnect_1.4.1         getPass_0.2-4           generics_0.1.4          gtable_0.3.6           
 [76] tzdb_0.5.0              hms_1.1.3               car_3.1-3               utf8_1.2.6              foreach_1.5.2          
 [81] pillar_1.10.2           vroom_1.6.5             later_1.4.2             splines_4.5.0           lattice_0.22-7         
 [86] gmp_0.7-5               bit_4.6.0               tidyselect_1.2.1        pbapply_1.7-2           miniUI_0.1.2           
 [91] git2r_0.36.2            xfun_0.52               expm_1.0-0              stringi_1.8.7           boot_1.3-31            
 [96] yaml_2.3.10             kSamples_1.2-10         evaluate_1.0.3          codetools_0.2-20        multcompView_0.1-10    
[101] cli_3.6.5               RcppParallel_5.1.10     systemfonts_1.2.3       xtable_1.8-4            parameters_0.26.0      
[106] repr_1.1.7              processx_3.8.6          jquerylib_0.1.4         Rcpp_1.0.14             zeallot_0.2.0          
[111] coda_0.19-4.1           parallel_4.5.0          MatrixModels_0.5-4      rstantools_2.4.0        bayestestR_0.16.0      
[116] Rmpfr_1.1-0             phangorn_2.12.1         scales_1.4.0            insight_1.3.0           crayon_1.5.3           
[121] combinat_0.0-8          rlang_1.1.6             fastmatch_1.1-6         cowplot_1.1.3           mnormt_2.1.1           </code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
