<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Alina G. Mikhailova" />

<meta name="date" content="2025-06-19" />

<title>Analysis of cytochrome b mutational spectra in fishes</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">TemperatureEffectsOnMtDNAspectra</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="I_CYTB_polymorphic_mutspec_analysis.html">Analysis of cytochrome b mutational spectra in fishes</a>
    </li>
    <li>
      <a href="Ia_Polymorphic_Data_Analysis_TrophLevel.KP.html">Analysis of the Cyt B neutral nucleotide data in Fishes</a>
    </li>
    <li>
      <a href="II_Cytb_Data_Analysis.html">Analysis of the Cyt B neutral nucleotide data in Fishes</a>
    </li>
    <li>
      <a href="IIa_Cytb_Data_Analysis_Simulation_Mutagenesis.html">Simulation of the mutagenesis in the Cyt B gene in Fishes</a>
    </li>
    <li>
      <a href="III_Comparison_of_all_Classes.html">SAnalysis of the mitochondrial mutational spectra between different classes</a>
    </li>
  </ul>
</li>
<li>
  <a href="methods.html">Methods</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="license.html">License</a>
</li>
<li>
  <a href="https://github.com/mitoclub/TemperatureEffectsOnMtDNAspectra">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Analysis of cytochrome b mutational spectra
in fishes</h1>
<h4 class="author">Alina G. Mikhailova</h4>
<h4 class="date">2025-06-19</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-06-19
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong>
<code>TemperatureEffectsOnMtDNAspectra/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240719code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240719)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240719code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240719)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcommitoclubTemperatureEffectsOnMtDNAspectratree14d12386eb61777a4587a52337c2d95577dea4dftargetblank14d1238a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/mitoclub/TemperatureEffectsOnMtDNAspectra/tree/14d12386eb61777a4587a52337c2d95577dea4df" target="_blank">14d1238</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcommitoclubTemperatureEffectsOnMtDNAspectratree14d12386eb61777a4587a52337c2d95577dea4dftargetblank14d1238a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/mitoclub/TemperatureEffectsOnMtDNAspectra/tree/14d12386eb61777a4587a52337c2d95577dea4df" target="_blank">14d1238</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/.RData
    Ignored:    analysis/.Rhistory
    Ignored:    code/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    data/Codons_of_CytB_gene_Chordata.txt
    Ignored:    data/ColdAndWarmFishesSpectra.txt
    Ignored:    data/Dataset_S1.csv
    Ignored:    data/ExpectedFractions.txt
    Ignored:    data/Extra_climate_data_2025.csv
    Ignored:    data/Extra_temperature_grow_data_2025.csv
    Ignored:    data/Extra_temperature_popQB_data_2025.csv
    Ignored:    data/Full_fish_ecology_table_2025.csv
    Ignored:    data/MIDORI2_LONGEST_NUC_GB265_Cytb_BLAST.fasta
    Ignored:    data/MIDORI2_LONGEST_NUC_GB265_Cytb_BLAST.fasta.fai
    Ignored:    data/MIDORI_WG_CYTB_FISHES.csv
    Ignored:    data/MolEvolRatesProxy.txt
    Ignored:    data/SimulationData.csv
    Ignored:    data/SimulationData2.csv
    Ignored:    data/Species_by_climate_zones.txt
    Ignored:    output/.DS_Store
    Ignored:    output/figures/.DS_Store
    Ignored:    output/tables/.DS_Store

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/I_CYTB_polymorphic_mutspec_analysis.Rmd</code>) and HTML
(<code>docs/I_CYTB_polymorphic_mutspec_analysis.html</code>) files. If
you’ve configured a remote Git repository (see
<code>?wflow_git_remote</code>), click on the hyperlinks in the table
below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mitoclub/TemperatureEffectsOnMtDNAspectra/14d12386eb61777a4587a52337c2d95577dea4df/docs/I_CYTB_polymorphic_mutspec_analysis.html" target="_blank">14d1238</a>
</td>
<td>
Alya Mikhailova
</td>
<td>
2025-06-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mitoclub/TemperatureEffectsOnMtDNAspectra/blob/4c79558748eab7f1d8d614feecd8f12d908c0faf/analysis/I_CYTB_polymorphic_mutspec_analysis.Rmd" target="_blank">4c79558</a>
</td>
<td>
Alya Mikhailova
</td>
<td>
2025-06-19
</td>
<td>
Added some updates
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mitoclub/TemperatureEffectsOnMtDNAspectra/blob/6ece1e577d495976c4a50a8619c489d8402654d3/analysis/I_CYTB_polymorphic_mutspec_analysis.Rmd" target="_blank">6ece1e5</a>
</td>
<td>
Alya Mikhailova
</td>
<td>
2025-06-18
</td>
<td>
First chapter all done
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mitoclub/TemperatureEffectsOnMtDNAspectra/6ece1e577d495976c4a50a8619c489d8402654d3/docs/I_CYTB_polymorphic_mutspec_analysis.html" target="_blank">6ece1e5</a>
</td>
<td>
Alya Mikhailova
</td>
<td>
2025-06-18
</td>
<td>
First chapter all done
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div
id="reading-mutation-spectra-database-l-strand-and-ecological-databases"
class="section level1">
<h1>Reading mutation spectra database (L-strand) and ecological
databases</h1>
<pre class="r"><code>mutSpec12 &lt;- read_csv(here(data_dir, &quot;MutSpecVertebrates12fish.csv&quot;))
mutSpec12Cb &lt;- dplyr::filter(mutSpec12, Gene == &quot;Cytb&quot;)
cat(&quot;N species =&quot;, length(unique(mutSpec12Cb$Species)))</code></pre>
<pre><code>N species = 840</code></pre>
<pre class="r"><code>kable(head(mutSpec12Cb))</code></pre>
<table>
<colgroup>
<col width="5%" />
<col width="13%" />
<col width="21%" />
<col width="4%" />
<col width="10%" />
<col width="12%" />
<col width="9%" />
<col width="11%" />
<col width="11%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">Gene</th>
<th align="left">Class</th>
<th align="left">Species</th>
<th align="left">Mut</th>
<th align="right">Observed</th>
<th align="right">TreeLeaves</th>
<th align="right">MSAsize</th>
<th align="right">Expected</th>
<th align="right">MutSpec</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Cytb</td>
<td align="left">Actinopteri</td>
<td align="left">Scombrops_gilberti</td>
<td align="left">A&gt;C</td>
<td align="right">4.00000</td>
<td align="right">68</td>
<td align="right">1140</td>
<td align="right">70.95588</td>
<td align="right">0.0326476</td>
</tr>
<tr class="even">
<td align="left">Cytb</td>
<td align="left">Actinopteri</td>
<td align="left">Scombrops_gilberti</td>
<td align="left">A&gt;G</td>
<td align="right">21.60250</td>
<td align="right">68</td>
<td align="right">1140</td>
<td align="right">111.14706</td>
<td align="right">0.1125605</td>
</tr>
<tr class="odd">
<td align="left">Cytb</td>
<td align="left">Actinopteri</td>
<td align="left">Scombrops_gilberti</td>
<td align="left">A&gt;T</td>
<td align="right">2.00000</td>
<td align="right">68</td>
<td align="right">1140</td>
<td align="right">70.95588</td>
<td align="right">0.0163238</td>
</tr>
<tr class="even">
<td align="left">Cytb</td>
<td align="left">Actinopteri</td>
<td align="left">Scombrops_gilberti</td>
<td align="left">C&gt;A</td>
<td align="right">3.00000</td>
<td align="right">68</td>
<td align="right">1140</td>
<td align="right">86.32353</td>
<td align="right">0.0201267</td>
</tr>
<tr class="odd">
<td align="left">Cytb</td>
<td align="left">Actinopteri</td>
<td align="left">Scombrops_gilberti</td>
<td align="left">C&gt;G</td>
<td align="right">0.00000</td>
<td align="right">68</td>
<td align="right">1140</td>
<td align="right">86.32353</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="left">Cytb</td>
<td align="left">Actinopteri</td>
<td align="left">Scombrops_gilberti</td>
<td align="left">C&gt;T</td>
<td align="right">10.66127</td>
<td align="right">68</td>
<td align="right">1140</td>
<td align="right">181.58824</td>
<td align="right">0.0340018</td>
</tr>
</tbody>
</table>
<pre class="r"><code>MutSpecForAnalysisCb &lt;- mutSpec12Cb %&gt;% dplyr::select(Class, Species, Mut, MutSpec)
MutSpecForAnalysisCb$Mut &lt;- gsub(&quot;&gt;&quot;, &quot;_&quot;, MutSpecForAnalysisCb$Mut)
MutSpecForAnalysisCb &lt;- MutSpecForAnalysisCb %&gt;% pivot_wider(names_from = Mut, values_from = MutSpec, , values_fn = median) 
colnames(MutSpecForAnalysisCb) &lt;- c(&quot;Class&quot;,     # Only CytB
                                  &quot;Species&quot;, 
                                  &quot;T_G.H&quot;, 
                                  &quot;T_C.H&quot;, 
                                  &quot;T_A.H&quot;, 
                                  &quot;G_T.H&quot;, 
                                  &quot;G_C.H&quot;, 
                                  &quot;G_A.H&quot;, 
                                  &quot;C_T.H&quot;, 
                                  &quot;C_G.H&quot;, 
                                  &quot;C_A.H&quot;, 
                                  &quot;A_T.H&quot;, 
                                  &quot;A_G.H&quot;, 
                                  &quot;A_C.H&quot;)</code></pre>
<pre class="r"><code>MutSpecForAnalysisBoth &lt;- mutSpec12 %&gt;% dplyr::select(Class, Species, Mut, MutSpec)
MutSpecForAnalysisBoth$Mut &lt;- gsub(&quot;&gt;&quot;, &quot;_&quot;, MutSpecForAnalysisBoth$Mut)
MutSpecForAnalysisBoth &lt;- MutSpecForAnalysisBoth %&gt;% pivot_wider(names_from = Mut, values_from = MutSpec, , values_fn = median) 
cat(&quot;N species =&quot;, length(unique(mutSpec12$Species)))</code></pre>
<pre><code>N species = 1035</code></pre>
<pre class="r"><code>colnames(MutSpecForAnalysisBoth) &lt;- c(&quot;Class&quot;,      # Both genes
                                  &quot;Species&quot;, 
                                  &quot;T_G.H&quot;, 
                                  &quot;T_C.H&quot;, 
                                  &quot;T_A.H&quot;, 
                                  &quot;G_T.H&quot;, 
                                  &quot;G_C.H&quot;, 
                                  &quot;G_A.H&quot;, 
                                  &quot;C_T.H&quot;, 
                                  &quot;C_G.H&quot;, 
                                  &quot;C_A.H&quot;, 
                                  &quot;A_T.H&quot;, 
                                  &quot;A_G.H&quot;, 
                                  &quot;A_C.H&quot;)</code></pre>
<pre class="r"><code>fishEcology &lt;- read.csv(here(data_dir, &quot;Full_fish_ecology_table_2025.csv&quot;))
kable(head(fishEcology))</code></pre>
<table style="width:100%;">
<colgroup>
<col width="16%" />
<col width="6%" />
<col width="6%" />
<col width="6%" />
<col width="20%" />
<col width="6%" />
<col width="4%" />
<col width="6%" />
<col width="6%" />
<col width="6%" />
<col width="8%" />
<col width="6%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">Species</th>
<th align="right">Wmax</th>
<th align="right">Lmax</th>
<th align="right">tmax</th>
<th align="left">FeedingType</th>
<th align="right">FoodTroph</th>
<th align="right">tm</th>
<th align="right">Lm</th>
<th align="right">Temp_min</th>
<th align="right">Temp_max</th>
<th align="left">Climate_zone</th>
<th align="right">Temp_avg</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Abbottina_rivularis</td>
<td align="right">NA</td>
<td align="right">9.10000</td>
<td align="right">2.00000</td>
<td align="left">NA</td>
<td align="right">NA</td>
<td align="right">1.000</td>
<td align="right">4.50000</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="left">subtropical</td>
<td align="right">NA</td>
</tr>
<tr class="even">
<td align="left">Abramis_brama</td>
<td align="right">4112.167</td>
<td align="right">46.91111</td>
<td align="right">14.41667</td>
<td align="left">hunting macrofauna (predator)</td>
<td align="right">3.15</td>
<td align="right">4.125</td>
<td align="right">28.75000</td>
<td align="right">10</td>
<td align="right">24</td>
<td align="left">temperate</td>
<td align="right">17</td>
</tr>
<tr class="odd">
<td align="left">Abudefduf_vaigiensis</td>
<td align="right">NA</td>
<td align="right">18.96667</td>
<td align="right">NA</td>
<td align="left">grazing on aquatic plants</td>
<td align="right">3.39</td>
<td align="right">NA</td>
<td align="right">12.00000</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="left">tropical</td>
<td align="right">NA</td>
</tr>
<tr class="even">
<td align="left">Acanthemblemaria_aspera</td>
<td align="right">NA</td>
<td align="right">3.00000</td>
<td align="right">4.00000</td>
<td align="left">hunting macrofauna (predator)</td>
<td align="right">3.12</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="left">NA</td>
<td align="right">NA</td>
</tr>
<tr class="odd">
<td align="left">Acanthocybium_solandri</td>
<td align="right">7200.000</td>
<td align="right">177.88000</td>
<td align="right">9.30000</td>
<td align="left">hunting macrofauna (predator)</td>
<td align="right">4.45</td>
<td align="right">2.000</td>
<td align="right">99.33333</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="left">subtropical</td>
<td align="right">NA</td>
</tr>
<tr class="even">
<td align="left">Acanthopagrus_berda</td>
<td align="right">660.000</td>
<td align="right">56.00000</td>
<td align="right">12.50000</td>
<td align="left">hunting macrofauna (predator)</td>
<td align="right">3.50</td>
<td align="right">NA</td>
<td align="right">21.00000</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="left">tropical</td>
<td align="right">NA</td>
</tr>
</tbody>
</table>
<pre class="r"><code>cor.test(fishEcology$tm, fishEcology$Temp_avg, method = &quot;spearman&quot;)</code></pre>
<pre><code>
    Spearman&#39;s rank correlation rho

data:  fishEcology$tm and fishEcology$Temp_avg
S = 232808, p-value = 0.0002596
alternative hypothesis: true rho is not equal to 0
sample estimates:
       rho 
-0.3558985 </code></pre>
</div>
<div id="calculating-of-average-mutspec-for-fisches"
class="section level1">
<h1>Calculating of average MutSpec for Fisches</h1>
<div id="cytb" class="section level2">
<h2>Cytb</h2>
<pre class="r"><code>AverageMutSpecCb &lt;- mutSpec12Cb %&gt;% group_by(Mut) %&gt;% summarise(meanmutspec = mean(MutSpec, na.rm = TRUE))
AverageMutSpecCb$Mut.H &lt;- c(&quot;T_G.H&quot;, 
                          &quot;T_C.H&quot;, 
                          &quot;T_A.H&quot;, 
                          &quot;G_T.H&quot;, 
                          &quot;G_C.H&quot;, 
                          &quot;G_A.H&quot;, 
                          &quot;C_T.H&quot;, 
                          &quot;C_G.H&quot;, 
                          &quot;C_A.H&quot;, 
                          &quot;A_T.H&quot;, 
                          &quot;A_G.H&quot;, 
                          &quot;A_C.H&quot;)

f1 = ggplot(AverageMutSpecCb,
  aes(
    x = AverageMutSpecCb$Mut.H,
    y = AverageMutSpecCb$meanmutspec,
    fill = AverageMutSpecCb$Mut.H
    )
  ) +
  geom_bar(stat = &quot;identity&quot;) +
  scale_fill_manual(
    values = c(
      &quot;#bdbdbd&quot;,
      &quot;#00a426&quot;,
      &quot;#bdbdbd&quot;,
      &quot;#bdbdbd&quot;,
      &quot;#bdbdbd&quot;,
      &quot;red&quot;,
      &quot;#900000&quot;,
      &quot;#bdbdbd&quot;,
      &quot;#bdbdbd&quot;,
      &quot;#bdbdbd&quot;,
      &quot;darkgreen&quot;,
      &quot;#bdbdbd&quot;
    )
  ) +
  labs(title = &quot;Average mitochondrial mutational spectrum in Fishes (cytB, N=840)&quot;, x = &quot;Substitution types&quot;, y = &quot;Normalised Frequencies&quot;) + theme(
    legend.position = &quot;none&quot;,
    panel.background = element_blank(),
    axis.line = element_line(color = &quot;black&quot;)
  )

ggsave(here(plots_dir,&quot;Average_mutspec_in_Fishes_CytB.pdf&quot;),
  plot = f1,
  width = 6,
  height = 6
)
f1</code></pre>
<p><img src="figure/I_CYTB_polymorphic_mutspec_analysis.Rmd/average_mutspec_plot_cytb-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-average_mutspec_plot_cytb-1">
Past versions of average_mutspec_plot_cytb-1.png
</button>
</p>
<div id="fig-average_mutspec_plot_cytb-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mitoclub/TemperatureEffectsOnMtDNAspectra/blob/6ece1e577d495976c4a50a8619c489d8402654d3/docs/figure/I_CYTB_polymorphic_mutspec_analysis.Rmd/average_mutspec_plot_cytb-1.png" target="_blank">6ece1e5</a>
</td>
<td>
Alya Mikhailova
</td>
<td>
2025-06-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="both-genes" class="section level2">
<h2>Both genes</h2>
<pre class="r"><code>AverageMutSpecBoth &lt;- mutSpec12 %&gt;% group_by(Mut) %&gt;% summarise(meanmutspec = mean(MutSpec, na.rm = TRUE))
AverageMutSpecBoth$Mut.H &lt;- c(&quot;T_G.H&quot;, 
                          &quot;T_C.H&quot;, 
                          &quot;T_A.H&quot;, 
                          &quot;G_T.H&quot;, 
                          &quot;G_C.H&quot;, 
                          &quot;G_A.H&quot;, 
                          &quot;C_T.H&quot;, 
                          &quot;C_G.H&quot;, 
                          &quot;C_A.H&quot;, 
                          &quot;A_T.H&quot;, 
                          &quot;A_G.H&quot;, 
                          &quot;A_C.H&quot;)

f1 = ggplot(AverageMutSpecBoth,
  aes(
    x = AverageMutSpecBoth$Mut.H,
    y = AverageMutSpecBoth$meanmutspec,
    fill = AverageMutSpecBoth$Mut.H
    )
  ) +
  geom_bar(stat = &quot;identity&quot;) +
  scale_fill_manual(
    values = c(
      &quot;#bdbdbd&quot;,
      &quot;#00a426&quot;,
      &quot;#bdbdbd&quot;,
      &quot;#bdbdbd&quot;,
      &quot;#bdbdbd&quot;,
      &quot;#ff0000&quot;,
      &quot;#900000&quot;,
      &quot;#bdbdbd&quot;,
      &quot;#bdbdbd&quot;,
      &quot;#bdbdbd&quot;,
      &quot;#005213&quot;,
      &quot;#bdbdbd&quot;
    )
  ) +
  labs(title = &quot;Average mitochondrial mutational spectrum in Fishes (cytB+COX1, N=1035)&quot;, x = &quot;Substitution types&quot;, y = &quot;Normalised Frequencies&quot;) + theme(
    legend.position = &quot;none&quot;,
    # Remove legend
    panel.background = element_blank(),
    axis.line = element_line(color = &quot;black&quot;)# Remove background
  )

f1</code></pre>
<p><img src="figure/I_CYTB_polymorphic_mutspec_analysis.Rmd/average_mutspec_plot_both-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-average_mutspec_plot_both-1">
Past versions of average_mutspec_plot_both-1.png
</button>
</p>
<div id="fig-average_mutspec_plot_both-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mitoclub/TemperatureEffectsOnMtDNAspectra/blob/6ece1e577d495976c4a50a8619c489d8402654d3/docs/figure/I_CYTB_polymorphic_mutspec_analysis.Rmd/average_mutspec_plot_both-1.png" target="_blank">6ece1e5</a>
</td>
<td>
Alya Mikhailova
</td>
<td>
2025-06-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="comparison-of-average-mutspec-between-cold-and-warm-fisches"
class="section level1">
<h1>Comparison of average MutSpec between cold and warm fisches</h1>
<pre class="r"><code>merging &lt;- list(MutSpecForAnalysisCb, fishEcology)
MutAndEcoCb &lt;- merging %&gt;% purrr::reduce(left_join, by= &quot;Species&quot;)

#write.table(MutAndEcoCb, file = here(tables_dir, &quot;Supplementary_Material_1_Mitochondrial_mutational_spectra_cytb_and_ecology_Fishes.txt&quot;), row.names = FALSE)
#write_csv(MutAndEcoCb, file = here(tables_dir, &quot;Supplementary_Material_1_Mitochondrial_mutational_spectra_cytb_and_ecology_Fishes.csv&quot;))

temperaturestat &lt;- summary(MutAndEcoCb$Temp_avg)

WarmFishesCb &lt;- filter(MutAndEcoCb, Temp_avg &gt; as.numeric(temperaturestat[&quot;3rd Qu.&quot;]))
WarmFishesCb &lt;- WarmFishesCb %&gt;% dplyr::select(Species, T_G.H, 
                                        T_C.H, 
                                        T_A.H, 
                                        G_T.H, 
                                        G_C.H, 
                                        G_A.H, 
                                        C_T.H, 
                                        C_G.H, 
                                        C_A.H, 
                                        A_T.H, 
                                        A_G.H, 
                                        A_C.H)
WarmFishesCb &lt;- pivot_longer(WarmFishesCb, cols = c(&quot;T_G.H&quot;, 
                                  &quot;T_C.H&quot;, 
                                  &quot;T_A.H&quot;, 
                                  &quot;G_T.H&quot;, 
                                  &quot;G_C.H&quot;, 
                                  &quot;G_A.H&quot;, 
                                  &quot;C_T.H&quot;, 
                                  &quot;C_G.H&quot;, 
                                  &quot;C_A.H&quot;, 
                                  &quot;A_T.H&quot;, 
                                  &quot;A_G.H&quot;, 
                                  &quot;A_C.H&quot;), names_to = &quot;Mut&quot;, values_to = &quot;MutSpec&quot; )
NwarmCb &lt;- length(unique(WarmFishesCb$Species))

ColdFishesCb &lt;- filter(MutAndEcoCb, Temp_avg &lt; as.numeric(temperaturestat[&quot;1st Qu.&quot;]))
ColdFishesCb &lt;- ColdFishesCb %&gt;% dplyr::select(Species, T_G.H, 
                                        T_C.H, 
                                        T_A.H, 
                                        G_T.H, 
                                        G_C.H, 
                                        G_A.H, 
                                        C_T.H, 
                                        C_G.H, 
                                        C_A.H, 
                                        A_T.H, 
                                        A_G.H, 
                                        A_C.H)
ColdFishesCb &lt;- pivot_longer(ColdFishesCb, cols = c(&quot;T_G.H&quot;, 
                                  &quot;T_C.H&quot;, 
                                  &quot;T_A.H&quot;, 
                                  &quot;G_T.H&quot;, 
                                  &quot;G_C.H&quot;, 
                                  &quot;G_A.H&quot;, 
                                  &quot;C_T.H&quot;, 
                                  &quot;C_G.H&quot;, 
                                  &quot;C_A.H&quot;, 
                                  &quot;A_T.H&quot;, 
                                  &quot;A_G.H&quot;, 
                                  &quot;A_C.H&quot;), names_to = &quot;Mut&quot;, values_to = &quot;MutSpec&quot; )
NcoldCb &lt;- length(unique(ColdFishesCb$Species))

f1a &lt;- ggplot(
  WarmFishesCb,
  aes(
    x = Mut,
    y = MutSpec
  )
) +
  geom_jitter(aes(colour = Mut), width = 0.1) +
  scale_color_manual(
    values = c(
      &quot;#bdbdbd&quot;,
      &quot;#00a426&quot;,
      &quot;#bdbdbd&quot;,
      &quot;#bdbdbd&quot;,
      &quot;#bdbdbd&quot;,
      &quot;red&quot;,
      &quot;#900000&quot;,
      &quot;#bdbdbd&quot;,
      &quot;#bdbdbd&quot;,
      &quot;#bdbdbd&quot;,
      &quot;darkgreen&quot;,
      &quot;#bdbdbd&quot;
    )
  ) +
  labs(
    title = &quot;Warm Fishes, &gt;23C&quot;,
    x = &quot;Substitution types&quot;,
    y = &quot;Normalised Frequencies&quot;
  ) +
  theme(
    legend.position = &quot;none&quot;,
    panel.background = element_blank(),
    axis.line = element_line(color = &quot;black&quot;)
  ) +
  geom_text(
    x = 2.5,
    y = 0.63,
    label = paste(&quot;N warm =&quot;, NwarmCb),
    size = 7
  ) +
  stat_summary(
    fun = &quot;median&quot;,
    geom = &quot;line&quot;,        
    aes(group = 2),      
    color = &quot;black&quot;,     
    size = 0.5             
  )


f1b &lt;- ggplot(
 ColdFishesCb,
  aes(
    x = Mut,
    y = MutSpec
  )
) +
  geom_jitter(aes(colour = Mut), width = 0.1) +
  scale_color_manual(
    values = c(
      &quot;#bdbdbd&quot;,
      &quot;#00a426&quot;,
      &quot;#bdbdbd&quot;,
      &quot;#bdbdbd&quot;,
      &quot;#bdbdbd&quot;,
      &quot;red&quot;,
      &quot;#900000&quot;,
      &quot;#bdbdbd&quot;,
      &quot;#bdbdbd&quot;,
      &quot;#bdbdbd&quot;,
      &quot;darkgreen&quot;,
      &quot;#bdbdbd&quot;
    )
  ) +
  labs(
    title = &quot;Cold Fishes, &lt;13C&quot;,
    x = &quot;Substitution types&quot;,
    y = &quot;Normalised Frequencies&quot;
  ) +
  theme(
    legend.position = &quot;none&quot;,
    panel.background = element_blank(),
    axis.line = element_line(color = &quot;black&quot;)
  ) +
  geom_text(
    x = 2.5,
    y = 0.56,
    label = paste(&quot;N cold =&quot;, NcoldCb),
    size = 7
  ) +
  stat_summary(
    fun = &quot;median&quot;,
    geom = &quot;line&quot;,        
    aes(group = 2),      
    color = &quot;black&quot;,     
    size = 0.5             
  )

f1 &lt;- ggarrange(f1b, f1a, ncol = 2)

ggsave(here(plots_dir,&quot;Average mitochondrial mutational spectrum in Fishes divided by groups (cytB).pdf&quot;),
  plot = f1,
  width = 12,
  height = 6
)
f1</code></pre>
<p><img src="figure/I_CYTB_polymorphic_mutspec_analysis.Rmd/average_mutspec_cytb_groups-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-average_mutspec_cytb_groups-1">
Past versions of average_mutspec_cytb_groups-1.png
</button>
</p>
<div id="fig-average_mutspec_cytb_groups-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mitoclub/TemperatureEffectsOnMtDNAspectra/blob/6ece1e577d495976c4a50a8619c489d8402654d3/docs/figure/I_CYTB_polymorphic_mutspec_analysis.Rmd/average_mutspec_cytb_groups-1.png" target="_blank">6ece1e5</a>
</td>
<td>
Alya Mikhailova
</td>
<td>
2025-06-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="studies-of-a_g-spectra-with-temperature-and-other-parameters-in-fishes"
class="section level1">
<h1>Studies of A_G spectra with temperature and other parameters in
fishes</h1>
<div id="a_g-freqs-dependence-on-temperature" class="section level2">
<h2>A_G freqs dependence on temperature</h2>
<pre class="r"><code>correlation_results &lt;- MutAndEcoCb |&gt;
  dplyr::select(
    &quot;A_G.H&quot;,
    &quot;A_T.H&quot;,
    &quot;A_C.H&quot;,
    &quot;T_A.H&quot;,
    &quot;T_G.H&quot;,
    &quot;T_C.H&quot;,
    &quot;C_A.H&quot;,
    &quot;C_T.H&quot;,
    &quot;C_G.H&quot;,
    &quot;G_T.H&quot;,
    &quot;G_A.H&quot;,
    &quot;G_C.H&quot;
  ) |&gt;
  map_dfr( ~ tidy(cor.test(.x, MutAndEcoCb$Temp_avg), method = &#39;spearman&#39;), .id = &quot;variable&quot;) |&gt;
  arrange(desc(abs(estimate)))

kable(correlation_results)</code></pre>
<table>
<colgroup>
<col width="7%" />
<col width="9%" />
<col width="9%" />
<col width="8%" />
<col width="8%" />
<col width="9%" />
<col width="9%" />
<col width="30%" />
<col width="9%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">variable</th>
<th align="right">estimate</th>
<th align="right">statistic</th>
<th align="right">p.value</th>
<th align="right">parameter</th>
<th align="right">conf.low</th>
<th align="right">conf.high</th>
<th align="left">method</th>
<th align="left">alternative</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">A_G.H</td>
<td align="right">0.3168787</td>
<td align="right">4.1594615</td>
<td align="right">0.0000527</td>
<td align="right">155</td>
<td align="right">0.1686096</td>
<td align="right">0.4511251</td>
<td align="left">Pearson’s product-moment correlation</td>
<td align="left">two.sided</td>
</tr>
<tr class="even">
<td align="left">T_C.H</td>
<td align="right">-0.1682837</td>
<td align="right">-2.1254273</td>
<td align="right">0.0351359</td>
<td align="right">155</td>
<td align="right">-0.3165770</td>
<td align="right">-0.0119609</td>
<td align="left">Pearson’s product-moment correlation</td>
<td align="left">two.sided</td>
</tr>
<tr class="odd">
<td align="left">G_A.H</td>
<td align="right">0.1233894</td>
<td align="right">1.5480154</td>
<td align="right">0.1236579</td>
<td align="right">155</td>
<td align="right">-0.0339040</td>
<td align="right">0.2747180</td>
<td align="left">Pearson’s product-moment correlation</td>
<td align="left">two.sided</td>
</tr>
<tr class="even">
<td align="left">A_T.H</td>
<td align="right">-0.1191124</td>
<td align="right">-1.4935699</td>
<td align="right">0.1373209</td>
<td align="right">155</td>
<td align="right">-0.2706999</td>
<td align="right">0.0382392</td>
<td align="left">Pearson’s product-moment correlation</td>
<td align="left">two.sided</td>
</tr>
<tr class="odd">
<td align="left">C_G.H</td>
<td align="right">-0.1154942</td>
<td align="right">-1.4475779</td>
<td align="right">0.1497547</td>
<td align="right">155</td>
<td align="right">-0.2672967</td>
<td align="right">0.0419020</td>
<td align="left">Pearson’s product-moment correlation</td>
<td align="left">two.sided</td>
</tr>
<tr class="even">
<td align="left">T_A.H</td>
<td align="right">0.0911624</td>
<td align="right">1.1397078</td>
<td align="right">0.2561662</td>
<td align="right">155</td>
<td align="right">-0.0664243</td>
<td align="right">0.2443118</td>
<td align="left">Pearson’s product-moment correlation</td>
<td align="left">two.sided</td>
</tr>
<tr class="odd">
<td align="left">T_G.H</td>
<td align="right">0.0798204</td>
<td align="right">0.9969366</td>
<td align="right">0.3203491</td>
<td align="right">155</td>
<td align="right">-0.0777904</td>
<td align="right">0.2335386</td>
<td align="left">Pearson’s product-moment correlation</td>
<td align="left">two.sided</td>
</tr>
<tr class="even">
<td align="left">A_C.H</td>
<td align="right">-0.0498000</td>
<td align="right">-0.6207752</td>
<td align="right">0.5356593</td>
<td align="right">155</td>
<td align="right">-0.2048403</td>
<td align="right">0.1076781</td>
<td align="left">Pearson’s product-moment correlation</td>
<td align="left">two.sided</td>
</tr>
<tr class="odd">
<td align="left">C_A.H</td>
<td align="right">-0.0488585</td>
<td align="right">-0.6090109</td>
<td align="right">0.5434088</td>
<td align="right">155</td>
<td align="right">-0.2039359</td>
<td align="right">0.1086108</td>
<td align="left">Pearson’s product-moment correlation</td>
<td align="left">two.sided</td>
</tr>
<tr class="even">
<td align="left">C_T.H</td>
<td align="right">-0.0413337</td>
<td align="right">-0.5150406</td>
<td align="right">0.6072583</td>
<td align="right">155</td>
<td align="right">-0.1966983</td>
<td align="right">0.1160558</td>
<td align="left">Pearson’s product-moment correlation</td>
<td align="left">two.sided</td>
</tr>
<tr class="odd">
<td align="left">G_C.H</td>
<td align="right">-0.0364204</td>
<td align="right">-0.4537317</td>
<td align="right">0.6506566</td>
<td align="right">155</td>
<td align="right">-0.1919634</td>
<td align="right">0.1209075</td>
<td align="left">Pearson’s product-moment correlation</td>
<td align="left">two.sided</td>
</tr>
<tr class="even">
<td align="left">G_T.H</td>
<td align="right">-0.0325314</td>
<td align="right">-0.4052270</td>
<td align="right">0.6858694</td>
<td align="right">155</td>
<td align="right">-0.1882105</td>
<td align="right">0.1247424</td>
<td align="left">Pearson’s product-moment correlation</td>
<td align="left">two.sided</td>
</tr>
</tbody>
</table>
<pre class="r"><code>#write.table(kable(correlation_results), file = here(tables_dir, &quot;Table1.txt&quot;), row.names = FALSE, col.names = FALSE)

SubsetClass&lt;- unique(MutAndEcoCb$Class)
spN &lt;- nrow(MutAndEcoCb[!is.na(MutAndEcoCb$Temp_avg),])
cat(&quot;N species =&quot;, spN, &quot;Taxa =&quot;, SubsetClass)</code></pre>
<pre><code>N species = 157 Taxa = Actinopteri Chondrichthyes</code></pre>
<pre class="r"><code>f1 &lt;- ggplot(MutAndEcoCb, aes(x = Temp_avg, y = A_G.H)) +
  geom_point(color = &quot;#73514f&quot;) +                  # Scatter plot
  geom_smooth(
    method = &quot;lm&quot;,
    color = &quot;black&quot;,
    # Regression line + CI
    fill = &quot;lightgray&quot;,
    se = TRUE
  ) +     # Set se=TRUE for confidence interval
  stat_cor(
    method = &quot;spearman&quot;,
    # Spearman correlation stats
    aes(label = paste(..r.label.., ..p.label.., &quot;N&quot;, spN, sep = &quot;~&quot;))
  ) +  # Include n
  labs(title = &quot;Positive assosiation of A&gt;G.H with temperature&quot;, x = &quot;Median annual water temperature, ?C&quot;, y = &quot;AH&gt;GH&quot;) +
  theme(
    legend.position = &quot;none&quot;,
    panel.background = element_blank(),
    axis.line = element_line(color = &quot;black&quot;)
  )
 
ggsave(here(plots_dir,&quot;Assosiation_of_A_G_with_temperature.pdf&quot;),
  plot = f1,
  width = 6,
  height = 6
) 
f1</code></pre>
<p><img src="figure/I_CYTB_polymorphic_mutspec_analysis.Rmd/statistics_with_temp_cytb-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-statistics_with_temp_cytb-1">
Past versions of statistics_with_temp_cytb-1.png
</button>
</p>
<div id="fig-statistics_with_temp_cytb-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mitoclub/TemperatureEffectsOnMtDNAspectra/blob/6ece1e577d495976c4a50a8619c489d8402654d3/docs/figure/I_CYTB_polymorphic_mutspec_analysis.Rmd/statistics_with_temp_cytb-1.png" target="_blank">6ece1e5</a>
</td>
<td>
Alya Mikhailova
</td>
<td>
2025-06-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="a_g-freqs-dependence-on-climate-zones" class="section level2">
<h2>A_G freqs dependence on climate zones</h2>
<pre class="r"><code>ClimateZonetest &lt;- na.omit(data.frame(subset(MutAndEcoCb, select = c(Species ,A_G.H, Climate_zone))))

ClimateZonetest$ClimateGroup &lt;- ifelse(ClimateZonetest$Climate_zone %in% c(&quot;tropical&quot;, &quot;subtropical&quot;), &quot;Warm&quot;, &quot;Cold&quot;)
ClimateZonetest$Climate_zone &lt;- factor(ClimateZonetest$Climate_zone, levels = c(&quot;tropical&quot;, &quot;subtropical&quot;, &quot;temperate&quot;, &quot;boreal&quot;, &quot;polar&quot;), ordered = TRUE)
AGmedian &lt;- median(ClimateZonetest$A_G.H)
ClimateZonetest$A_G.Freq &lt;- &quot;low_A_G&quot;
ClimateZonetest[ClimateZonetest$A_G.H &gt;= AGmedian,]$A_G.Freq &lt;- &quot;high_A_G&quot;
ClimateZonetest$A_G.Freq &lt;- as.factor(ClimateZonetest$A_G.Freq)
ClimateZonePie &lt;- ClimateZonetest[,2-3]
ClimPl &lt;- c(&quot;#F2AF4AFF&quot;, &quot;#EB7F54FF&quot;, &quot;#C36377FF&quot;, &quot;#61599DFF&quot;, &quot;#1D457FFF&quot;)
names(ClimPl) &lt;- c(&quot;tropical&quot;, &quot;subtropical&quot;, &quot;temperate&quot;, &quot;boreal&quot;, &quot;polar&quot;) 
s1 &lt;- ggpiestats(data = ClimateZonePie, x = Climate_zone, y = A_G.Freq) +
  scale_fill_manual(values = ClimPl)
s1</code></pre>
<p><img src="figure/I_CYTB_polymorphic_mutspec_analysis.Rmd/comparison_climate_A_G-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-comparison_climate_A_G-1">
Past versions of comparison_climate_A_G-1.png
</button>
</p>
<div id="fig-comparison_climate_A_G-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mitoclub/TemperatureEffectsOnMtDNAspectra/blob/14d12386eb61777a4587a52337c2d95577dea4df/docs/figure/I_CYTB_polymorphic_mutspec_analysis.Rmd/comparison_climate_A_G-1.png" target="_blank">14d1238</a>
</td>
<td>
Alya Mikhailova
</td>
<td>
2025-06-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mitoclub/TemperatureEffectsOnMtDNAspectra/blob/6ece1e577d495976c4a50a8619c489d8402654d3/docs/figure/I_CYTB_polymorphic_mutspec_analysis.Rmd/comparison_climate_A_G-1.png" target="_blank">6ece1e5</a>
</td>
<td>
Alya Mikhailova
</td>
<td>
2025-06-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>tidy(wilcox.test(ClimateZonetest[ClimateZonetest$ClimateGroup == &quot;Warm&quot;,]$A_G.H, ClimateZonetest[ClimateZonetest$ClimateGroup == &quot;Cold&quot;,]$A_G.H))</code></pre>
<pre><code># A tibble: 1 × 4
  statistic    p.value method                                        alternative
      &lt;dbl&gt;      &lt;dbl&gt; &lt;chr&gt;                                         &lt;chr&gt;      
1    17568. 0.00000467 Wilcoxon rank sum test with continuity corre… two.sided  </code></pre>
</div>
<div id="ag-freqs-dependence-on-time-of-maturation"
class="section level2">
<h2>A&gt;G freqs dependence on time of maturation</h2>
<pre class="r"><code>correlation_results1 &lt;- MutAndEcoCb |&gt;
  dplyr::select(
    &quot;A_G.H&quot;,
    &quot;A_T.H&quot;,
    &quot;A_C.H&quot;,
    &quot;T_A.H&quot;,
    &quot;T_G.H&quot;,
    &quot;T_C.H&quot;,
    &quot;C_A.H&quot;,
    &quot;C_T.H&quot;,
    &quot;C_G.H&quot;,
    &quot;G_T.H&quot;,
    &quot;G_A.H&quot;,
    &quot;G_C.H&quot;
  ) |&gt;
  map_dfr( ~ tidy(cor.test(.x, MutAndEcoCb$tm), method = &#39;spearman&#39;), .id = &quot;variable&quot;) |&gt;
  arrange(desc(abs(estimate)))
kable(correlation_results1)</code></pre>
<table>
<colgroup>
<col width="7%" />
<col width="9%" />
<col width="9%" />
<col width="8%" />
<col width="8%" />
<col width="9%" />
<col width="8%" />
<col width="30%" />
<col width="9%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">variable</th>
<th align="right">estimate</th>
<th align="right">statistic</th>
<th align="right">p.value</th>
<th align="right">parameter</th>
<th align="right">conf.low</th>
<th align="right">conf.high</th>
<th align="left">method</th>
<th align="left">alternative</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">G_A.H</td>
<td align="right">0.1395474</td>
<td align="right">1.8959691</td>
<td align="right">0.0595575</td>
<td align="right">181</td>
<td align="right">-0.0056231</td>
<td align="right">0.2789573</td>
<td align="left">Pearson’s product-moment correlation</td>
<td align="left">two.sided</td>
</tr>
<tr class="even">
<td align="left">T_C.H</td>
<td align="right">-0.1237639</td>
<td align="right">-1.6779743</td>
<td align="right">0.0950779</td>
<td align="right">181</td>
<td align="right">-0.2640796</td>
<td align="right">0.0216820</td>
<td align="left">Pearson’s product-moment correlation</td>
<td align="left">two.sided</td>
</tr>
<tr class="odd">
<td align="left">T_G.H</td>
<td align="right">-0.0766398</td>
<td align="right">-1.0341248</td>
<td align="right">0.3024575</td>
<td align="right">181</td>
<td align="right">-0.2192589</td>
<td align="right">0.0691860</td>
<td align="left">Pearson’s product-moment correlation</td>
<td align="left">two.sided</td>
</tr>
<tr class="even">
<td align="left">C_A.H</td>
<td align="right">-0.0740346</td>
<td align="right">-0.9987746</td>
<td align="right">0.3192373</td>
<td align="right">181</td>
<td align="right">-0.2167634</td>
<td align="right">0.0717931</td>
<td align="left">Pearson’s product-moment correlation</td>
<td align="left">two.sided</td>
</tr>
<tr class="odd">
<td align="left">T_A.H</td>
<td align="right">-0.0647810</td>
<td align="right">-0.8733736</td>
<td align="right">0.3836167</td>
<td align="right">181</td>
<td align="right">-0.2078842</td>
<td align="right">0.0810372</td>
<td align="left">Pearson’s product-moment correlation</td>
<td align="left">two.sided</td>
</tr>
<tr class="even">
<td align="left">A_G.H</td>
<td align="right">0.0643911</td>
<td align="right">0.8680947</td>
<td align="right">0.3864920</td>
<td align="right">181</td>
<td align="right">-0.0814261</td>
<td align="right">0.2075095</td>
<td align="left">Pearson’s product-moment correlation</td>
<td align="left">two.sided</td>
</tr>
<tr class="odd">
<td align="left">C_G.H</td>
<td align="right">-0.0565248</td>
<td align="right">-0.7616810</td>
<td align="right">0.4472417</td>
<td align="right">181</td>
<td align="right">-0.1999421</td>
<td align="right">0.0892638</td>
<td align="left">Pearson’s product-moment correlation</td>
<td align="left">two.sided</td>
</tr>
<tr class="even">
<td align="left">C_T.H</td>
<td align="right">0.0556726</td>
<td align="right">0.7501620</td>
<td align="right">0.4541311</td>
<td align="right">181</td>
<td align="right">-0.0901117</td>
<td align="right">0.1991212</td>
<td align="left">Pearson’s product-moment correlation</td>
<td align="left">two.sided</td>
</tr>
<tr class="odd">
<td align="left">G_T.H</td>
<td align="right">0.0374879</td>
<td align="right">0.5047025</td>
<td align="right">0.6143817</td>
<td align="right">181</td>
<td align="right">-0.1081569</td>
<td align="right">0.1815572</td>
<td align="left">Pearson’s product-moment correlation</td>
<td align="left">two.sided</td>
</tr>
<tr class="even">
<td align="left">G_C.H</td>
<td align="right">-0.0298235</td>
<td align="right">-0.4014128</td>
<td align="right">0.6885898</td>
<td align="right">181</td>
<td align="right">-0.1741269</td>
<td align="right">0.1157338</td>
<td align="left">Pearson’s product-moment correlation</td>
<td align="left">two.sided</td>
</tr>
<tr class="odd">
<td align="left">A_T.H</td>
<td align="right">-0.0236919</td>
<td align="right">-0.3188316</td>
<td align="right">0.7502219</td>
<td align="right">181</td>
<td align="right">-0.1681706</td>
<td align="right">0.1217833</td>
<td align="left">Pearson’s product-moment correlation</td>
<td align="left">two.sided</td>
</tr>
<tr class="even">
<td align="left">A_C.H</td>
<td align="right">-0.0010672</td>
<td align="right">-0.0143581</td>
<td align="right">0.9885601</td>
<td align="right">181</td>
<td align="right">-0.1461013</td>
<td align="right">0.1440117</td>
<td align="left">Pearson’s product-moment correlation</td>
<td align="left">two.sided</td>
</tr>
</tbody>
</table>
</div>
<div id="pgls" class="section level2">
<h2>PGLS</h2>
<pre class="r"><code>create_enhanced_summary_table &lt;- function(model_summary, pgls_model) {
  # Extract coefficients
  coef_table &lt;- model_summary$coefficients
  
  # Create enhanced dataframe
  enhanced_df &lt;- data.frame(
    Parameter = rownames(coef_table),
    Estimate = round(coef_table[, &quot;Estimate&quot;], 6),
    Std_Error = round(coef_table[, &quot;Std. Error&quot;], 6),
    t_value = round(coef_table[, &quot;t value&quot;], 3),
    p_value = format.pval(coef_table[, &quot;Pr(&gt;|t|)&quot;], digits = 3),
    Significance = case_when(
      coef_table[, &quot;Pr(&gt;|t|)&quot;] &lt; 0.001 ~ &quot;***&quot;,
      coef_table[, &quot;Pr(&gt;|t|)&quot;] &lt; 0.01 ~ &quot;**&quot;,
      coef_table[, &quot;Pr(&gt;|t|)&quot;] &lt; 0.05 ~ &quot;*&quot;,
      coef_table[, &quot;Pr(&gt;|t|)&quot;] &lt; 0.1 ~ &quot;.&quot;,
      TRUE ~ &quot;&quot;
    ),
    CI_Lower = round(coef_table[, &quot;Estimate&quot;] - 1.96 * coef_table[, &quot;Std. Error&quot;], 6),
    CI_Upper = round(coef_table[, &quot;Estimate&quot;] + 1.96 * coef_table[, &quot;Std. Error&quot;], 6)
  )
  
  # Model fit statistics
  model_stats &lt;- data.frame(
    Statistic = c(
      &quot;R-squared&quot;,
      &quot;Adjusted R-squared&quot;,
      &quot;F-statistic&quot;,
      &quot;Lambda&quot;,
      &quot;AIC&quot;,
      &quot;BIC&quot;
    ),
    Value = c(
      round(model_summary$r.squared, 4),
      round(model_summary$adj.r.squared, 4),
      round(model_summary$fstatistic[1], 3),
      round(pgls_model$param[&quot;lambda&quot;], 6),
      round(AIC(pgls_model), 2),
      round(BIC(pgls_model), 2)
    )
  )
  
  return(list(coefficients = enhanced_df, model_stats = model_stats))
}</code></pre>
<pre class="r"><code>tree &lt;- read.tree(file = here(data_dir, &quot;chordates_species.nwk&quot;))
tree &lt;- ape::drop.tip(tree, &quot;Tenebrosternarchus_preto&quot;)
MutAndEcoPGLS &lt;- MutAndEcoCb %&gt;% dplyr::select(Species, A_G.H, C_T.H, G_A.H, T_C.H, Temp_avg)

row.names(MutAndEcoPGLS) = MutAndEcoPGLS$Species
tree_pruned = treedata(tree, MutAndEcoPGLS, sort=T, warnings=T)$phy
data&lt;-as.data.frame(treedata(tree_pruned, MutAndEcoPGLS, sort=T, warnings=T)$data)
data$Species = as.character(data$Species)
data$A_G.H = as.numeric(data$A_G.H)
data$C_T.H = as.numeric(data$C_T.H)
data$G_A.H = as.numeric(data$G_A.H)
data$T_C.H = as.numeric(data$T_C.H)
data$Temp_avg = as.numeric(data$Temp_avg)
data$Tv &lt;- 1-(data$A_G.H + data$G_A.H + data$T_C.H + data$C_T.H)
data_comp &lt;- comparative.data(tree_pruned, data, Species, vcv=TRUE)

mod &lt;- pgls(A_G.H ~ Temp_avg, data_comp, lambda=&quot;ML&quot;)
summ &lt;- summary( pgls(A_G.H ~ Temp_avg, data_comp, lambda=&quot;ML&quot;))


pgls_results &lt;- create_enhanced_summary_table(summ, mod)

kable(pgls_results$coefficients)</code></pre>
<table>
<colgroup>
<col width="13%" />
<col width="13%" />
<col width="9%" />
<col width="10%" />
<col width="8%" />
<col width="9%" />
<col width="14%" />
<col width="9%" />
<col width="9%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="left">Parameter</th>
<th align="right">Estimate</th>
<th align="right">Std_Error</th>
<th align="right">t_value</th>
<th align="left">p_value</th>
<th align="left">Significance</th>
<th align="right">CI_Lower</th>
<th align="right">CI_Upper</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">(Intercept)</td>
<td align="left">(Intercept)</td>
<td align="right">0.068292</td>
<td align="right">0.024395</td>
<td align="right">2.799</td>
<td align="left">0.0058</td>
<td align="left">**</td>
<td align="right">0.020478</td>
<td align="right">0.116106</td>
</tr>
<tr class="even">
<td align="left">Temp_avg</td>
<td align="left">Temp_avg</td>
<td align="right">0.005357</td>
<td align="right">0.001273</td>
<td align="right">4.207</td>
<td align="left">4.47e-05</td>
<td align="left">***</td>
<td align="right">0.002861</td>
<td align="right">0.007852</td>
</tr>
</tbody>
</table>
<pre class="r"><code>kable(pgls_results$model)</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">Statistic</th>
<th align="right">Value</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">R-squared</td>
<td align="right">0.106800</td>
</tr>
<tr class="even">
<td align="left">Adjusted R-squared</td>
<td align="right">0.100800</td>
</tr>
<tr class="odd">
<td align="left">F-statistic</td>
<td align="right">17.698000</td>
</tr>
<tr class="even">
<td align="left">Lambda</td>
<td align="right">0.000001</td>
</tr>
<tr class="odd">
<td align="left">AIC</td>
<td align="right">-303.430000</td>
</tr>
<tr class="even">
<td align="left">BIC</td>
<td align="right">-297.410000</td>
</tr>
</tbody>
</table>
<pre class="r"><code>#write.table(kable(pgls_results$coefficients), file = here(tables_dir, &quot;Table2a.txt&quot;), row.names = FALSE, col.names = FALSE)
#write.table(kable(pgls_results$model), file = here(tables_dir, &quot;Table2b.txt&quot;), row.names = FALSE, col.names = FALSE)</code></pre>
<pre class="r"><code>tree &lt;- read.tree(file = here(data_dir, &quot;chordates_species.nwk&quot;))
tree &lt;- ape::drop.tip(tree, &quot;Tenebrosternarchus_preto&quot;)

row.names(ClimateZonetest) = ClimateZonetest$Species
tree_pruned = treedata(tree, ClimateZonetest, sort=T, warnings=T)$phy
data&lt;-as.data.frame(treedata(tree_pruned, ClimateZonetest, sort=T, warnings=T)$data)
data$Species = as.character(data$Species)
data$A_G.H = as.numeric(data$A_G.H)
data$ClimateGroup = as.factor(data$ClimateGroup)
data_comp &lt;- comparative.data(tree_pruned, data, Species, vcv=TRUE)



mod &lt;- pgls(A_G.H ~ ClimateGroup, data_comp, lambda=&quot;ML&quot;)
summ &lt;- summary(pgls(A_G.H ~ ClimateGroup, data_comp, lambda=&quot;ML&quot;))


pgls_results &lt;- create_enhanced_summary_table(summ, mod)

kable(pgls_results$coefficients)</code></pre>
<table style="width:100%;">
<colgroup>
<col width="16%" />
<col width="16%" />
<col width="8%" />
<col width="9%" />
<col width="7%" />
<col width="8%" />
<col width="12%" />
<col width="8%" />
<col width="8%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="left">Parameter</th>
<th align="right">Estimate</th>
<th align="right">Std_Error</th>
<th align="right">t_value</th>
<th align="left">p_value</th>
<th align="left">Significance</th>
<th align="right">CI_Lower</th>
<th align="right">CI_Upper</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">(Intercept)</td>
<td align="left">(Intercept)</td>
<td align="right">0.137514</td>
<td align="right">0.007763</td>
<td align="right">17.713</td>
<td align="left">&lt; 2e-16</td>
<td align="left">***</td>
<td align="right">0.122298</td>
<td align="right">0.152731</td>
</tr>
<tr class="even">
<td align="left">ClimateGroupWarm</td>
<td align="left">ClimateGroupWarm</td>
<td align="right">0.034651</td>
<td align="right">0.010353</td>
<td align="right">3.347</td>
<td align="left">0.000917</td>
<td align="left">***</td>
<td align="right">0.014360</td>
<td align="right">0.054942</td>
</tr>
</tbody>
</table>
<pre class="r"><code>kable(pgls_results$model)</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">Statistic</th>
<th align="right">Value</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">R-squared</td>
<td align="right">0.034800</td>
</tr>
<tr class="even">
<td align="left">Adjusted R-squared</td>
<td align="right">0.031700</td>
</tr>
<tr class="odd">
<td align="left">F-statistic</td>
<td align="right">11.203000</td>
</tr>
<tr class="even">
<td align="left">Lambda</td>
<td align="right">0.000001</td>
</tr>
<tr class="odd">
<td align="left">AIC</td>
<td align="right">-611.130000</td>
</tr>
<tr class="even">
<td align="left">BIC</td>
<td align="right">-603.640000</td>
</tr>
</tbody>
</table>
<pre class="r"><code>la.A_G &lt;- phylosig(tree_pruned, data$A_G.H, method = &quot;lambda&quot;, test = TRUE) #Phylogenetic signal lambda : 0.30935 , P-value (based on LR test) : 3.47115e-05
la.C_T &lt;- phylosig(tree_pruned, data$C_T.H, method = &quot;lambda&quot;, test = TRUE) #Phylogenetic signal lambda : 0.216484, P-value (based on LR test) : 0.000556152 
la.G_A &lt;- phylosig(tree_pruned, data$G_A.H, method = &quot;lambda&quot;, test = TRUE) #Phylogenetic signal lambda : 0.522227, P-value (based on LR test) : 5.35018e-05 
la.T_C &lt;- phylosig(tree_pruned, data$T_C.H, method = &quot;lambda&quot;, test = TRUE) #Phylogenetic signal lambda : 0.853065 
la.Tv &lt;- phylosig(tree_pruned, data$Tv, method = &quot;lambda&quot;, test = TRUE)     #Phylogenetic signal lambda : 0.138137 
                                                                            #####################################

signal_strength &lt;- data.frame(
  Level = c(&quot;None&quot;, &quot;Weak&quot;, &quot;Moderate&quot;, &quot;Strong&quot;),
  Range = c(&quot;0-0.15&quot;, &quot;0.15-0.35&quot;, &quot;0.35-0.75&quot;, &quot;0.75-1.0&quot;)
)
signal_strength$Level &lt;- factor(
  signal_strength$Level,
  levels = c(&quot;None&quot;, &quot;Weak&quot;, &quot;Moderate&quot;, &quot;Strong&quot;),
  ordered = TRUE
)

f1a &lt;- ggplot(signal_strength, aes(x=2, y=Level, fill=Level)) +
  geom_col(alpha=0.8, width=1) +
  geom_point(aes(x=1, y=2), size=15, color=&quot;#e74c3c&quot;) + 
  annotate(&quot;text&quot;, x=1, y=2, label=paste(&quot;λ =&quot;, la.A_G$lambda), size=5, fontface=&quot;bold&quot;) +
  scale_fill_manual(values=c(&quot;#ecf0f1&quot;, &quot;#bdc3c7&quot;, &quot;#3498db&quot;, &quot;#2c3e50&quot;)) +
  labs(title=&quot;Phylogenetic Signal Strength of A_G.H&quot;, ) +
  theme_void() +
  theme(legend.position = &quot;none&quot;)

f1b &lt;- ggplot(signal_strength, aes(x=2, y=Level, fill=Level)) +
  geom_col(alpha=0.8, width=1) +
  geom_point(aes(x=1, y=2), size=15, color=&quot;#e74c3c&quot;) + 
  annotate(&quot;text&quot;, x=1, y=2, label=paste(&quot;λ =&quot;, la.C_T$lambda), size=5, fontface=&quot;bold&quot;) +
  scale_fill_manual(values=c(&quot;#ecf0f1&quot;, &quot;#bdc3c7&quot;, &quot;#3498db&quot;, &quot;#2c3e50&quot;)) +
  labs(title=&quot;Phylogenetic Signal Strength of C_T.H&quot;) +
  theme_void()

f1c &lt;- ggplot(signal_strength, aes(x=2, y=Level, fill=Level)) +
  geom_col(alpha=0.8, width=1) +
  geom_point(aes(x=1, y=3), size=15, color=&quot;#e74c3c&quot;) + 
  annotate(&quot;text&quot;, x=1, y=3, label=paste(&quot;λ =&quot;, la.G_A$lambda), size=5, fontface=&quot;bold&quot;) +
  scale_fill_manual(values=c(&quot;#ecf0f1&quot;, &quot;#bdc3c7&quot;, &quot;#3498db&quot;, &quot;#2c3e50&quot;)) +
  labs(title=&quot;Phylogenetic Signal Strength of G_A.H&quot;) +
  theme_void() +
  theme(legend.position = &quot;none&quot;)

f1d &lt;- ggplot(signal_strength, aes(x=2, y=Level, fill=Level)) +
  geom_col(alpha=0.8, width=1) +
  geom_point(aes(x=1, y=4), size=15, color=&quot;#e74c3c&quot;) + 
  annotate(&quot;text&quot;, x=1, y=4, label=paste(&quot;λ =&quot;, la.T_C$lambda), size=5, fontface=&quot;bold&quot;) +
  scale_fill_manual(values=c(&quot;#ecf0f1&quot;, &quot;#bdc3c7&quot;, &quot;#3498db&quot;, &quot;#2c3e50&quot;)) +
  labs(title=&quot;Phylogenetic Signal Strength of T_C.H&quot;) +
  theme_void() +
  theme(legend.position = &quot;none&quot;)

fig &lt;- (f1a | f1b) / (f1c | f1d) 

MutAndEcoPGLS &lt;- MutAndEcoPGLS[!is.na(MutAndEcoPGLS$Temp_avg),]
row.names(MutAndEcoPGLS) = MutAndEcoPGLS$Species
tree_pruned = treedata(tree, MutAndEcoPGLS, sort=T, warnings=T)$phy
data&lt;-as.data.frame(treedata(tree_pruned, MutAndEcoPGLS, sort=T, warnings=T)$data)
data$Species = as.character(data$Species)
data$A_G.H = as.numeric(data$A_G.H)
data$C_T.H = as.numeric(data$C_T.H)
data$G_A.H = as.numeric(data$G_A.H)
data$T_C.H = as.numeric(data$T_C.H)
data$Temp_avg = as.numeric(data$Temp_avg)
data$Tv &lt;- 1-(data$A_G.H + data$G_A.H + data$T_C.H + data$C_T.H)

la.temp &lt;- phylosig(tree_pruned, data$Temp_avg, method = &quot;lambda&quot;, test = TRUE) ########Phylogenetic signal lambda : 0.921383, P-value (based on LR test) : 8.81583e-12 


f1e &lt;- ggplot(signal_strength, aes(x=2, y=Level, fill=Level)) +
  geom_col(alpha=0.8, width=1) +
  geom_point(aes(x=1, y=4), size=15, color=&quot;#e74c3c&quot;) + # Position at &quot;Moderate&quot;
  annotate(&quot;text&quot;, x=1, y=4, label=paste(&quot;λ =&quot;, la.temp$lambda), size=7, fontface=&quot;bold&quot;) +
  scale_fill_manual(values=c(&quot;#ecf0f1&quot;, &quot;#bdc3c7&quot;, &quot;#3498db&quot;, &quot;#2c3e50&quot;)) +
  labs(title=&quot;Phylogenetic Signal Strength of temperature&quot;) +
  theme_void() +
  theme(legend.position = &quot;none&quot;)

f1f &lt;- ggplot(signal_strength, aes(x=2, y=Level, fill=Level)) +
  geom_col(alpha=0.8, width=1) +
  geom_point(aes(x=1, y=1), size=20, color=&quot;#e74c3c&quot;) + # Position at &quot;Moderate&quot;
  annotate(&quot;text&quot;, x=1, y=1, label=paste(&quot;λ =&quot;, la.Tv$lambda), size=7, fontface=&quot;bold&quot;) +
  scale_fill_manual(values=c(&quot;#ecf0f1&quot;, &quot;#bdc3c7&quot;, &quot;#3498db&quot;, &quot;#2c3e50&quot;)) +
  labs(title=&quot;Phylogenetic Signal Strength of transversions&quot;) +
  theme_void() 

figf &lt;- (f1e | f1f)

ggsave(here(plots_dir,&quot;Lambda_temperature_and_Tv.pdf&quot;),
  plot = figf,
  width = 9,
  height = 9
) 

ggsave(here(plots_dir,&quot;Lambda_transitions.pdf&quot;),
  plot = fig,
  width = 8,
  height = 8
) 

create_lambda_summary_table &lt;- function(la_result, varia) {
  values &lt;- c(&quot;variable&quot; = varia,
    `Lambda (λ) estimate` = la_result$lambda,
    `Log-likelihood (λ)` = la_result$logL,
    `Likelihood Ratio (λ=0)` = la_result$logL0,
    `P-value (LR test)` = la_result$P
  )
  df_wide &lt;- as.data.frame(t(values), check.names = FALSE)
  rownames(df_wide) &lt;- NULL
  return(df_wide)
}

AG &lt;- create_lambda_summary_table(la.A_G, &quot;A&gt;G.H&quot;)
TC &lt;- create_lambda_summary_table(la.T_C, &quot;T&gt;C.H&quot;)
GA &lt;- create_lambda_summary_table(la.G_A, &quot;G&gt;A.H&quot;)
CT &lt;- create_lambda_summary_table(la.C_T, &quot;C&gt;T.H&quot;) 

lambda_results &lt;- bind_rows(AG, TC, GA, CT, .id = NULL)

#write.table(kable(lambda_results), file = here(tables_dir, &quot;Table3.txt&quot;), row.names = FALSE, col.names = FALSE)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.5.0 (2025-04-11)
Platform: aarch64-apple-darwin20
Running under: macOS Sequoia 15.5

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.5-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.5-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.1

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Vienna
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] rfishbase_5.0.1    ggstatsplot_0.13.1 broom_1.0.8        ggpubr_0.6.0      
 [5] ggExtra_0.10.1     geiger_2.0.11      phytools_2.4-4     maps_3.4.3        
 [9] caper_1.0.3        mvtnorm_1.3-3      MASS_7.3-65        ape_5.8-1         
[13] magrittr_2.0.3     lubridate_1.9.4    forcats_1.0.0      stringr_1.5.1     
[17] dplyr_1.1.4        purrr_1.0.4        readr_2.1.5        tidyr_1.3.1       
[21] tibble_3.3.0       ggplot2_3.5.2      tidyverse_2.0.0    knitr_1.50        
[25] here_1.0.1         workflowr_1.7.1   

loaded via a namespace (and not attached):
  [1] RColorBrewer_1.1-3      rstudioapi_0.17.1       jsonlite_2.0.0         
  [4] datawizard_1.1.0        correlation_0.8.7       subplex_1.9            
  [7] estimability_1.5.1      farver_2.1.2            rmarkdown_2.29         
 [10] fs_1.6.6                ragg_1.4.0              vctrs_0.6.5            
 [13] paletteer_1.6.0         effectsize_1.0.1        rstatix_0.7.2          
 [16] htmltools_0.5.8.1       DEoptim_2.2-8           deSolve_1.40           
 [19] Formula_1.2-5           sass_0.4.10             bslib_0.9.0            
 [22] emmeans_1.11.1          cachem_1.1.0            whisker_0.4.1          
 [25] igraph_2.1.4            mime_0.13               lifecycle_1.0.4        
 [28] iterators_1.0.14        pkgconfig_2.0.3         Matrix_1.7-3           
 [31] R6_2.6.1                fastmap_1.2.0           shiny_1.10.0           
 [34] BayesFactor_0.9.12-4.7  digest_0.6.37           numDeriv_2016.8-1.1    
 [37] rematch2_2.1.2          patchwork_1.3.0         ps_1.9.1               
 [40] rprojroot_2.0.4         textshaping_1.0.1       labeling_0.4.3         
 [43] clusterGeneration_1.3.8 timechange_0.3.0        mgcv_1.9-3             
 [46] httr_1.4.7              abind_1.4-8             compiler_4.5.0         
 [49] bit64_4.6.0-1           withr_3.0.2             doParallel_1.0.17      
 [52] backports_1.5.0         optimParallel_1.0-2     carData_3.0-5          
 [55] ggsignif_0.6.4          scatterplot3d_0.3-44    tools_4.5.0            
 [58] httpuv_1.6.16           statsExpressions_1.7.0  glue_1.8.0             
 [61] quadprog_1.5-8          callr_3.7.6             nlme_3.1-168           
 [64] promises_1.3.3          grid_4.5.0              getPass_0.2-4          
 [67] generics_0.1.4          gtable_0.3.6            tzdb_0.5.0             
 [70] hms_1.1.3               utf8_1.2.6              car_3.1-3              
 [73] foreach_1.5.2           pillar_1.10.2           vroom_1.6.5            
 [76] later_1.4.2             splines_4.5.0           lattice_0.22-7         
 [79] bit_4.6.0               tidyselect_1.2.1        pbapply_1.7-2          
 [82] miniUI_0.1.2            git2r_0.36.2            xfun_0.52              
 [85] expm_1.0-0              stringi_1.8.7           yaml_2.3.10            
 [88] evaluate_1.0.3          codetools_0.2-20        cli_3.6.5              
 [91] RcppParallel_5.1.10     xtable_1.8-4            parameters_0.26.0      
 [94] systemfonts_1.2.3       processx_3.8.6          jquerylib_0.1.4        
 [97] Rcpp_1.0.14             zeallot_0.2.0           coda_0.19-4.1          
[100] parallel_4.5.0          MatrixModels_0.5-4      rstantools_2.4.0       
[103] bayestestR_0.16.0       phangorn_2.12.1         scales_1.4.0           
[106] insight_1.3.0           crayon_1.5.3            combinat_0.0-8         
[109] rlang_1.1.6             fastmatch_1.1-6         cowplot_1.1.3          
[112] mnormt_2.1.1           </code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
