<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Alina G. Mikhailova" />

<meta name="date" content="2025-06-02" />

<title>Analysis of the polymorphic data in Fishes obtained by NeMu pipeline</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">TemperatureEffectsOnMtDNAspectra</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="I_Polymorphic_Data_Analysis.html">Analysis of the polymorphic data in Fishes obtained by NeMu pipeline</a>
    </li>
  </ul>
</li>
<li>
  <a href="methods.html">Methods</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="license.html">License</a>
</li>
<li>
  <a href="https://github.com/mitoclub/TemperatureEffectsOnMtDNAspectra">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Analysis of the polymorphic data in Fishes
obtained by NeMu pipeline</h1>
<h4 class="author">Alina G. Mikhailova</h4>
<h4 class="date">2025-06-02</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-06-02
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong>
<code>TemperatureEffectsOnMtDNAspectra/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240719code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240719)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240719code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240719)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcommitoclubTemperatureEffectsOnMtDNAspectratreebdaf5f7622b912b7dda1a29557c923639eef7177targetblankbdaf5f7a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/mitoclub/TemperatureEffectsOnMtDNAspectra/tree/bdaf5f7622b912b7dda1a29557c923639eef7177" target="_blank">bdaf5f7</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcommitoclubTemperatureEffectsOnMtDNAspectratreebdaf5f7622b912b7dda1a29557c923639eef7177targetblankbdaf5f7a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/mitoclub/TemperatureEffectsOnMtDNAspectra/tree/bdaf5f7622b912b7dda1a29557c923639eef7177" target="_blank">bdaf5f7</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/.RData
    Ignored:    analysis/.Rhistory
    Ignored:    code/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    output/.DS_Store
    Ignored:    output/tables/.DS_Store

Untracked files:
    Untracked:  analysis/III_Comparison_of_all_Classes.Rmd
    Untracked:  analysis/II_Whole_Genomes_Data_Analysis.Rmd
    Untracked:  analysis/IIa_Whole_Genomes_Data_Analysis_Simulation_Mutagenesis.Rmd
    Untracked:  code/Obtaining_codon_usage_from_MIDORI.R
    Untracked:  data/AllGenesCodonUsageNoOverlap.txt
    Untracked:  data/ColdAndWarmFishesSpectra.txt
    Untracked:  data/Dataset_S1.csv
    Untracked:  data/Dataset_S1.xlsx
    Untracked:  data/ExpectedFractions.txt
    Untracked:  data/Extra_climate_data_2025.csv
    Untracked:  data/Extra_temperature_grow_data_2025.csv
    Untracked:  data/Extra_temperature_popQB_data_2025.csv
    Untracked:  data/Full_fish_ecology_table_2025.csv
    Untracked:  data/MIDORI2_LONGEST_NUC_GB265_Cytb_BLAST.fasta
    Untracked:  data/MIDORI2_LONGEST_NUC_GB265_Cytb_BLAST.fasta.fai
    Untracked:  data/MIDORI_WG_CYTB_FISHES.csv
    Untracked:  data/SimulationData.csv
    Untracked:  output/figures/PolymorphicData.Actinopterygii.AverageMutSpecColdAndWarmCytB.pdf
    Untracked:  output/figures/Simulation_plot.pdf

Unstaged changes:
    Modified:   analysis/I_Polymorphic_Data_Analysis.Rmd
    Modified:   analysis/Ia_Polymorphic_Data_Analysis_TrophLevel.KP.Rmd
    Modified:   output/figures/PolymorphicData.Actinopterygii.A_GwitnTemperatureCytB.pdf
    Modified:   output/figures/PolymorphicData.Actinopterygii.AverageMutSpecColdAndWarm.pdf
    Modified:   output/figures/PolymorphicData.Actinopterygii.AverageMutSpecCytB.pdf
    Deleted:    output/tables/Extra_climate_data_2025.csv
    Deleted:    output/tables/Extra_temperature_grow_data_2025.csv
    Deleted:    output/tables/Extra_temperature_popQB_data_2025.csv
    Deleted:    output/tables/Full_fish_ecology_table_2025.csv

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/I_Polymorphic_Data_Analysis.Rmd</code>) and HTML
(<code>docs/I_Polymorphic_Data_Analysis.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mitoclub/TemperatureEffectsOnMtDNAspectra/blob/bdaf5f7622b912b7dda1a29557c923639eef7177/analysis/I_Polymorphic_Data_Analysis.Rmd" target="_blank">bdaf5f7</a>
</td>
<td>
Alya Mikhailova
</td>
<td>
2025-05-26
</td>
<td>
all analyses on CytB and PGLS for Troph
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mitoclub/TemperatureEffectsOnMtDNAspectra/bdaf5f7622b912b7dda1a29557c923639eef7177/docs/I_Polymorphic_Data_Analysis.html" target="_blank">bdaf5f7</a>
</td>
<td>
Alya Mikhailova
</td>
<td>
2025-05-26
</td>
<td>
all analyses on CytB and PGLS for Troph
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mitoclub/TemperatureEffectsOnMtDNAspectra/1bd7c34389d68ab36e344b3cb575d13fdee9c036/docs/I_Polymorphic_Data_Analysis.html" target="_blank">1bd7c34</a>
</td>
<td>
Alya Mikhailova
</td>
<td>
2025-05-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mitoclub/TemperatureEffectsOnMtDNAspectra/8f5f58cd988819a2c2bbb481846483a95022eaf9/docs/I_Polymorphic_Data_Analysis.html" target="_blank">8f5f58c</a>
</td>
<td>
Alya Mikhailova
</td>
<td>
2025-05-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mitoclub/TemperatureEffectsOnMtDNAspectra/blob/27e35c43fb537215c44b39d9a072adb002f9c517/analysis/I_Polymorphic_Data_Analysis.Rmd" target="_blank">27e35c4</a>
</td>
<td>
Alya Mikhailova
</td>
<td>
2025-05-23
</td>
<td>
Presentation at lab meeting Chapter 1
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mitoclub/TemperatureEffectsOnMtDNAspectra/27e35c43fb537215c44b39d9a072adb002f9c517/docs/I_Polymorphic_Data_Analysis.html" target="_blank">27e35c4</a>
</td>
<td>
Alya Mikhailova
</td>
<td>
2025-05-23
</td>
<td>
Presentation at lab meeting Chapter 1
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div
id="reading-mutation-spectra-database-l-strand-and-ecological-databases"
class="section level1">
<h1>Reading mutation spectra database (L-strand) and ecological
databases</h1>
<pre class="r"><code>#mutSpec12all &lt;- read_csv(here(data_dir, &quot;MutSpecVertebrates12all.csv.gz&quot;))
mutSpec12 &lt;- read_csv(here(data_dir, &quot;MutSpecVertebrates12fish.csv&quot;))
mutSpec12Cb &lt;- dplyr::filter(mutSpec12, Gene == &quot;Cytb&quot;)
cat(&quot;N species =&quot;, length(unique(mutSpec12Cb$Species)))</code></pre>
<pre><code>N species = 840</code></pre>
<pre class="r"><code>head(mutSpec12Cb)</code></pre>
<pre><code># A tibble: 6 × 9
  Gene  Class       Species   Mut   Observed TreeLeaves MSAsize Expected MutSpec
  &lt;chr&gt; &lt;chr&gt;       &lt;chr&gt;     &lt;chr&gt;    &lt;dbl&gt;      &lt;dbl&gt;   &lt;dbl&gt;    &lt;dbl&gt;   &lt;dbl&gt;
1 Cytb  Actinopteri Scombrop… A&gt;C        4           68    1140     71.0  0.0326
2 Cytb  Actinopteri Scombrop… A&gt;G       21.6         68    1140    111.   0.113 
3 Cytb  Actinopteri Scombrop… A&gt;T        2           68    1140     71.0  0.0163
4 Cytb  Actinopteri Scombrop… C&gt;A        3           68    1140     86.3  0.0201
5 Cytb  Actinopteri Scombrop… C&gt;G        0           68    1140     86.3  0     
6 Cytb  Actinopteri Scombrop… C&gt;T       10.7         68    1140    182.   0.0340</code></pre>
<pre class="r"><code>MutSpecForAnalysisCb &lt;- mutSpec12Cb %&gt;% dplyr::select(Class, Species, Mut, MutSpec)
MutSpecForAnalysisCb$Mut &lt;- gsub(&quot;&gt;&quot;, &quot;_&quot;, MutSpecForAnalysisCb$Mut)
MutSpecForAnalysisCb &lt;- MutSpecForAnalysisCb %&gt;% pivot_wider(names_from = Mut, values_from = MutSpec, , values_fn = median) 
colnames(MutSpecForAnalysisCb) &lt;- c(&quot;Class&quot;,
                                  &quot;Species&quot;, 
                                  &quot;T_G.H&quot;, 
                                  &quot;T_C.H&quot;, 
                                  &quot;T_A.H&quot;, 
                                  &quot;G_T.H&quot;, 
                                  &quot;G_C.H&quot;, 
                                  &quot;G_A.H&quot;, 
                                  &quot;C_T.H&quot;, 
                                  &quot;C_G.H&quot;, 
                                  &quot;C_A.H&quot;, 
                                  &quot;A_T.H&quot;, 
                                  &quot;A_G.H&quot;, 
                                  &quot;A_C.H&quot;)

sortedCytB &lt;- MutSpecForAnalysisCb %&gt;% arrange(desc(A_G.H))
#write.table(sortedCytB, file = here(output_dir, &quot;Sortet_by_A_G.H.txt&quot;), col.names = TRUE, row.names = FALSE)


MutSpecForAnalysis &lt;- mutSpec12 %&gt;% dplyr::select(Class, Species, Mut, MutSpec)
MutSpecForAnalysis &lt;- mutSpec12 %&gt;% dplyr::select(Class, Species, Mut, MutSpec)
MutSpecForAnalysis$Mut &lt;- gsub(&quot;&gt;&quot;, &quot;_&quot;, MutSpecForAnalysis$Mut)
MutSpecForAnalysis &lt;- MutSpecForAnalysis %&gt;% pivot_wider(names_from = Mut, values_from = MutSpec, , values_fn = median) 
colnames(MutSpecForAnalysis) &lt;- c(&quot;Class&quot;,
                                  &quot;Species&quot;, 
                                  &quot;T_G.H&quot;, 
                                  &quot;T_C.H&quot;, 
                                  &quot;T_A.H&quot;, 
                                  &quot;G_T.H&quot;, 
                                  &quot;G_C.H&quot;, 
                                  &quot;G_A.H&quot;, 
                                  &quot;C_T.H&quot;, 
                                  &quot;C_G.H&quot;, 
                                  &quot;C_A.H&quot;, 
                                  &quot;A_T.H&quot;, 
                                  &quot;A_G.H&quot;, 
                                  &quot;A_C.H&quot;)</code></pre>
<pre class="r"><code>fishEcology &lt;- read.csv(here(data_dir, &quot;Full_fish_ecology_table_2025.csv&quot;))
head(fishEcology)</code></pre>
<pre><code>                  Species     Wmax      Lmax     tmax
1     Abbottina_rivularis       NA   9.10000  2.00000
2           Abramis_brama 4112.167  46.91111 14.41667
3    Abudefduf_vaigiensis       NA  18.96667       NA
4 Acanthemblemaria_aspera       NA   3.00000  4.00000
5  Acanthocybium_solandri 7200.000 177.88000  9.30000
6     Acanthopagrus_berda  660.000  56.00000 12.50000
                    FeedingType FoodTroph    tm       Lm Temp_min Temp_max
1                          &lt;NA&gt;        NA 1.000  4.50000       NA       NA
2 hunting macrofauna (predator)      3.15 4.125 28.75000       10       24
3     grazing on aquatic plants      3.39    NA 12.00000       NA       NA
4 hunting macrofauna (predator)      3.12    NA       NA       NA       NA
5 hunting macrofauna (predator)      4.45 2.000 99.33333       NA       NA
6 hunting macrofauna (predator)      3.50    NA 21.00000       NA       NA
  Climate_zone Temp_avg
1  subtropical       NA
2    temperate       17
3     tropical       NA
4         &lt;NA&gt;       NA
5  subtropical       NA
6     tropical       NA</code></pre>
<pre class="r"><code>cor.test(fishEcology$tm, fishEcology$Temp_avg)</code></pre>
<pre><code>
    Pearson&#39;s product-moment correlation

data:  fishEcology$tm and fishEcology$Temp_avg
t = -3.7319, df = 99, p-value = 0.0003172
alternative hypothesis: true correlation is not equal to 0
95 percent confidence interval:
 -0.5115159 -0.1672228
sample estimates:
       cor 
-0.3511846 </code></pre>
</div>
<div id="calculating-of-average-mutspec-for-fisches"
class="section level1">
<h1>Calculating of average MutSpec for Fisches</h1>
<div id="cytb" class="section level2">
<h2>Cytb</h2>
<pre class="r"><code>AverageMutSpecCb &lt;- mutSpec12Cb %&gt;% group_by(Mut) %&gt;% summarise(meanmutspec = mean(MutSpec, na.rm = TRUE))
AverageMutSpecCb$Mut.H &lt;- c(&quot;T_G.H&quot;, 
                          &quot;T_C.H&quot;, 
                          &quot;T_A.H&quot;, 
                          &quot;G_T.H&quot;, 
                          &quot;G_C.H&quot;, 
                          &quot;G_A.H&quot;, 
                          &quot;C_T.H&quot;, 
                          &quot;C_G.H&quot;, 
                          &quot;C_A.H&quot;, 
                          &quot;A_T.H&quot;, 
                          &quot;A_G.H&quot;, 
                          &quot;A_C.H&quot;)

f1a = ggplot(AverageMutSpecCb,
  aes(
    x = AverageMutSpecCb$Mut.H,
    y = AverageMutSpecCb$meanmutspec,
    fill = AverageMutSpecCb$Mut.H
    )
  ) +
  geom_bar(stat = &quot;identity&quot;) +
  scale_fill_manual(
    values = c(
      &quot;#bdbdbd&quot;,
      &quot;#00a426&quot;,
      &quot;#bdbdbd&quot;,
      &quot;#bdbdbd&quot;,
      &quot;#bdbdbd&quot;,
      &quot;#ff0000&quot;,
      &quot;#900000&quot;,
      &quot;#bdbdbd&quot;,
      &quot;#bdbdbd&quot;,
      &quot;#bdbdbd&quot;,
      &quot;#005213&quot;,
      &quot;#bdbdbd&quot;
    )
  ) +
  labs(title = &quot;Average mitochondrial mutational spectrum in Fishes&quot;, x = &quot;Substitution types&quot;, y = &quot;Normalised Frequencies&quot;) + theme(
    legend.position = &quot;none&quot;,
    # Remove legend
    panel.background = element_blank(),
    axis.line = element_line(color = &quot;black&quot;)# Remove background
  )

ggsave(here(plots_dir,&quot;PolymorphicData.Actinopterygii.AverageMutSpecCytB.pdf&quot;),
  plot = f1a,
  width = 6,
  height = 6
)
f1a</code></pre>
<p><img src="figure/I_Polymorphic_Data_Analysis.Rmd/plots%20average%20cytb-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-plots-average-cytb-1">
Past versions of "plots average cytb-1.png"
</button>
</p>
<div id="fig-no-spaces-plots-average-cytb-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mitoclub/TemperatureEffectsOnMtDNAspectra/blob/bdaf5f7622b912b7dda1a29557c923639eef7177/docs/figure/I_Polymorphic_Data_Analysis.Rmd/plots average cytb-1.png" target="_blank">bdaf5f7</a>
</td>
<td>
Alya Mikhailova
</td>
<td>
2025-05-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cytbcox1" class="section level2">
<h2>Cytb+COX1</h2>
<pre class="r"><code>AverageMutSpec &lt;- mutSpec12 %&gt;% group_by(Mut) %&gt;% summarise(meanmutspec = mean(MutSpec, na.rm = TRUE))
AverageMutSpec$Mut.H &lt;- c(&quot;T_G.H&quot;, 
                          &quot;T_C.H&quot;, 
                          &quot;T_A.H&quot;, 
                          &quot;G_T.H&quot;, 
                          &quot;G_C.H&quot;, 
                          &quot;G_A.H&quot;, 
                          &quot;C_T.H&quot;, 
                          &quot;C_G.H&quot;, 
                          &quot;C_A.H&quot;, 
                          &quot;A_T.H&quot;, 
                          &quot;A_G.H&quot;, 
                          &quot;A_C.H&quot;)

sf1 = ggplot(AverageMutSpec,
  aes(
    x = AverageMutSpec$Mut.H,
    y = AverageMutSpec$meanmutspec,
    fill = AverageMutSpec$Mut.H
    )
  ) +
  geom_bar(stat = &quot;identity&quot;) +
  scale_fill_manual(
    values = c(
      &quot;#bdbdbd&quot;,
      &quot;#00a426&quot;,
      &quot;#bdbdbd&quot;,
      &quot;#bdbdbd&quot;,
      &quot;#bdbdbd&quot;,
      &quot;#ff0000&quot;,
      &quot;#900000&quot;,
      &quot;#bdbdbd&quot;,
      &quot;#bdbdbd&quot;,
      &quot;#bdbdbd&quot;,
      &quot;#005213&quot;,
      &quot;#bdbdbd&quot;
    )
  ) +
  labs(title = &quot;Average mitochondrial mutational spectrum in Fishes&quot;, x = &quot;Substitution types&quot;, y = &quot;Normalised Frequencies&quot;) + theme(
    legend.position = &quot;none&quot;,
    # Remove legend
    panel.background = element_blank(),
    axis.line = element_line(color = &quot;black&quot;)# Remove background
  )

#ggsave(here(plots_dir,&quot;PolymorphicData.Actinopterygii.AverageMutSpec.pdf&quot;),
#  plot = sf1,
#  width = 6,
#  height = 6
#)
sf1</code></pre>
<p><img src="figure/I_Polymorphic_Data_Analysis.Rmd/plots%20average%20all-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-plots-average-all-1">
Past versions of "plots average all-1.png"
</button>
</p>
<div id="fig-no-spaces-plots-average-all-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mitoclub/TemperatureEffectsOnMtDNAspectra/blob/bdaf5f7622b912b7dda1a29557c923639eef7177/docs/figure/I_Polymorphic_Data_Analysis.Rmd/plots average all-1.png" target="_blank">bdaf5f7</a>
</td>
<td>
Alya Mikhailova
</td>
<td>
2025-05-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="comparison-of-average-mutspec-between-cold-and-warm-fisches"
class="section level1">
<h1>Comparison of average MutSpec between cold and warm fisches</h1>
<p>##Cytb</p>
<pre class="r"><code>merging &lt;- list(MutSpecForAnalysisCb, fishEcology)
MutAndEcoCb &lt;- merging %&gt;% reduce(left_join, by= &quot;Species&quot;)
temperaturestat &lt;- summary (MutAndEcoCb$Temp_avg)

WarmFishes &lt;- filter(MutAndEcoCb, Temp_avg &gt; as.numeric(temperaturestat[&quot;3rd Qu.&quot;]))
WarmFishes &lt;- pivot_longer(WarmFishes, cols = c(&quot;T_G.H&quot;, 
                                  &quot;T_C.H&quot;, 
                                  &quot;T_A.H&quot;, 
                                  &quot;G_T.H&quot;, 
                                  &quot;G_C.H&quot;, 
                                  &quot;G_A.H&quot;, 
                                  &quot;C_T.H&quot;, 
                                  &quot;C_G.H&quot;, 
                                  &quot;C_A.H&quot;, 
                                  &quot;A_T.H&quot;, 
                                  &quot;A_G.H&quot;, 
                                  &quot;A_C.H&quot;), names_to = &quot;Mut&quot;, values_to = &quot;MutSpec&quot; )
AverageMutSpecWarm &lt;- WarmFishes %&gt;% group_by(Mut) %&gt;% summarise(meanmutspec = mean(MutSpec, na.rm = TRUE), error = 
                                                                 sd(MutSpec, na.rm = TRUE))
Nwarm &lt;- length(unique(WarmFishes$Species))

ColdFishes &lt;- filter(MutAndEcoCb, Temp_avg &lt; as.numeric(temperaturestat[&quot;1st Qu.&quot;]))
ColdFishes &lt;- pivot_longer(ColdFishes, cols = c(&quot;T_G.H&quot;, 
                                  &quot;T_C.H&quot;, 
                                  &quot;T_A.H&quot;, 
                                  &quot;G_T.H&quot;, 
                                  &quot;G_C.H&quot;, 
                                  &quot;G_A.H&quot;, 
                                  &quot;C_T.H&quot;, 
                                  &quot;C_G.H&quot;, 
                                  &quot;C_A.H&quot;, 
                                  &quot;A_T.H&quot;, 
                                  &quot;A_G.H&quot;, 
                                  &quot;A_C.H&quot;), names_to = &quot;Mut&quot;, values_to = &quot;MutSpec&quot; )
AverageMutSpecCold &lt;- ColdFishes %&gt;% group_by(Mut) %&gt;% summarise(meanmutspec = mean(MutSpec, na.rm = TRUE), error =
                                                                sd(MutSpec, na.rm = TRUE))
Ncold &lt;- length(unique(ColdFishes$Species))

f1bbw = ggplot(AverageMutSpecWarm,
  aes(
    x = AverageMutSpecWarm$Mut,
    y = AverageMutSpecWarm$meanmutspec,
    fill = AverageMutSpecWarm$Mut
    )
  ) +
  geom_bar(stat = &quot;identity&quot;) +
  scale_fill_manual(values = c(
                    &quot;#bdbdbd&quot;,
                    &quot;#00a426&quot;,
                    &quot;#bdbdbd&quot;,
                    &quot;#bdbdbd&quot;,
                    &quot;#bdbdbd&quot;,
                    &quot;#ff0000&quot;,
                    &quot;#900000&quot;,
                    &quot;#bdbdbd&quot;,
                    &quot;#bdbdbd&quot;,
                    &quot;#bdbdbd&quot;,
                    &quot;#005213&quot;,
                    &quot;#bdbdbd&quot;)) +
  labs(title = &quot;Average mitochondrial mutational spectrum in Warm Fishes, &lt;23C&quot;, x = &quot;Substitution types&quot;, y = &quot;Normalised Frequencies&quot;) + 
  theme(legend.position = &quot;none&quot;,                        # Remove legend
        panel.background = element_blank(),
        axis.line = element_line(color = &quot;black&quot;)) +     # Remove background
  geom_text(x=2, y=0.3, label=paste(&quot;N warm =&quot;, Nwarm), size=7) +
  geom_errorbar(aes(x=AverageMutSpecWarm$Mut, ymin=AverageMutSpecWarm$meanmutspec-AverageMutSpecWarm$error,
                    ymax=AverageMutSpecWarm$meanmutspec+AverageMutSpecWarm$error), width=0.4, colour=&quot;navy&quot;, alpha=0.9,
                    size=1.3)


f1bbc = ggplot(AverageMutSpecCold,
  aes(
    x = AverageMutSpecCold$Mut,
    y = AverageMutSpecCold$meanmutspec,
    fill = AverageMutSpecCold$Mut
    )
  ) +
  geom_bar(stat = &quot;identity&quot;) +
  scale_fill_manual(
    values = c(
      &quot;#bdbdbd&quot;,
      &quot;#00a426&quot;,
      &quot;#bdbdbd&quot;,
      &quot;#bdbdbd&quot;,
      &quot;#bdbdbd&quot;,
      &quot;#ff0000&quot;,
      &quot;#900000&quot;,
      &quot;#bdbdbd&quot;,
      &quot;#bdbdbd&quot;,
      &quot;#bdbdbd&quot;,
      &quot;#005213&quot;,
      &quot;#bdbdbd&quot;
    )
  ) +
  labs(title = &quot;Average mitochondrial mutational spectrum in Cold Fishes, &gt; 13C &quot;, x = &quot;Substitution types&quot;, y = &quot;Normalised Frequencies&quot;) + theme(
    legend.position = &quot;none&quot;,
    # Remove legend
    panel.background = element_blank(),
    axis.line = element_line(color = &quot;black&quot;)# Remove background
  )+ 
  geom_text(x=2, y=0.3, label=paste(&quot;N cold =&quot;, Ncold), size = 7)+
  geom_errorbar(aes(x=AverageMutSpecCold$Mut, ymin=AverageMutSpecCold$meanmutspec-AverageMutSpecCold$error,
                    ymax=AverageMutSpecCold$meanmutspec+AverageMutSpecCold$error), width=0.4, colour=&quot;navy&quot;, alpha=0.9,
                    size=1.3)

both &lt;- ggarrange(f1bbc, f1bbw, ncol = 2)
ggsave(here(plots_dir,&quot;PolymorphicData.Actinopterygii.AverageMutSpecColdAndWarmCytB.pdf&quot;),
  plot = both,
  width = 12,
  height = 6
)
both</code></pre>
<p><img src="figure/I_Polymorphic_Data_Analysis.Rmd/barplots%20cold%20warm%20cytb-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-barplots-cold-warm-cytb-1">
Past versions of "barplots cold warm cytb-1.png"
</button>
</p>
<div id="fig-no-spaces-barplots-cold-warm-cytb-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mitoclub/TemperatureEffectsOnMtDNAspectra/blob/bdaf5f7622b912b7dda1a29557c923639eef7177/docs/figure/I_Polymorphic_Data_Analysis.Rmd/barplots cold warm cytb-1.png" target="_blank">bdaf5f7</a>
</td>
<td>
Alya Mikhailova
</td>
<td>
2025-05-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="cytbcox1-1" class="section level2">
<h2>Cytb+COX1</h2>
<pre class="r"><code>merging &lt;- list(MutSpecForAnalysis, fishEcology)
MutAndEco &lt;- merging %&gt;% reduce(left_join, by= &quot;Species&quot;)
temperaturestat &lt;- summary (MutAndEco$Temp_avg)

WarmFishes &lt;- filter(MutAndEco, Temp_avg &gt; as.numeric(temperaturestat[&quot;3rd Qu.&quot;]))
WarmFishes &lt;- pivot_longer(WarmFishes, cols = c(&quot;T_G.H&quot;, 
                                  &quot;T_C.H&quot;, 
                                  &quot;T_A.H&quot;, 
                                  &quot;G_T.H&quot;, 
                                  &quot;G_C.H&quot;, 
                                  &quot;G_A.H&quot;, 
                                  &quot;C_T.H&quot;, 
                                  &quot;C_G.H&quot;, 
                                  &quot;C_A.H&quot;, 
                                  &quot;A_T.H&quot;, 
                                  &quot;A_G.H&quot;, 
                                  &quot;A_C.H&quot;), names_to = &quot;Mut&quot;, values_to = &quot;MutSpec&quot; )
AverageMutSpecWarm &lt;- WarmFishes %&gt;% group_by(Mut) %&gt;% summarise(meanmutspec = mean(MutSpec, na.rm = TRUE), error =
                                                                sd(MutSpec, na.rm = TRUE))
Nwarm &lt;- length(unique(WarmFishes$Species))

ColdFishes &lt;- filter(MutAndEco, Temp_avg &lt; as.numeric(temperaturestat[&quot;1st Qu.&quot;]))
ColdFishes &lt;- pivot_longer(ColdFishes, cols = c(&quot;T_G.H&quot;, 
                                  &quot;T_C.H&quot;, 
                                  &quot;T_A.H&quot;, 
                                  &quot;G_T.H&quot;, 
                                  &quot;G_C.H&quot;, 
                                  &quot;G_A.H&quot;, 
                                  &quot;C_T.H&quot;, 
                                  &quot;C_G.H&quot;, 
                                  &quot;C_A.H&quot;, 
                                  &quot;A_T.H&quot;, 
                                  &quot;A_G.H&quot;, 
                                  &quot;A_C.H&quot;), names_to = &quot;Mut&quot;, values_to = &quot;MutSpec&quot; )
AverageMutSpecCold &lt;- ColdFishes %&gt;% group_by(Mut) %&gt;% summarise(meanmutspec = mean(MutSpec, na.rm = TRUE), error =
                                                                sd(MutSpec, na.rm = TRUE))
Ncold &lt;- length(unique(ColdFishes$Species))

sf2w = ggplot(AverageMutSpecWarm,
  aes(
    x = AverageMutSpecWarm$Mut,
    y = AverageMutSpecWarm$meanmutspec,
    fill = AverageMutSpecWarm$Mut
    )
  ) +
  geom_bar(stat = &quot;identity&quot;) +
  scale_fill_manual(
    values = c(
      &quot;#bdbdbd&quot;,
      &quot;#00a426&quot;,
      &quot;#bdbdbd&quot;,
      &quot;#bdbdbd&quot;,
      &quot;#bdbdbd&quot;,
      &quot;#ff0000&quot;,
      &quot;#900000&quot;,
      &quot;#bdbdbd&quot;,
      &quot;#bdbdbd&quot;,
      &quot;#bdbdbd&quot;,
      &quot;#005213&quot;,
      &quot;#bdbdbd&quot;
    )
  ) +
  labs(title = &quot;Average mitochondrial mutational spectrum in Warm Fishes, &lt;23C&quot;, x = &quot;Substitution types&quot;, y = &quot;Normalised Frequencies&quot;) + theme(
    legend.position = &quot;none&quot;,
    # Remove legend
    panel.background = element_blank(),
    axis.line = element_line(color = &quot;black&quot;)# Remove background
  ) + 
  geom_text(x=2, y=0.3, label=paste(&quot;N warm =&quot;, Nwarm), size = 7)


sf2c = ggplot(AverageMutSpecCold,
  aes(
    x = AverageMutSpecCold$Mut,
    y = AverageMutSpecCold$meanmutspec,
    fill = AverageMutSpecCold$Mut
    )
  ) +
  geom_bar(stat = &quot;identity&quot;) +
  scale_fill_manual(
    values = c(
      &quot;#bdbdbd&quot;,
      &quot;#00a426&quot;,
      &quot;#bdbdbd&quot;,
      &quot;#bdbdbd&quot;,
      &quot;#bdbdbd&quot;,
      &quot;#ff0000&quot;,
      &quot;#900000&quot;,
      &quot;#bdbdbd&quot;,
      &quot;#bdbdbd&quot;,
      &quot;#bdbdbd&quot;,
      &quot;#005213&quot;,
      &quot;#bdbdbd&quot;
    )
  ) +
  labs(title = &quot;Average mitochondrial mutational spectrum in Cold Fishes, &gt; 13C &quot;, x = &quot;Substitution types&quot;, y = &quot;Normalised Frequencies&quot;) + theme(
    legend.position = &quot;none&quot;,
    # Remove legend
    panel.background = element_blank(),
    axis.line = element_line(color = &quot;black&quot;)# Remove background
  )+ 
  geom_text(x=2, y=0.3, label=paste(&quot;N cold =&quot;, Ncold), size = 7)

both &lt;- ggarrange(sf2c, sf2w, ncol = 2)
#ggsave(here(plots_dir,&quot;PolymorphicData.Actinopterygii.AverageMutSpecColdAndWarm.pdf&quot;),
#  plot = both,
#  width = 12,
#  height = 6
#)
both</code></pre>
<p><img src="figure/I_Polymorphic_Data_Analysis.Rmd/barplots%20cold%20warm-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-barplots-cold-warm-1">
Past versions of "barplots cold warm-1.png"
</button>
</p>
<div id="fig-no-spaces-barplots-cold-warm-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mitoclub/TemperatureEffectsOnMtDNAspectra/blob/bdaf5f7622b912b7dda1a29557c923639eef7177/docs/figure/I_Polymorphic_Data_Analysis.Rmd/barplots cold warm-1.png" target="_blank">bdaf5f7</a>
</td>
<td>
Alya Mikhailova
</td>
<td>
2025-05-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div
id="studies-of-ag-spectra-with-temperature-and-other-parameters-in-fishes"
class="section level1">
<h1>Studies of A&gt;G spectra with temperature and other parameters in
fishes</h1>
<div id="ag-freqs-dependence-on-climate-zones" class="section level2">
<h2>A&gt;G freqs dependence on climate zones</h2>
<p>###CytB</p>
<pre class="r"><code>ClimateZonetest &lt;- na.omit(data.frame(subset(MutAndEcoCb, select = c(A_G.H, Climate_zone))))
ClimateZonetest$Climate_zone &lt;- as.factor(ClimateZonetest$Climate_zone)
AGmedian &lt;- median(ClimateZonetest$A_G.H)
ClimateZonetest$A_G.Freq &lt;- &quot;low&quot;
ClimateZonetest[ClimateZonetest$A_G.H &gt;= AGmedian,]$A_G.Freq &lt;- &quot;high&quot;
ClimateZonetest$A_G.Freq &lt;- as.factor(ClimateZonetest$A_G.Freq)
ClimateZonetest &lt;- ClimateZonetest[,2-3]
sf3 &lt;- ggpiestats(data = ClimateZonetest, x = Climate_zone, y = A_G.Freq)
sf3</code></pre>
<p><img src="figure/I_Polymorphic_Data_Analysis.Rmd/piecharts-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-piecharts-1">
Past versions of piecharts-1.png
</button>
</p>
<div id="fig-piecharts-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mitoclub/TemperatureEffectsOnMtDNAspectra/blob/bdaf5f7622b912b7dda1a29557c923639eef7177/docs/figure/I_Polymorphic_Data_Analysis.Rmd/piecharts-1.png" target="_blank">bdaf5f7</a>
</td>
<td>
Alya Mikhailova
</td>
<td>
2025-05-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ClimateZonetest &lt;- na.omit(data.frame(subset(MutAndEcoCb, select = c(A_G.H, Climate_zone))))
cold &lt;- c(&quot;boreal&quot;, &quot;polar&quot;, &quot;temperate&quot;)
warm &lt;- c(&quot;subtropical&quot;, &quot;tropical&quot;)
wilcox.test(ClimateZonetest[ClimateZonetest$Climate_zone %in% cold,]$A_G.H, 
            ClimateZonetest[ClimateZonetest$Climate_zone %in% warm,]$A_G.H)</code></pre>
<pre><code>
    Wilcoxon rank sum test with continuity correction

data:  ClimateZonetest[ClimateZonetest$Climate_zone %in% cold, ]$A_G.H and ClimateZonetest[ClimateZonetest$Climate_zone %in% warm, ]$A_G.H
W = 9602.5, p-value = 4.671e-06
alternative hypothesis: true location shift is not equal to 0</code></pre>
</div>
<div id="ag-freqs-dependence-on-temperature" class="section level2">
<h2>A&gt;G freqs dependence on temperature</h2>
<pre class="r"><code>tidy(cor.test(MutAndEcoCb$A_G.H, MutAndEcoCb$Temp_avg, method = &#39;spearman&#39;))</code></pre>
<pre><code># A tibble: 1 × 5
  estimate statistic   p.value method                          alternative
     &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt; &lt;chr&gt;                           &lt;chr&gt;      
1    0.330   431970. 0.0000240 Spearman&#39;s rank correlation rho two.sided  </code></pre>
<pre class="r"><code>tidy(cor.test(MutAndEcoCb$A_T.H, MutAndEcoCb$Temp_avg, method = &#39;spearman&#39;))</code></pre>
<pre><code># A tibble: 1 × 5
  estimate statistic p.value method                          alternative
     &lt;dbl&gt;     &lt;dbl&gt;   &lt;dbl&gt; &lt;chr&gt;                           &lt;chr&gt;      
1  -0.0619   684862.   0.441 Spearman&#39;s rank correlation rho two.sided  </code></pre>
<pre class="r"><code>tidy(cor.test(MutAndEcoCb$A_C.H, MutAndEcoCb$Temp_avg, method = &#39;spearman&#39;))</code></pre>
<pre><code># A tibble: 1 × 5
  estimate statistic p.value method                          alternative
     &lt;dbl&gt;     &lt;dbl&gt;   &lt;dbl&gt; &lt;chr&gt;                           &lt;chr&gt;      
1   0.0139   636015.   0.863 Spearman&#39;s rank correlation rho two.sided  </code></pre>
<pre class="r"><code>tidy(cor.test(MutAndEcoCb$T_A.H, MutAndEcoCb$Temp_avg, method = &#39;spearman&#39;))</code></pre>
<pre><code># A tibble: 1 × 5
  estimate statistic p.value method                          alternative
     &lt;dbl&gt;     &lt;dbl&gt;   &lt;dbl&gt; &lt;chr&gt;                           &lt;chr&gt;      
1    0.141   554337.  0.0792 Spearman&#39;s rank correlation rho two.sided  </code></pre>
<pre class="r"><code>tidy(cor.test(MutAndEcoCb$T_G.H, MutAndEcoCb$Temp_avg, method = &#39;spearman&#39;))</code></pre>
<pre><code># A tibble: 1 × 5
  estimate statistic p.value method                          alternative
     &lt;dbl&gt;     &lt;dbl&gt;   &lt;dbl&gt; &lt;chr&gt;                           &lt;chr&gt;      
1   0.0237   629663.   0.768 Spearman&#39;s rank correlation rho two.sided  </code></pre>
<pre class="r"><code>tidy(cor.test(MutAndEcoCb$T_C.H, MutAndEcoCb$Temp_avg, method = &#39;spearman&#39;))</code></pre>
<pre><code># A tibble: 1 × 5
  estimate statistic p.value method                          alternative
     &lt;dbl&gt;     &lt;dbl&gt;   &lt;dbl&gt; &lt;chr&gt;                           &lt;chr&gt;      
1   -0.187   765856.  0.0187 Spearman&#39;s rank correlation rho two.sided  </code></pre>
<pre class="r"><code>tidy(cor.test(MutAndEcoCb$C_A.H, MutAndEcoCb$Temp_avg, method = &#39;spearman&#39;))</code></pre>
<pre><code># A tibble: 1 × 5
  estimate statistic p.value method                          alternative
     &lt;dbl&gt;     &lt;dbl&gt;   &lt;dbl&gt; &lt;chr&gt;                           &lt;chr&gt;      
1   -0.196   771323.  0.0139 Spearman&#39;s rank correlation rho two.sided  </code></pre>
<pre class="r"><code>tidy(cor.test(MutAndEcoCb$C_T.H, MutAndEcoCb$Temp_avg, method = &#39;spearman&#39;))</code></pre>
<pre><code># A tibble: 1 × 5
  estimate statistic p.value method                          alternative
     &lt;dbl&gt;     &lt;dbl&gt;   &lt;dbl&gt; &lt;chr&gt;                           &lt;chr&gt;      
1 -0.00214   646338.   0.979 Spearman&#39;s rank correlation rho two.sided  </code></pre>
<pre class="r"><code>tidy(cor.test(MutAndEcoCb$C_G.H, MutAndEcoCb$Temp_avg, method = &#39;spearman&#39;))</code></pre>
<pre><code># A tibble: 1 × 5
  estimate statistic p.value method                          alternative
     &lt;dbl&gt;     &lt;dbl&gt;   &lt;dbl&gt; &lt;chr&gt;                           &lt;chr&gt;      
1   -0.144   737735.  0.0723 Spearman&#39;s rank correlation rho two.sided  </code></pre>
<pre class="r"><code>tidy(cor.test(MutAndEcoCb$G_T.H, MutAndEcoCb$Temp_avg, method = &#39;spearman&#39;))</code></pre>
<pre><code># A tibble: 1 × 5
  estimate statistic p.value method                          alternative
     &lt;dbl&gt;     &lt;dbl&gt;   &lt;dbl&gt; &lt;chr&gt;                           &lt;chr&gt;      
1  -0.0629   685542.   0.434 Spearman&#39;s rank correlation rho two.sided  </code></pre>
<pre class="r"><code>tidy(cor.test(MutAndEcoCb$G_A.H, MutAndEcoCb$Temp_avg, method = &#39;spearman&#39;))</code></pre>
<pre><code># A tibble: 1 × 5
  estimate statistic p.value method                          alternative
     &lt;dbl&gt;     &lt;dbl&gt;   &lt;dbl&gt; &lt;chr&gt;                           &lt;chr&gt;      
1   0.0938   584432.   0.242 Spearman&#39;s rank correlation rho two.sided  </code></pre>
<pre class="r"><code>tidy(cor.test(MutAndEcoCb$G_C.H, MutAndEcoCb$Temp_avg, method = &#39;spearman&#39;))</code></pre>
<pre><code># A tibble: 1 × 5
  estimate statistic p.value method                          alternative
     &lt;dbl&gt;     &lt;dbl&gt;   &lt;dbl&gt; &lt;chr&gt;                           &lt;chr&gt;      
1  0.00772   639977.   0.924 Spearman&#39;s rank correlation rho two.sided  </code></pre>
<pre class="r"><code>SubsetClass&lt;- unique(MutAndEcoCb$Class)
spN &lt;- nrow(MutAndEcoCb[!is.na(MutAndEcoCb$Temp_avg),])
cat(&quot;N species =&quot;, spN, &quot;Taxa =&quot;, SubsetClass)</code></pre>
<pre><code>N species = 157 Taxa = Actinopteri Chondrichthyes</code></pre>
<pre class="r"><code>f1b &lt;- ggplot(MutAndEcoCb, aes(x=Temp_avg, y=A_G.H)) + 
    geom_point(color=&quot;#73514f&quot;) +                  # Scatter plot
    geom_smooth(method = &quot;lm&quot;, color = &quot;black&quot;,        # Regression line + CI
                fill = &quot;lightgray&quot;, se = TRUE) +     # Set se=TRUE for confidence interval
    stat_cor(method = &quot;spearman&quot;,                    # Spearman correlation stats
            aes(label = paste(..r.label.., ..p.label.., &quot;N&quot;, spN, sep = &quot;~&quot;))) +  # Include n
    labs(x = &quot;Median annual water temperature, ?C&quot;, y = &quot;AH&gt;GH&quot;) +   # Axis labels
    theme_minimal()                                  # Theme customization
 
ggsave(here(plots_dir,&quot;PolymorphicData.Actinopterygii.A_GwitnTemperatureCytB.pdf&quot;),
  plot = f1b,
  width = 6,
  height = 6
) 
f1b</code></pre>
<p><img src="figure/I_Polymorphic_Data_Analysis.Rmd/statistics%20with%20temp%20cytb-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-statistics-with-temp-cytb-1">
Past versions of "statistics with temp cytb-1.png"
</button>
</p>
<div id="fig-no-spaces-statistics-with-temp-cytb-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mitoclub/TemperatureEffectsOnMtDNAspectra/blob/bdaf5f7622b912b7dda1a29557c923639eef7177/docs/figure/I_Polymorphic_Data_Analysis.Rmd/statistics with temp cytb-1.png" target="_blank">bdaf5f7</a>
</td>
<td>
Alya Mikhailova
</td>
<td>
2025-05-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="ag-freqs-dependence-on-time-of-maturation"
class="section level2">
<h2>A&gt;G freqs dependence on time of maturation</h2>
<pre class="r"><code>tidy(cor.test(MutAndEcoCb$A_G.H, MutAndEcoCb$tm, method = &#39;spearman&#39;))</code></pre>
<pre><code># A tibble: 1 × 5
  estimate statistic p.value method                          alternative
     &lt;dbl&gt;     &lt;dbl&gt;   &lt;dbl&gt; &lt;chr&gt;                           &lt;chr&gt;      
1   0.0305   990222.   0.682 Spearman&#39;s rank correlation rho two.sided  </code></pre>
<pre class="r"><code>tidy(cor.test(MutAndEcoCb$A_T.H, MutAndEcoCb$tm, method = &#39;spearman&#39;))</code></pre>
<pre><code># A tibble: 1 × 5
  estimate statistic p.value method                          alternative
     &lt;dbl&gt;     &lt;dbl&gt;   &lt;dbl&gt; &lt;chr&gt;                           &lt;chr&gt;      
1  -0.0439  1066221.   0.555 Spearman&#39;s rank correlation rho two.sided  </code></pre>
<pre class="r"><code>tidy(cor.test(MutAndEcoCb$A_C.H, MutAndEcoCb$tm, method = &#39;spearman&#39;))</code></pre>
<pre><code># A tibble: 1 × 5
  estimate statistic p.value method                          alternative
     &lt;dbl&gt;     &lt;dbl&gt;   &lt;dbl&gt; &lt;chr&gt;                           &lt;chr&gt;      
1  0.00757  1013648.   0.919 Spearman&#39;s rank correlation rho two.sided  </code></pre>
<pre class="r"><code>tidy(cor.test(MutAndEcoCb$T_A.H, MutAndEcoCb$tm, method = &#39;spearman&#39;))</code></pre>
<pre><code># A tibble: 1 × 5
  estimate statistic p.value method                          alternative
     &lt;dbl&gt;     &lt;dbl&gt;   &lt;dbl&gt; &lt;chr&gt;                           &lt;chr&gt;      
1  -0.0450  1067299.   0.546 Spearman&#39;s rank correlation rho two.sided  </code></pre>
<pre class="r"><code>tidy(cor.test(MutAndEcoCb$T_G.H, MutAndEcoCb$tm, method = &#39;spearman&#39;))</code></pre>
<pre><code># A tibble: 1 × 5
  estimate statistic p.value method                          alternative
     &lt;dbl&gt;     &lt;dbl&gt;   &lt;dbl&gt; &lt;chr&gt;                           &lt;chr&gt;      
1  -0.0752  1098178.   0.312 Spearman&#39;s rank correlation rho two.sided  </code></pre>
<pre class="r"><code>tidy(cor.test(MutAndEcoCb$T_C.H, MutAndEcoCb$tm, method = &#39;spearman&#39;))</code></pre>
<pre><code># A tibble: 1 × 5
  estimate statistic p.value method                          alternative
     &lt;dbl&gt;     &lt;dbl&gt;   &lt;dbl&gt; &lt;chr&gt;                           &lt;chr&gt;      
1  -0.0436  1065919.   0.558 Spearman&#39;s rank correlation rho two.sided  </code></pre>
<pre class="r"><code>tidy(cor.test(MutAndEcoCb$C_A.H, MutAndEcoCb$tm, method = &#39;spearman&#39;))</code></pre>
<pre><code># A tibble: 1 × 5
  estimate statistic p.value method                          alternative
     &lt;dbl&gt;     &lt;dbl&gt;   &lt;dbl&gt; &lt;chr&gt;                           &lt;chr&gt;      
1  -0.0539  1076449.   0.469 Spearman&#39;s rank correlation rho two.sided  </code></pre>
<pre class="r"><code>tidy(cor.test(MutAndEcoCb$C_T.H, MutAndEcoCb$tm, method = &#39;spearman&#39;))</code></pre>
<pre><code># A tibble: 1 × 5
  estimate statistic p.value method                          alternative
     &lt;dbl&gt;     &lt;dbl&gt;   &lt;dbl&gt; &lt;chr&gt;                           &lt;chr&gt;      
1  -0.0371  1059268.   0.618 Spearman&#39;s rank correlation rho two.sided  </code></pre>
<pre class="r"><code>tidy(cor.test(MutAndEcoCb$C_G.H, MutAndEcoCb$tm, method = &#39;spearman&#39;))</code></pre>
<pre><code># A tibble: 1 × 5
  estimate statistic p.value method                          alternative
     &lt;dbl&gt;     &lt;dbl&gt;   &lt;dbl&gt; &lt;chr&gt;                           &lt;chr&gt;      
1  -0.0546  1077179.   0.463 Spearman&#39;s rank correlation rho two.sided  </code></pre>
<pre class="r"><code>tidy(cor.test(MutAndEcoCb$G_T.H, MutAndEcoCb$tm, method = &#39;spearman&#39;))</code></pre>
<pre><code># A tibble: 1 × 5
  estimate statistic p.value method                          alternative
     &lt;dbl&gt;     &lt;dbl&gt;   &lt;dbl&gt; &lt;chr&gt;                           &lt;chr&gt;      
1  -0.0399  1062088.   0.592 Spearman&#39;s rank correlation rho two.sided  </code></pre>
<pre class="r"><code>tidy(cor.test(MutAndEcoCb$G_A.H, MutAndEcoCb$tm, method = &#39;spearman&#39;))</code></pre>
<pre><code># A tibble: 1 × 5
  estimate statistic p.value method                          alternative
     &lt;dbl&gt;     &lt;dbl&gt;   &lt;dbl&gt; &lt;chr&gt;                           &lt;chr&gt;      
1   0.0204  1000596.   0.784 Spearman&#39;s rank correlation rho two.sided  </code></pre>
<pre class="r"><code>tidy(cor.test(MutAndEcoCb$G_C.H, MutAndEcoCb$tm, method = &#39;spearman&#39;))</code></pre>
<pre><code># A tibble: 1 × 5
  estimate statistic p.value method                          alternative
     &lt;dbl&gt;     &lt;dbl&gt;   &lt;dbl&gt; &lt;chr&gt;                           &lt;chr&gt;      
1   -0.104  1127853.   0.160 Spearman&#39;s rank correlation rho two.sided  </code></pre>
<pre><code>
      Acanthuriformes        Anabantiformes      Centrarchiformes 
                    4                     6                    10 
        Cypriniformes    Cyprinodontiformes       Eupercaria/misc 
                   44                    10                     4 
        Galaxiiformes           Gobiiformes Perciformes/Percoidei 
                    4                     9                     7 
        Salmoniformes         Scombriformes          Siluriformes 
                   11                    10                     7 </code></pre>
<pre><code>
    Spearman&#39;s rank correlation rho

data:  MediansByOrder$Median_temperature and MediansByOrder$Median_A_G
S = 96, p-value = 0.02216
alternative hypothesis: true rho is not equal to 0
sample estimates:
      rho 
0.6643357 </code></pre>
<p><img src="figure/I_Polymorphic_Data_Analysis.Rmd/statistics%20with%20temp%20orders-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-statistics-with-temp-orders-1">
Past versions of "statistics with temp orders-1.png"
</button>
</p>
<div id="fig-no-spaces-statistics-with-temp-orders-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mitoclub/TemperatureEffectsOnMtDNAspectra/blob/bdaf5f7622b912b7dda1a29557c923639eef7177/docs/figure/I_Polymorphic_Data_Analysis.Rmd/statistics with temp orders-1.png" target="_blank">bdaf5f7</a>
</td>
<td>
Alya Mikhailova
</td>
<td>
2025-05-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="lm" class="section level2">
<h2>LM</h2>
<pre class="r"><code>MutAndEcoCb$DeltaT &lt;- MutAndEcoCb$Temp_max - MutAndEcoCb$Temp_min
summary(lm(A_G.H ~ Temp_avg + DeltaT + Wmax + tmax + tm + FoodTroph, data = MutAndEcoCb))</code></pre>
<pre><code>
Call:
lm(formula = A_G.H ~ Temp_avg + DeltaT + Wmax + tmax + tm + FoodTroph, 
    data = MutAndEcoCb)

Residuals:
      Min        1Q    Median        3Q       Max 
-0.099553 -0.045146 -0.007559  0.043250  0.260853 

Coefficients:
              Estimate Std. Error t value Pr(&gt;|t|)  
(Intercept)  6.088e-02  8.030e-02   0.758   0.4525  
Temp_avg     3.685e-03  2.173e-03   1.696   0.0972 .
DeltaT       2.509e-03  1.279e-03   1.961   0.0563 .
Wmax        -1.043e-07  1.244e-07  -0.838   0.4065  
tmax         8.756e-04  1.510e-03   0.580   0.5650  
tm          -2.989e-03  5.653e-03  -0.529   0.5997  
FoodTroph   -3.790e-03  1.793e-02  -0.211   0.8336  
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 0.06626 on 43 degrees of freedom
  (790 observations deleted due to missingness)
Multiple R-squared:  0.2232,    Adjusted R-squared:  0.1148 
F-statistic: 2.059 on 6 and 43 DF,  p-value: 0.07822</code></pre>
<pre class="r"><code>summary(lm(A_G.H ~ Temp_avg + FoodTroph + Wmax, data = MutAndEcoCb))</code></pre>
<pre><code>
Call:
lm(formula = A_G.H ~ Temp_avg + FoodTroph + Wmax, data = MutAndEcoCb)

Residuals:
     Min       1Q   Median       3Q      Max 
-0.11850 -0.05580 -0.01314  0.03427  0.35432 

Coefficients:
              Estimate Std. Error t value Pr(&gt;|t|)   
(Intercept)  6.346e-03  6.602e-02   0.096  0.92366   
Temp_avg     5.610e-03  1.681e-03   3.337  0.00128 **
FoodTroph    1.920e-02  1.623e-02   1.183  0.24027   
Wmax        -1.094e-07  1.105e-07  -0.990  0.32519   
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 0.0852 on 81 degrees of freedom
  (755 observations deleted due to missingness)
Multiple R-squared:  0.1307,    Adjusted R-squared:  0.09855 
F-statistic: 4.061 on 3 and 81 DF,  p-value: 0.009665</code></pre>
<pre class="r"><code>summary(lm(A_G.H ~ Temp_avg + tmax + tm, data = MutAndEcoCb))</code></pre>
<pre><code>
Call:
lm(formula = A_G.H ~ Temp_avg + tmax + tm, data = MutAndEcoCb)

Residuals:
     Min       1Q   Median       3Q      Max 
-0.12248 -0.04048 -0.01461  0.02858  0.25412 

Coefficients:
             Estimate Std. Error t value Pr(&gt;|t|)  
(Intercept)  0.078515   0.037345   2.102   0.0392 *
Temp_avg     0.004412   0.001877   2.350   0.0216 *
tmax         0.002146   0.001463   1.467   0.1470  
tm          -0.006565   0.005507  -1.192   0.2373  
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 0.07328 on 69 degrees of freedom
  (767 observations deleted due to missingness)
Multiple R-squared:  0.1355,    Adjusted R-squared:  0.09794 
F-statistic: 3.606 on 3 and 69 DF,  p-value: 0.01757</code></pre>
<pre class="r"><code>summary(lm(A_G.H ~ Temp_avg + DeltaT, data = MutAndEcoCb))</code></pre>
<pre><code>
Call:
lm(formula = A_G.H ~ Temp_avg + DeltaT, data = MutAndEcoCb)

Residuals:
     Min       1Q   Median       3Q      Max 
-0.16983 -0.05484 -0.01756  0.02488  0.35404 

Coefficients:
            Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept) 0.041680   0.032431   1.285    0.201    
Temp_avg    0.005853   0.001353   4.325  2.8e-05 ***
DeltaT      0.001108   0.000973   1.139    0.256    
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 0.08856 on 147 degrees of freedom
  (690 observations deleted due to missingness)
Multiple R-squared:  0.1131,    Adjusted R-squared:  0.101 
F-statistic:  9.37 on 2 and 147 DF,  p-value: 0.0001479</code></pre>
</div>
<div id="pgls" class="section level2">
<h2>PGLS</h2>
<pre class="r"><code>tree = read.tree(file = here(data_dir, &quot;chordates_species.nwk&quot;))
tree &lt;- ape::drop.tip(tree, &quot;Tenebrosternarchus_preto&quot;)
MutAndEcoPGLS &lt;- MutAndEco %&gt;% dplyr::select(Species, A_G.H, C_T.H, G_A.H, T_C.H, Temp_avg)

row.names(MutAndEcoPGLS) = MutAndEcoPGLS$Species
tree_pruned = treedata(tree, MutAndEcoPGLS, sort=T, warnings=T)$phy
data&lt;-as.data.frame(treedata(tree_pruned, MutAndEcoPGLS, sort=T, warnings=T)$data)
data$Species = as.character(data$Species)
data$A_G.H = as.numeric(data$A_G.H)
data$C_T.H = as.numeric(data$C_T.H)
data$G_A.H = as.numeric(data$G_A.H)
data$T_C.H = as.numeric(data$T_C.H)
data$Temp_avg = as.numeric(data$Temp_avg)
data$Tv &lt;- 1-(data$A_G.H + data$G_A.H + data$T_C.H + data$C_T.H)
data_comp &lt;- comparative.data(tree_pruned, data, Species, vcv=TRUE)


summary(pgls(A_G.H ~ Temp_avg, data_comp, lambda=&quot;ML&quot;))</code></pre>
<pre><code>
Call:
pgls(formula = A_G.H ~ Temp_avg, data = data_comp, lambda = &quot;ML&quot;)

Residuals:
       Min         1Q     Median         3Q        Max 
-0.0080311 -0.0023654 -0.0005517  0.0015528  0.0167251 

Branch length transformations:

kappa  [Fix]  : 1.000
lambda [ ML]  : 0.000
   lower bound : 0.000, p = 1    
   upper bound : 1.000, p = &lt; 2.22e-16
   95.0% CI   : (NA, 0.175)
delta  [Fix]  : 1.000

Coefficients:
             Estimate Std. Error t value  Pr(&gt;|t|)    
(Intercept) 0.0790861  0.0225661  3.5046 0.0005951 ***
Temp_avg    0.0046805  0.0011718  3.9942 9.924e-05 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 0.00386 on 158 degrees of freedom
Multiple R-squared: 0.09171,    Adjusted R-squared: 0.08596 
F-statistic: 15.95 on 1 and 158 DF,  p-value: 9.924e-05 </code></pre>
<pre class="r"><code>#la.A_G &lt;- phylosig(tree_pruned, data$A_G.H, method = &quot;lambda&quot;, test = TRUE) #Phylogenetic signal lambda : 0.280122 
#la.C_T &lt;- phylosig(tree_pruned, data$C_T.H, method = &quot;lambda&quot;, test = TRUE) #Phylogenetic signal lambda : 0.119451 
#la.G_A &lt;- phylosig(tree_pruned, data$G_A.H, method = &quot;lambda&quot;, test = TRUE) #Phylogenetic signal lambda : 0.719915 
#la.T_C &lt;- phylosig(tree_pruned, data$T_C.H, method = &quot;lambda&quot;, test = TRUE) #Phylogenetic signal lambda : 0.813846 
#la.Tv &lt;- phylosig(tree_pruned, data$Tv, method = &quot;lambda&quot;, test = TRUE) #Phylogenetic signal lambda : 0.113722
#la.temp &lt;- phylosig(tree_pruned, data$Temp_avg, method = &quot;lambda&quot;, test = TRUE) #Phylogenetic signal lambda : 0.92032</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.5.0 (2025-04-11)
Platform: aarch64-apple-darwin20
Running under: macOS Sequoia 15.5

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.5-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.5-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.1

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Vienna
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] rfishbase_5.0.1    ggstatsplot_0.13.1 broom_1.0.8        ggpubr_0.6.0      
 [5] ggExtra_0.10.1     geiger_2.0.11      phytools_2.4-4     maps_3.4.3        
 [9] caper_1.0.3        mvtnorm_1.3-3      MASS_7.3-65        ape_5.8-1         
[13] magrittr_2.0.3     lubridate_1.9.4    forcats_1.0.0      stringr_1.5.1     
[17] dplyr_1.1.4        purrr_1.0.4        readr_2.1.5        tidyr_1.3.1       
[21] tibble_3.2.1       ggplot2_3.5.2      tidyverse_2.0.0    knitr_1.50        
[25] here_1.0.1         workflowr_1.7.1   

loaded via a namespace (and not attached):
  [1] RColorBrewer_1.1-3      rstudioapi_0.17.1       jsonlite_2.0.0         
  [4] datawizard_1.1.0        correlation_0.8.7       subplex_1.9            
  [7] estimability_1.5.1      farver_2.1.2            rmarkdown_2.29         
 [10] fs_1.6.6                ragg_1.4.0              vctrs_0.6.5            
 [13] memoise_2.0.1           paletteer_1.6.0         effectsize_1.0.1       
 [16] rstatix_0.7.2           htmltools_0.5.8.1       curl_6.2.3             
 [19] DEoptim_2.2-8           deSolve_1.40            Formula_1.2-5          
 [22] sass_0.4.10             bslib_0.9.0             emmeans_1.11.1         
 [25] cachem_1.1.0            whisker_0.4.1           igraph_2.1.4           
 [28] mime_0.13               lifecycle_1.0.4         iterators_1.0.14       
 [31] pkgconfig_2.0.3         Matrix_1.7-3            R6_2.6.1               
 [34] fastmap_1.2.0           shiny_1.10.0            BayesFactor_0.9.12-4.7 
 [37] digest_0.6.37           numDeriv_2016.8-1.1     rematch2_2.1.2         
 [40] patchwork_1.3.0         ps_1.9.1                rprojroot_2.0.4        
 [43] prismatic_1.1.2         textshaping_1.0.1       labeling_0.4.3         
 [46] clusterGeneration_1.3.8 timechange_0.3.0        mgcv_1.9-3             
 [49] httr_1.4.7              abind_1.4-8             compiler_4.5.0         
 [52] bit64_4.6.0-1           withr_3.0.2             doParallel_1.0.17      
 [55] duckdbfs_0.1.0          backports_1.5.0         optimParallel_1.0-2    
 [58] DBI_1.2.3               carData_3.0-5           duckdb_1.3.0           
 [61] ggsignif_0.6.4          scatterplot3d_0.3-44    tools_4.5.0            
 [64] httpuv_1.6.16           statsExpressions_1.7.0  glue_1.8.0             
 [67] quadprog_1.5-8          callr_3.7.6             nlme_3.1-168           
 [70] promises_1.3.2          grid_4.5.0              getPass_0.2-4          
 [73] generics_0.1.4          gtable_0.3.6            tzdb_0.5.0             
 [76] hms_1.1.3               car_3.1-3               utf8_1.2.5             
 [79] foreach_1.5.2           pillar_1.10.2           vroom_1.6.5            
 [82] later_1.4.2             splines_4.5.0           lattice_0.22-7         
 [85] bit_4.6.0               tidyselect_1.2.1        pbapply_1.7-2          
 [88] miniUI_0.1.2            git2r_0.36.2            xfun_0.52              
 [91] expm_1.0-0              stringi_1.8.7           yaml_2.3.10            
 [94] evaluate_1.0.3          codetools_0.2-20        cli_3.6.5              
 [97] RcppParallel_5.1.10     xtable_1.8-4            parameters_0.26.0      
[100] systemfonts_1.2.3       processx_3.8.6          jquerylib_0.1.4        
[103] Rcpp_1.0.14             dbplyr_2.5.0            zeallot_0.2.0          
[106] coda_0.19-4.1           parallel_4.5.0          MatrixModels_0.5-4     
[109] rstantools_2.4.0        blob_1.2.4              bayestestR_0.16.0      
[112] phangorn_2.12.1         scales_1.4.0            insight_1.3.0          
[115] crayon_1.5.3            combinat_0.0-8          rlang_1.1.6            
[118] cowplot_1.1.3           fastmatch_1.1-6         mnormt_2.1.1           </code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
