<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Konstantin Popadin" />


<title>MutSpec and TrophLevel</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">TemperatureEffectsOnMtDNAspectra</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="I_Polymorphic_Data_Analysis.html">Analysis of the polymorphic data in Fishes obtained by NeMu pipeline</a>
    </li>
  </ul>
</li>
<li>
  <a href="methods.html">Methods</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="license.html">License</a>
</li>
<li>
  <a href="https://github.com/mitoclub/TemperatureEffectsOnMtDNAspectra">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">MutSpec and TrophLevel</h1>
<h4 class="author">Konstantin Popadin</h4>
<h4 class="date">23/05/2025</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-06-02
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong>
<code>TemperatureEffectsOnMtDNAspectra/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240719code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240719)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240719code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240719)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcommitoclubTemperatureEffectsOnMtDNAspectratree530b2a62fc373b56fd5e6ef0b3970c3a6d4e274ctargetblank530b2a6a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/mitoclub/TemperatureEffectsOnMtDNAspectra/tree/530b2a62fc373b56fd5e6ef0b3970c3a6d4e274c" target="_blank">530b2a6</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcommitoclubTemperatureEffectsOnMtDNAspectratree530b2a62fc373b56fd5e6ef0b3970c3a6d4e274ctargetblank530b2a6a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/mitoclub/TemperatureEffectsOnMtDNAspectra/tree/530b2a62fc373b56fd5e6ef0b3970c3a6d4e274c" target="_blank">530b2a6</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/.RData
    Ignored:    analysis/.Rhistory
    Ignored:    code/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    data/AllGenesCodonUsageNoOverlap.txt
    Ignored:    data/ColdAndWarmFishesSpectra.txt
    Ignored:    data/Dataset_S1.csv
    Ignored:    data/ExpectedFractions.txt
    Ignored:    data/Extra_climate_data_2025.csv
    Ignored:    data/Extra_temperature_grow_data_2025.csv
    Ignored:    data/Extra_temperature_popQB_data_2025.csv
    Ignored:    data/Full_fish_ecology_table_2025.csv
    Ignored:    data/MIDORI2_LONGEST_NUC_GB265_Cytb_BLAST.fasta
    Ignored:    data/MIDORI2_LONGEST_NUC_GB265_Cytb_BLAST.fasta.fai
    Ignored:    data/MIDORI_WG_CYTB_FISHES.csv
    Ignored:    data/SimulationData.csv
    Ignored:    output/.DS_Store
    Ignored:    output/tables/

Unstaged changes:
    Modified:   output/figures/PolymorphicData.Actinopterygii.A_GwitnTemperatureCytB.pdf
    Modified:   output/figures/PolymorphicData.Actinopterygii.AverageMutSpecColdAndWarmCytB.pdf
    Modified:   output/figures/PolymorphicData.Actinopterygii.AverageMutSpecCytB.pdf

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/Ia_Polymorphic_Data_Analysis_TrophLevel.KP.Rmd</code>)
and HTML
(<code>docs/Ia_Polymorphic_Data_Analysis_TrophLevel.KP.html</code>)
files. If you’ve configured a remote Git repository (see
<code>?wflow_git_remote</code>), click on the hyperlinks in the table
below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mitoclub/TemperatureEffectsOnMtDNAspectra/blob/ad28257961a99798122a3b968b1be129307f6f34/analysis/Ia_Polymorphic_Data_Analysis_TrophLevel.KP.Rmd" target="_blank">ad28257</a>
</td>
<td>
Alya Mikhailova
</td>
<td>
2025-06-02
</td>
<td>
Violin plots for tempersture and A&gt;G
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mitoclub/TemperatureEffectsOnMtDNAspectra/ad28257961a99798122a3b968b1be129307f6f34/docs/Ia_Polymorphic_Data_Analysis_TrophLevel.KP.html" target="_blank">ad28257</a>
</td>
<td>
Alya Mikhailova
</td>
<td>
2025-06-02
</td>
<td>
Violin plots for tempersture and A&gt;G
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mitoclub/TemperatureEffectsOnMtDNAspectra/blob/bdaf5f7622b912b7dda1a29557c923639eef7177/analysis/Ia_Polymorphic_Data_Analysis_TrophLevel.KP.Rmd" target="_blank">bdaf5f7</a>
</td>
<td>
Alya Mikhailova
</td>
<td>
2025-05-26
</td>
<td>
all analyses on CytB and PGLS for Troph
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mitoclub/TemperatureEffectsOnMtDNAspectra/bdaf5f7622b912b7dda1a29557c923639eef7177/docs/Ia_Polymorphic_Data_Analysis_TrophLevel.KP.html" target="_blank">bdaf5f7</a>
</td>
<td>
Alya Mikhailova
</td>
<td>
2025-05-26
</td>
<td>
all analyses on CytB and PGLS for Troph
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="background" class="section level2">
<h2>Background</h2>
<p><br> read data <br><br></p>
</div>
<div id="ag-climate.-works" class="section level2">
<h2>1. A&gt;G ~ CLIMATE. works</h2>
<hr />
<pre class="r"><code>### CLIMATE
Mut = read_csv(here(data_dir, &quot;MutSpecVertebrates12fish.csv&quot;))</code></pre>
<pre><code>Rows: 13380 Columns: 9
── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;,&quot;
chr (4): Gene, Class, Species, Mut
dbl (5): Observed, TreeLeaves, MSAsize, Expected, MutSpec

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<pre class="r"><code>names(Mut)</code></pre>
<pre><code>[1] &quot;Gene&quot;       &quot;Class&quot;      &quot;Species&quot;    &quot;Mut&quot;        &quot;Observed&quot;  
[6] &quot;TreeLeaves&quot; &quot;MSAsize&quot;    &quot;Expected&quot;   &quot;MutSpec&quot;   </code></pre>
<pre class="r"><code>head(Mut)</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["Gene"],"name":[1],"type":["chr"],"align":["left"]},{"label":["Class"],"name":[2],"type":["chr"],"align":["left"]},{"label":["Species"],"name":[3],"type":["chr"],"align":["left"]},{"label":["Mut"],"name":[4],"type":["chr"],"align":["left"]},{"label":["Observed"],"name":[5],"type":["dbl"],"align":["right"]},{"label":["TreeLeaves"],"name":[6],"type":["dbl"],"align":["right"]},{"label":["MSAsize"],"name":[7],"type":["dbl"],"align":["right"]},{"label":["Expected"],"name":[8],"type":["dbl"],"align":["right"]},{"label":["MutSpec"],"name":[9],"type":["dbl"],"align":["right"]}],"data":[{"1":"Cytb","2":"Actinopteri","3":"Scombrops_gilberti","4":"A>C","5":"4.00000","6":"68","7":"1140","8":"70.95588","9":"0.03264762"},{"1":"Cytb","2":"Actinopteri","3":"Scombrops_gilberti","4":"A>G","5":"21.60250","6":"68","7":"1140","8":"111.14706","9":"0.11256049"},{"1":"Cytb","2":"Actinopteri","3":"Scombrops_gilberti","4":"A>T","5":"2.00000","6":"68","7":"1140","8":"70.95588","9":"0.01632381"},{"1":"Cytb","2":"Actinopteri","3":"Scombrops_gilberti","4":"C>A","5":"3.00000","6":"68","7":"1140","8":"86.32353","9":"0.02012667"},{"1":"Cytb","2":"Actinopteri","3":"Scombrops_gilberti","4":"C>G","5":"0.00000","6":"68","7":"1140","8":"86.32353","9":"0.00000000"},{"1":"Cytb","2":"Actinopteri","3":"Scombrops_gilberti","4":"C>T","5":"10.66127","6":"68","7":"1140","8":"181.58824","9":"0.03400175"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<pre class="r"><code>dim(Mut)</code></pre>
<pre><code>[1] 13380     9</code></pre>
<pre class="r"><code>Eco = read.csv(here(data_dir, &quot;Full_fish_ecology_table_2025.csv&quot;))
dim(Eco)</code></pre>
<pre><code>[1] 4590   12</code></pre>
<pre class="r"><code>Eco &lt;- Eco[!is.na(Eco$Climate_zone), c(1, 11)]  # Filter and select columns

MutEco = merge(Mut,Eco)

MutEco$WarmClimate &lt;- ifelse(MutEco$Climate_zone %in% c(&quot;tropical&quot;, &quot;subtropical&quot;), 1, 0)
table(MutEco$Gene)</code></pre>
<pre><code>
 CO1 Cytb 
1032 3996 </code></pre>
<pre class="r"><code>MutEco = MutEco[MutEco$Gene == &#39;Cytb&#39;,]
MutEco = MutEco[MutEco$Mut == &#39;T&gt;C&#39;,]
dim(MutEco)</code></pre>
<pre><code>[1] 333  11</code></pre>
<pre class="r"><code>cor.test(MutEco$MutSpec,MutEco$WarmClimate, method = &#39;spearman&#39;)</code></pre>
<pre><code>Warning in cor.test.default(MutEco$MutSpec, MutEco$WarmClimate, method =
&quot;spearman&quot;): Cannot compute exact p-value with ties</code></pre>
<pre><code>
    Spearman&#39;s rank correlation rho

data:  MutEco$MutSpec and MutEco$WarmClimate
S = 4607473, p-value = 3.417e-06
alternative hypothesis: true rho is not equal to 0
sample estimates:
      rho 
0.2513389 </code></pre>
<pre class="r"><code>boxplot(MutEco$MutSpec[MutEco$WarmClimate == 0],MutEco$MutSpec[MutEco$WarmClimate == 1], notch = TRUE)</code></pre>
<p><img src="figure/Ia_Polymorphic_Data_Analysis_TrophLevel.KP.Rmd/unnamed-chunk-1-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-1-1">
Past versions of unnamed-chunk-1-1.png
</button>
</p>
<div id="fig-unnamed-chunk-1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mitoclub/TemperatureEffectsOnMtDNAspectra/blob/bdaf5f7622b912b7dda1a29557c923639eef7177/docs/figure/Ia_Polymorphic_Data_Analysis_TrophLevel.KP.Rmd/unnamed-chunk-1-1.png" target="_blank">bdaf5f7</a>
</td>
<td>
Alya Mikhailova
</td>
<td>
2025-05-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>wilcox.test(MutEco$MutSpec[MutEco$WarmClimate == 0],MutEco$MutSpec[MutEco$WarmClimate == 1])</code></pre>
<pre><code>
    Wilcoxon rank sum test with continuity correction

data:  MutEco$MutSpec[MutEco$WarmClimate == 0] and MutEco$MutSpec[MutEco$WarmClimate == 1]
W = 9602.5, p-value = 4.671e-06
alternative hypothesis: true location shift is not equal to 0</code></pre>
</div>
<div id="pgls1-ag-warmclimate-for-333-fishes" class="section level2">
<h2>PGLS1: A&gt;G ~ WarmClimate for 333 fishes</h2>
<pre class="r"><code>tree = read.tree(file = here(data_dir, &quot;chordates_species.nwk&quot;))
tree &lt;- ape::drop.tip(tree, &quot;Tenebrosternarchus_preto&quot;)
MutEcoPGLS &lt;- MutEco %&gt;% dplyr::select(Species, MutSpec, WarmClimate)

row.names(MutEcoPGLS) = MutEcoPGLS$Species
tree_pruned = treedata(tree, MutEcoPGLS, sort=T, warnings=T)$phy
data&lt;-as.data.frame(treedata(tree_pruned, MutEcoPGLS, sort=T, warnings=T)$data)
data$Species = as.character(data$Species)
data$WarmClimate = as.numeric(data$WarmClimate)
data$MutSpec = as.numeric(data$MutSpec)
data_comp &lt;- comparative.data(tree_pruned, data, Species, vcv=TRUE)


summary(pgls(MutSpec ~ WarmClimate, data_comp, lambda=&quot;ML&quot;))</code></pre>
<pre><code>
Call:
pgls(formula = MutSpec ~ WarmClimate, data = data_comp, lambda = &quot;ML&quot;)

Residuals:
       Min         1Q     Median         3Q        Max 
-0.0080064 -0.0026449 -0.0007283  0.0013327  0.0172939 

Branch length transformations:

kappa  [Fix]  : 1.000
lambda [ ML]  : 0.000
   lower bound : 0.000, p = 1    
   upper bound : 1.000, p = &lt; 2.22e-16
   95.0% CI   : (NA, 0.368)
delta  [Fix]  : 1.000

Coefficients:
             Estimate Std. Error t value  Pr(&gt;|t|)    
(Intercept) 0.1375144  0.0077634 17.7133 &lt; 2.2e-16 ***
WarmClimate 0.0346512  0.0103527  3.3471  0.000917 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 0.004226 on 311 degrees of freedom
Multiple R-squared: 0.03477,    Adjusted R-squared: 0.03167 
F-statistic:  11.2 on 1 and 311 DF,  p-value: 0.000917 </code></pre>
<p><br></p>
<p><br><br></p>
</div>
<div
id="ag-foodtroph-doesnt-work.-may-be-because-climate-is-shaping-more"
class="section level2">
<h2>2. A&gt;G ~ FoodTroph (Doesn’t work. may be because Climate is
shaping more?)</h2>
<pre class="r"><code>Mut = read_csv(here(data_dir, &quot;MutSpecVertebrates12fish.csv&quot;)) </code></pre>
<pre><code>Rows: 13380 Columns: 9
── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;,&quot;
chr (4): Gene, Class, Species, Mut
dbl (5): Observed, TreeLeaves, MSAsize, Expected, MutSpec

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<pre class="r"><code>names(Mut)</code></pre>
<pre><code>[1] &quot;Gene&quot;       &quot;Class&quot;      &quot;Species&quot;    &quot;Mut&quot;        &quot;Observed&quot;  
[6] &quot;TreeLeaves&quot; &quot;MSAsize&quot;    &quot;Expected&quot;   &quot;MutSpec&quot;   </code></pre>
<pre class="r"><code>head(Mut)</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["Gene"],"name":[1],"type":["chr"],"align":["left"]},{"label":["Class"],"name":[2],"type":["chr"],"align":["left"]},{"label":["Species"],"name":[3],"type":["chr"],"align":["left"]},{"label":["Mut"],"name":[4],"type":["chr"],"align":["left"]},{"label":["Observed"],"name":[5],"type":["dbl"],"align":["right"]},{"label":["TreeLeaves"],"name":[6],"type":["dbl"],"align":["right"]},{"label":["MSAsize"],"name":[7],"type":["dbl"],"align":["right"]},{"label":["Expected"],"name":[8],"type":["dbl"],"align":["right"]},{"label":["MutSpec"],"name":[9],"type":["dbl"],"align":["right"]}],"data":[{"1":"Cytb","2":"Actinopteri","3":"Scombrops_gilberti","4":"A>C","5":"4.00000","6":"68","7":"1140","8":"70.95588","9":"0.03264762"},{"1":"Cytb","2":"Actinopteri","3":"Scombrops_gilberti","4":"A>G","5":"21.60250","6":"68","7":"1140","8":"111.14706","9":"0.11256049"},{"1":"Cytb","2":"Actinopteri","3":"Scombrops_gilberti","4":"A>T","5":"2.00000","6":"68","7":"1140","8":"70.95588","9":"0.01632381"},{"1":"Cytb","2":"Actinopteri","3":"Scombrops_gilberti","4":"C>A","5":"3.00000","6":"68","7":"1140","8":"86.32353","9":"0.02012667"},{"1":"Cytb","2":"Actinopteri","3":"Scombrops_gilberti","4":"C>G","5":"0.00000","6":"68","7":"1140","8":"86.32353","9":"0.00000000"},{"1":"Cytb","2":"Actinopteri","3":"Scombrops_gilberti","4":"C>T","5":"10.66127","6":"68","7":"1140","8":"181.58824","9":"0.03400175"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<pre class="r"><code>dim(Mut)</code></pre>
<pre><code>[1] 13380     9</code></pre>
<pre class="r"><code>Eco = read.csv(here(data_dir, &quot;Full_fish_ecology_table_2025.csv&quot;))
dim(Eco)</code></pre>
<pre><code>[1] 4590   12</code></pre>
<pre class="r"><code>summary(Eco$FoodTroph)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max.    NA&#39;s 
  2.000   3.010   3.305   3.313   3.640   4.630    4138 </code></pre>
<pre class="r"><code>str(Eco$FoodTroph)</code></pre>
<pre><code> num [1:4590] NA 3.15 3.39 3.12 4.45 ...</code></pre>
<pre class="r"><code>Eco &lt;- Eco[!is.na(Eco$FoodTroph), c(1, 6)]  # Filter and select columns

MutEco = merge(Mut,Eco)
table(MutEco$Gene)</code></pre>
<pre><code>
 CO1 Cytb 
1416 4572 </code></pre>
<pre class="r"><code>MutEco = MutEco[MutEco$Gene == &#39;Cytb&#39;,]
MutEco = MutEco[MutEco$Mut == &#39;T&gt;C&#39;,]
dim(MutEco)</code></pre>
<pre><code>[1] 381  10</code></pre>
<pre class="r"><code>cor.test(MutEco$MutSpec,MutEco$FoodTroph, method = &#39;spearman&#39;)</code></pre>
<pre><code>Warning in cor.test.default(MutEco$MutSpec, MutEco$FoodTroph, method =
&quot;spearman&quot;): Cannot compute exact p-value with ties</code></pre>
<pre><code>
    Spearman&#39;s rank correlation rho

data:  MutEco$MutSpec and MutEco$FoodTroph
S = 8534993, p-value = 0.1491
alternative hypothesis: true rho is not equal to 0
sample estimates:
      rho 
0.0740608 </code></pre>
<pre class="r"><code>wilcox.test(MutEco$MutSpec[MutEco$FoodTroph &gt;= 3.5],MutEco$MutSpec[MutEco$FoodTroph &lt; 3.5])</code></pre>
<pre><code>
    Wilcoxon rank sum test with continuity correction

data:  MutEco$MutSpec[MutEco$FoodTroph &gt;= 3.5] and MutEco$MutSpec[MutEco$FoodTroph &lt; 3.5]
W = 18604, p-value = 0.1159
alternative hypothesis: true location shift is not equal to 0</code></pre>
<pre class="r"><code># PGLS2: A&gt;G ~ FoodTroph for 381 fishes</code></pre>
</div>
<div id="pgls2-ag-foodtroph-for-381-fishes" class="section level2">
<h2>PGLS2: A&gt;G ~ FoodTroph for 381 fishes</h2>
<pre class="r"><code>tree = read.tree(file = here(data_dir, &quot;chordates_species.nwk&quot;))
tree &lt;- ape::drop.tip(tree, &quot;Tenebrosternarchus_preto&quot;)
MutEcoPGLS &lt;- MutEco %&gt;% dplyr::select(Species, MutSpec, FoodTroph)

row.names(MutEcoPGLS) = MutEcoPGLS$Species
tree_pruned = treedata(tree, MutEcoPGLS, sort=T, warnings=T)$phy
data&lt;-as.data.frame(treedata(tree_pruned, MutEcoPGLS, sort=T, warnings=T)$data)
data$Species = as.character(data$Species)
data$FoodTroph = as.numeric(data$FoodTroph)
data$MutSpec = as.numeric(data$MutSpec)
data_comp &lt;- comparative.data(tree_pruned, data, Species, vcv=TRUE)


summary(pgls(MutSpec ~ FoodTroph, data_comp, lambda=&quot;ML&quot;))</code></pre>
<pre><code>
Call:
pgls(formula = MutSpec ~ FoodTroph, data = data_comp, lambda = &quot;ML&quot;)

Residuals:
       Min         1Q     Median         3Q        Max 
-0.0075404 -0.0029752 -0.0009356  0.0015578  0.0243950 

Branch length transformations:

kappa  [Fix]  : 1.000
lambda [ ML]  : 0.000
   lower bound : 0.000, p = 1    
   upper bound : 1.000, p = &lt; 2.22e-16
   95.0% CI   : (NA, 0.506)
delta  [Fix]  : 1.000

Coefficients:
               Estimate  Std. Error t value  Pr(&gt;|t|)    
(Intercept)  0.16410285  0.02890311  5.6777 2.826e-08 ***
FoodTroph   -0.00097839  0.00862896 -0.1134    0.9098    
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 0.004544 on 358 degrees of freedom
Multiple R-squared: 3.591e-05,  Adjusted R-squared: -0.002757 
F-statistic: 0.01286 on 1 and 358 DF,  p-value: 0.9098 </code></pre>
<p><br></p>
<p><br><br></p>
</div>
<div id="ag-climate-foodtroph" class="section level2">
<h2>3 A&gt;G ~ Climate + FoodTroph</h2>
<pre class="r"><code>Mut = read_csv(here(data_dir, &quot;MutSpecVertebrates12fish.csv&quot;))</code></pre>
<pre><code>Rows: 13380 Columns: 9
── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;,&quot;
chr (4): Gene, Class, Species, Mut
dbl (5): Observed, TreeLeaves, MSAsize, Expected, MutSpec

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<pre class="r"><code>names(Mut)</code></pre>
<pre><code>[1] &quot;Gene&quot;       &quot;Class&quot;      &quot;Species&quot;    &quot;Mut&quot;        &quot;Observed&quot;  
[6] &quot;TreeLeaves&quot; &quot;MSAsize&quot;    &quot;Expected&quot;   &quot;MutSpec&quot;   </code></pre>
<pre class="r"><code>head(Mut)</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["Gene"],"name":[1],"type":["chr"],"align":["left"]},{"label":["Class"],"name":[2],"type":["chr"],"align":["left"]},{"label":["Species"],"name":[3],"type":["chr"],"align":["left"]},{"label":["Mut"],"name":[4],"type":["chr"],"align":["left"]},{"label":["Observed"],"name":[5],"type":["dbl"],"align":["right"]},{"label":["TreeLeaves"],"name":[6],"type":["dbl"],"align":["right"]},{"label":["MSAsize"],"name":[7],"type":["dbl"],"align":["right"]},{"label":["Expected"],"name":[8],"type":["dbl"],"align":["right"]},{"label":["MutSpec"],"name":[9],"type":["dbl"],"align":["right"]}],"data":[{"1":"Cytb","2":"Actinopteri","3":"Scombrops_gilberti","4":"A>C","5":"4.00000","6":"68","7":"1140","8":"70.95588","9":"0.03264762"},{"1":"Cytb","2":"Actinopteri","3":"Scombrops_gilberti","4":"A>G","5":"21.60250","6":"68","7":"1140","8":"111.14706","9":"0.11256049"},{"1":"Cytb","2":"Actinopteri","3":"Scombrops_gilberti","4":"A>T","5":"2.00000","6":"68","7":"1140","8":"70.95588","9":"0.01632381"},{"1":"Cytb","2":"Actinopteri","3":"Scombrops_gilberti","4":"C>A","5":"3.00000","6":"68","7":"1140","8":"86.32353","9":"0.02012667"},{"1":"Cytb","2":"Actinopteri","3":"Scombrops_gilberti","4":"C>G","5":"0.00000","6":"68","7":"1140","8":"86.32353","9":"0.00000000"},{"1":"Cytb","2":"Actinopteri","3":"Scombrops_gilberti","4":"C>T","5":"10.66127","6":"68","7":"1140","8":"181.58824","9":"0.03400175"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<pre class="r"><code>dim(Mut)</code></pre>
<pre><code>[1] 13380     9</code></pre>
<pre class="r"><code>Eco = read.csv(here(data_dir, &quot;Full_fish_ecology_table_2025.csv&quot;))
dim(Eco)</code></pre>
<pre><code>[1] 4590   12</code></pre>
<pre class="r"><code>summary(Eco$FoodTroph)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max.    NA&#39;s 
  2.000   3.010   3.305   3.313   3.640   4.630    4138 </code></pre>
<pre class="r"><code>str(Eco$FoodTroph)</code></pre>
<pre><code> num [1:4590] NA 3.15 3.39 3.12 4.45 ...</code></pre>
<pre class="r"><code>Eco &lt;- Eco[!is.na(Eco$FoodTroph) &amp; !is.na(Eco$Climate_zone), c(1, 6,11)]  # Filter and select columns
table(Eco$Climate_zone)</code></pre>
<pre><code>
     boreal       polar subtropical   temperate    tropical 
          4          10          92         100          87 </code></pre>
<pre class="r"><code>MutEco = merge(Mut,Eco)
MutEco$WarmClimate &lt;- ifelse(MutEco$Climate_zone %in% c(&quot;tropical&quot;, &quot;subtropical&quot;), 1, 0)
MutEco$Predator &lt;- ifelse(MutEco$FoodTroph &gt;= 3.5, 1, 0) # recommendation from chatGPT
MutEco = MutEco[MutEco$Gene == &#39;Cytb&#39;,]
MutEco = MutEco[MutEco$Mut == &#39;T&gt;C&#39;,]
dim(MutEco)</code></pre>
<pre><code>[1] 264  13</code></pre>
<pre class="r"><code>model &lt;- lm(MutSpec ~ FoodTroph + WarmClimate, data = MutEco)
summary(model)</code></pre>
<pre><code>
Call:
lm(formula = MutSpec ~ FoodTroph + WarmClimate, data = MutEco)

Residuals:
     Min       1Q   Median       3Q      Max 
-0.18288 -0.05532 -0.01415  0.02401  0.37897 

Coefficients:
            Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept) 0.069334   0.033577   2.065 0.039917 *  
FoodTroph   0.019153   0.009588   1.997 0.046810 *  
WarmClimate 0.037898   0.011345   3.340 0.000959 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 0.0905 on 261 degrees of freedom
Multiple R-squared:  0.05439,   Adjusted R-squared:  0.04714 
F-statistic: 7.506 on 2 and 261 DF,  p-value: 0.0006769</code></pre>
<pre class="r"><code>model &lt;- lm(MutSpec ~ scale(FoodTroph) + scale(WarmClimate), data = MutEco)
summary(model)</code></pre>
<pre><code>
Call:
lm(formula = MutSpec ~ scale(FoodTroph) + scale(WarmClimate), 
    data = MutEco)

Residuals:
     Min       1Q   Median       3Q      Max 
-0.18288 -0.05532 -0.01415  0.02401  0.37897 

Coefficients:
                   Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)        0.156485   0.005570  28.095  &lt; 2e-16 ***
scale(FoodTroph)   0.011147   0.005581   1.997 0.046810 *  
scale(WarmClimate) 0.018641   0.005581   3.340 0.000959 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 0.0905 on 261 degrees of freedom
Multiple R-squared:  0.05439,   Adjusted R-squared:  0.04714 
F-statistic: 7.506 on 2 and 261 DF,  p-value: 0.0006769</code></pre>
<pre class="r"><code>cor.test(MutEco$MutSpec[MutEco$WarmClimate == 1],MutEco$FoodTroph[MutEco$WarmClimate == 1], method = &#39;spearman&#39;) # only predators in tropcs are super oxidized!</code></pre>
<pre><code>Warning in cor.test.default(MutEco$MutSpec[MutEco$WarmClimate == 1],
MutEco$FoodTroph[MutEco$WarmClimate == : Cannot compute exact p-value with ties</code></pre>
<pre><code>
    Spearman&#39;s rank correlation rho

data:  MutEco$MutSpec[MutEco$WarmClimate == 1] and MutEco$FoodTroph[MutEco$WarmClimate == 1]
S = 510734, p-value = 0.00891
alternative hypothesis: true rho is not equal to 0
sample estimates:
      rho 
0.2081101 </code></pre>
<pre class="r"><code>cor.test(MutEco$MutSpec[MutEco$WarmClimate == 0],MutEco$FoodTroph[MutEco$WarmClimate == 0], method = &#39;spearman&#39;)</code></pre>
<pre><code>Warning in cor.test.default(MutEco$MutSpec[MutEco$WarmClimate == 0],
MutEco$FoodTroph[MutEco$WarmClimate == : Cannot compute exact p-value with ties</code></pre>
<pre><code>
    Spearman&#39;s rank correlation rho

data:  MutEco$MutSpec[MutEco$WarmClimate == 0] and MutEco$FoodTroph[MutEco$WarmClimate == 0]
S = 186045, p-value = 0.3635
alternative hypothesis: true rho is not equal to 0
sample estimates:
       rho 
0.08871308 </code></pre>
<pre class="r"><code>MutEco$ClimateAndTroph = MutEco$WarmClimate +  MutEco$Predator
table(MutEco$ClimateAndTroph)</code></pre>
<pre><code>
  0   1   2 
 65 134  65 </code></pre>
<pre class="r"><code># since the effect of Food and Climate is similar in scale model - I add together! 

# Prepare data groups
group1 &lt;- na.omit(MutEco$MutSpec[MutEco$WarmClimate == 0 &amp; MutEco$Predator == 0])
group2 &lt;- na.omit(MutEco$MutSpec[MutEco$WarmClimate == 0 &amp; MutEco$Predator == 1])
group3 &lt;- na.omit(MutEco$MutSpec[MutEco$WarmClimate == 1 &amp; MutEco$Predator == 0])
group4 &lt;- na.omit(MutEco$MutSpec[MutEco$WarmClimate == 1 &amp; MutEco$Predator == 1])

# Combine into list
groups &lt;- list(group1, group2, group3, group4)
labels &lt;- c(&quot;Cold\nNonPredat&quot;, &quot;Cold\nPredat&quot;, &quot;Warm\nNonPredat&quot;, &quot;Warm\nPredat&quot;)

# Run boxplot with notches and vertical labels
bp &lt;- boxplot(
  groups,
  names = labels,
  notch = TRUE,
  col = c(&quot;lightblue&quot;, &quot;skyblue&quot;, &quot;lightcoral&quot;, &quot;salmon&quot;),
  ylab = &quot;MutSpec&quot;,
  las = 2,
  main = &quot;MutSpec (Ah&gt;Gh) by Climate and Predator Status&quot;
)

# Add sample sizes above each box
sample_sizes &lt;- sapply(groups, length)
text(
  x = 1:4,
  y = bp$stats[5, ] + 0.02,  # slightly above top whisker
  labels = paste0(&quot;n = &quot;, sample_sizes),
  cex = 0.8
)</code></pre>
<p><img src="figure/Ia_Polymorphic_Data_Analysis_TrophLevel.KP.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mitoclub/TemperatureEffectsOnMtDNAspectra/blob/bdaf5f7622b912b7dda1a29557c923639eef7177/docs/figure/Ia_Polymorphic_Data_Analysis_TrophLevel.KP.Rmd/unnamed-chunk-5-1.png" target="_blank">bdaf5f7</a>
</td>
<td>
Alya Mikhailova
</td>
<td>
2025-05-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># PGLS3: A&gt;G ~ WarmClimate + Predator for 264 species
# PGLS4: A&gt;G ~ ClimateAndTroph   # little hope for 264 </code></pre>
</div>
<div
id="pgls3-4-ag-warmclimate-predator-for-264-species-ag-climateandtroph-little-hope-for-264"
class="section level2">
<h2>PGLS3-4: A&gt;G ~ WarmClimate + Predator for 264 species; A&gt;G ~
ClimateAndTroph # little hope for 264</h2>
<pre class="r"><code>tree = read.tree(file = here(data_dir, &quot;chordates_species.nwk&quot;))
tree &lt;- ape::drop.tip(tree, &quot;Tenebrosternarchus_preto&quot;)
MutEcoPGLS &lt;- MutEco %&gt;% dplyr::select(Species, MutSpec, WarmClimate, Predator, ClimateAndTroph)

row.names(MutEcoPGLS) = MutEcoPGLS$Species
tree_pruned = treedata(tree, MutEcoPGLS, sort=T, warnings=T)$phy
data&lt;-as.data.frame(treedata(tree_pruned, MutEcoPGLS, sort=T, warnings=T)$data)
data$Species = as.character(data$Species)
data$WarmClimate = as.numeric(data$WarmClimate)
data$Predator = as.numeric(data$Predator)
data$ClimateAndTroph = as.numeric(data$ClimateAndTroph)
data$MutSpec = as.numeric(data$MutSpec)
data_comp &lt;- comparative.data(tree_pruned, data, Species, vcv=TRUE)

summary(pgls(MutSpec ~ WarmClimate + Predator, data_comp, lambda=&quot;ML&quot;))</code></pre>
<pre><code>
Call:
pgls(formula = MutSpec ~ WarmClimate + Predator, data = data_comp, 
    lambda = &quot;ML&quot;)

Residuals:
      Min        1Q    Median        3Q       Max 
-0.008516 -0.002757 -0.000699  0.001200  0.017604 

Branch length transformations:

kappa  [Fix]  : 1.000
lambda [ ML]  : 0.000
   lower bound : 0.000, p = 1    
   upper bound : 1.000, p = &lt; 2.22e-16
   95.0% CI   : (NA, 0.078)
delta  [Fix]  : 1.000

Coefficients:
            Estimate Std. Error t value  Pr(&gt;|t|)    
(Intercept) 0.128591   0.010016 12.8387 &lt; 2.2e-16 ***
WarmClimate 0.036910   0.011700  3.1545  0.001804 ** 
Predator    0.017622   0.011813  1.4918  0.137019    
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 0.004264 on 250 degrees of freedom
Multiple R-squared: 0.04707,    Adjusted R-squared: 0.03944 
F-statistic: 6.174 on 2 and 250 DF,  p-value: 0.002414 </code></pre>
<pre class="r"><code>summary(pgls(MutSpec ~ ClimateAndTroph, data_comp, lambda=&quot;ML&quot;))</code></pre>
<pre><code>
Call:
pgls(formula = MutSpec ~ ClimateAndTroph, data = data_comp, lambda = &quot;ML&quot;)

Residuals:
       Min         1Q     Median         3Q        Max 
-0.0086071 -0.0028482 -0.0007425  0.0012158  0.0179653 

Branch length transformations:

kappa  [Fix]  : 1.000
lambda [ ML]  : 0.000
   lower bound : 0.000, p = 1    
   upper bound : 1.000, p = &lt; 2.22e-16
   95.0% CI   : (NA, 0.083)
delta  [Fix]  : 1.000

Coefficients:
                 Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)     0.1303702  0.0099020 13.1660  &lt; 2e-16 ***
ClimateAndTroph 0.0273562  0.0082438  3.3184  0.00104 ** 
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 0.004267 on 251 degrees of freedom
Multiple R-squared: 0.04203,    Adjusted R-squared: 0.03821 
F-statistic: 11.01 on 1 and 251 DF,  p-value: 0.00104 </code></pre>
<pre class="r"><code>MutEcoPGLS &lt;- MutEcoPGLS[MutEcoPGLS$WarmClimate == 0,] # if WarmClimate == 1, Predator: est = 0.0088876,  p =  0.5845   

row.names(MutEcoPGLS) = MutEcoPGLS$Species
tree_pruned = treedata(tree, MutEcoPGLS, sort=T, warnings=T)$phy
data&lt;-as.data.frame(treedata(tree_pruned, MutEcoPGLS, sort=T, warnings=T)$data)
data$Species = as.character(data$Species)
data$WarmClimate = as.numeric(data$WarmClimate)
data$Predator = as.numeric(data$Predator)
data$ClimateAndTroph = as.numeric(data$ClimateAndTroph)
data$MutSpec = as.numeric(data$MutSpec)
data_comp &lt;- comparative.data(tree_pruned, data, Species, vcv=TRUE)

summary(pgls(MutSpec ~ Predator, data_comp, lambda=&quot;ML&quot;))</code></pre>
<pre><code>
Call:
pgls(formula = MutSpec ~ Predator, data = data_comp, lambda = &quot;ML&quot;)

Residuals:
      Min        1Q    Median        3Q       Max 
-0.006815 -0.002936 -0.001018  0.001346  0.014921 

Branch length transformations:

kappa  [Fix]  : 1.000
lambda [ ML]  : 0.000
   lower bound : 0.000, p = 1    
   upper bound : 1.000, p = 1.8227e-11
   95.0% CI   : (NA, 0.220)
delta  [Fix]  : 1.000

Coefficients:
            Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept) 0.123824   0.010465 11.8325  &lt; 2e-16 ***
Predator    0.030133   0.016953  1.7774  0.07845 .  
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 0.00451 on 103 degrees of freedom
Multiple R-squared: 0.02976,    Adjusted R-squared: 0.02034 
F-statistic: 3.159 on 1 and 103 DF,  p-value: 0.07845 </code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.5.0 (2025-04-11)
Platform: aarch64-apple-darwin20
Running under: macOS Sequoia 15.5

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.5-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.5-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.1

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Vienna
tzcode source: internal

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] geiger_2.0.11       phytools_2.4-4      maps_3.4.3         
 [4] caper_1.0.3         mvtnorm_1.3-3       ape_5.8-1          
 [7] here_1.0.1          reshape2_1.4.4      MASS_7.3-65        
[10] factoextra_1.0.7    lsa_0.73.3          SnowballC_0.7.1    
[13] Biostrings_2.76.0   GenomeInfoDb_1.44.0 XVector_0.48.0     
[16] IRanges_2.42.0      S4Vectors_0.46.0    BiocGenerics_0.54.0
[19] generics_0.1.4      lubridate_1.9.4     forcats_1.0.0      
[22] stringr_1.5.1       dplyr_1.1.4         purrr_1.0.4        
[25] readr_2.1.5         tidyr_1.3.1         tibble_3.2.1       
[28] ggplot2_3.5.2       tidyverse_2.0.0     knitr_1.50         
[31] workflowr_1.7.1    

loaded via a namespace (and not attached):
 [1] subplex_1.9             tidyselect_1.2.1        farver_2.1.2           
 [4] optimParallel_1.0-2     fastmap_1.2.0           combinat_0.0-8         
 [7] promises_1.3.2          digest_0.6.37           timechange_0.3.0       
[10] lifecycle_1.0.4         processx_3.8.6          magrittr_2.0.3         
[13] compiler_4.5.0          rlang_1.1.6             sass_0.4.10            
[16] tools_4.5.0             igraph_2.1.4            yaml_2.3.10            
[19] phangorn_2.12.1         clusterGeneration_1.3.8 bit_4.6.0              
[22] scatterplot3d_0.3-44    mnormt_2.1.1            plyr_1.8.9             
[25] RColorBrewer_1.1-3      expm_1.0-0              numDeriv_2016.8-1.1    
[28] withr_3.0.2             grid_4.5.0              git2r_0.36.2           
[31] iterators_1.0.14        scales_1.4.0            cli_3.6.5              
[34] rmarkdown_2.29          crayon_1.5.3            rstudioapi_0.17.1      
[37] httr_1.4.7              tzdb_0.5.0              cachem_1.1.0           
[40] parallel_4.5.0          vctrs_0.6.5             Matrix_1.7-3           
[43] jsonlite_2.0.0          callr_3.7.6             hms_1.1.3              
[46] bit64_4.6.0-1           ggrepel_0.9.6           foreach_1.5.2          
[49] jquerylib_0.1.4         glue_1.8.0              codetools_0.2-20       
[52] ps_1.9.1                DEoptim_2.2-8           stringi_1.8.7          
[55] gtable_0.3.6            later_1.4.2             quadprog_1.5-8         
[58] UCSC.utils_1.4.0        pillar_1.10.2           htmltools_0.5.8.1      
[61] deSolve_1.40            GenomeInfoDbData_1.2.14 R6_2.6.1               
[64] doParallel_1.0.17       rprojroot_2.0.4         vroom_1.6.5            
[67] evaluate_1.0.3          lattice_0.22-7          httpuv_1.6.16          
[70] bslib_0.9.0             fastmatch_1.1-6         Rcpp_1.0.14            
[73] coda_0.19-4.1           nlme_3.1-168            whisker_0.4.1          
[76] xfun_0.52               fs_1.6.6                getPass_0.2-4          
[79] pkgconfig_2.0.3        </code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
