<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Alina G. Mikhailova" />

<meta name="date" content="2025-06-02" />

<title>Analysis of the whole genomes data in Fishes from Mikhailova et al. 2022</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">TemperatureEffectsOnMtDNAspectra</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="I_Polymorphic_Data_Analysis.html">Analysis of the polymorphic data in Fishes obtained by NeMu pipeline</a>
    </li>
  </ul>
</li>
<li>
  <a href="methods.html">Methods</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="license.html">License</a>
</li>
<li>
  <a href="https://github.com/mitoclub/TemperatureEffectsOnMtDNAspectra">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Analysis of the whole genomes data in
Fishes from Mikhailova et al. 2022</h1>
<h4 class="author">Alina G. Mikhailova</h4>
<h4 class="date">2025-06-02</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-06-02
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong>
<code>TemperatureEffectsOnMtDNAspectra/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R
Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240719code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240719)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240719code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240719)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcommitoclubTemperatureEffectsOnMtDNAspectratreebdaf5f7622b912b7dda1a29557c923639eef7177targetblankbdaf5f7a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/mitoclub/TemperatureEffectsOnMtDNAspectra/tree/bdaf5f7622b912b7dda1a29557c923639eef7177" target="_blank">bdaf5f7</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcommitoclubTemperatureEffectsOnMtDNAspectratreebdaf5f7622b912b7dda1a29557c923639eef7177targetblankbdaf5f7a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/mitoclub/TemperatureEffectsOnMtDNAspectra/tree/bdaf5f7622b912b7dda1a29557c923639eef7177" target="_blank">bdaf5f7</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/.RData
    Ignored:    analysis/.Rhistory
    Ignored:    code/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    output/.DS_Store
    Ignored:    output/tables/.DS_Store

Untracked files:
    Untracked:  analysis/III_Comparison_of_all_Classes.Rmd
    Untracked:  analysis/II_Whole_Genomes_Data_Analysis.Rmd
    Untracked:  analysis/II_Whole_Genomes_Data_Analysis _Simulation_Mutagenesis.Rmd
    Untracked:  analysis/IIa_Whole_Genomes_Data_Analysis_Simulation_Mutagenesis.Rmd
    Untracked:  code/Obtaining_codon_usage_from_MIDORI.R
    Untracked:  data/AllGenesCodonUsageNoOverlap.txt
    Untracked:  data/ColdAndWarmFishesSpectra.txt
    Untracked:  data/Dataset_S1.csv
    Untracked:  data/Dataset_S1.xlsx
    Untracked:  data/ExpectedFractions.txt
    Untracked:  data/Extra_climate_data_2025.csv
    Untracked:  data/Extra_temperature_grow_data_2025.csv
    Untracked:  data/Extra_temperature_popQB_data_2025.csv
    Untracked:  data/Full_fish_ecology_table_2025.csv
    Untracked:  data/MIDORI2_LONGEST_NUC_GB265_Cytb_BLAST.fasta
    Untracked:  data/MIDORI2_LONGEST_NUC_GB265_Cytb_BLAST.fasta.fai
    Untracked:  data/MIDORI_WG_CYTB_FISHES.csv
    Untracked:  data/SimulationData.csv
    Untracked:  output/figures/PolymorphicData.Actinopterygii.AverageMutSpecColdAndWarmCytB.pdf
    Untracked:  output/figures/Simulation_plot.pdf

Unstaged changes:
    Modified:   analysis/I_Polymorphic_Data_Analysis.Rmd
    Modified:   analysis/Ia_Polymorphic_Data_Analysis_TrophLevel.KP.Rmd
    Modified:   output/figures/PolymorphicData.Actinopterygii.A_GwitnTemperatureCytB.pdf
    Modified:   output/figures/PolymorphicData.Actinopterygii.AverageMutSpecColdAndWarm.pdf
    Modified:   output/figures/PolymorphicData.Actinopterygii.AverageMutSpecCytB.pdf
    Deleted:    output/tables/Extra_climate_data_2025.csv
    Deleted:    output/tables/Extra_temperature_grow_data_2025.csv
    Deleted:    output/tables/Extra_temperature_popQB_data_2025.csv
    Deleted:    output/tables/Full_fish_ecology_table_2025.csv

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div
id="reading-codon-usage-database-l-strand-calculating-neutral-nucleotide-content-and-reading-ecology"
class="section level1">
<h1>Reading codon usage database (L-strand), calculating neutral
nucleotide content and reading ecology</h1>
<pre class="r"><code>mtDNAcodonUsage &lt;- read.table(here(data_dir, &quot;AllGenesCodonUsageNoOverlap.txt&quot;), header = TRUE, sep = &#39;\t&#39;)
mtDNAcodonUsage = mtDNAcodonUsage[mtDNAcodonUsage$Gene != &#39;ND6&#39;,] # check Cyt B

mtDNAcodonUsage = aggregate(list(mtDNAcodonUsage$NeutralA,mtDNAcodonUsage$NeutralT,mtDNAcodonUsage$NeutralG,mtDNAcodonUsage$NeutralC), by = list(mtDNAcodonUsage$Species), FUN = sum)
names(mtDNAcodonUsage) = c(&#39;Species&#39;,&#39;NeutralA&#39;,&#39;NeutralT&#39;,&#39;NeutralG&#39;,&#39;NeutralC&#39;)
mtDNAcodonUsage$FrT.H = mtDNAcodonUsage$NeutralA / (mtDNAcodonUsage$NeutralA + mtDNAcodonUsage$NeutralT +
                                                    mtDNAcodonUsage$NeutralG + mtDNAcodonUsage$NeutralC)
mtDNAcodonUsage$FrA.H = mtDNAcodonUsage$NeutralT / (mtDNAcodonUsage$NeutralA + mtDNAcodonUsage$NeutralT + 
                                                    mtDNAcodonUsage$NeutralG + mtDNAcodonUsage$NeutralC) 
mtDNAcodonUsage$FrC.H = mtDNAcodonUsage$NeutralG / (mtDNAcodonUsage$NeutralA + mtDNAcodonUsage$NeutralT + 
                                                    mtDNAcodonUsage$NeutralG + mtDNAcodonUsage$NeutralC) 
mtDNAcodonUsage$FrG.H = mtDNAcodonUsage$NeutralC / (mtDNAcodonUsage$NeutralA + mtDNAcodonUsage$NeutralT + 
                                                    mtDNAcodonUsage$NeutralG + mtDNAcodonUsage$NeutralC) 

head(mtDNAcodonUsage)</code></pre>
<pre><code>                  Species NeutralA NeutralT NeutralG NeutralC     FrT.H
1     Abalistes_stellaris      799      346      106      723 0.4047619
2 Abbottina_obtusirostris      851      348      192      557 0.4368583
3     Abbottina_rivularis      830      342      206      574 0.4252049
4          Ablennes_hians      768      499       59      656 0.3874874
5           Abramis_brama      867      395      137      549 0.4450719
6        Abronia_graminea      902      334       89      484 0.4986180
      FrA.H      FrC.H     FrG.H
1 0.1752786 0.05369807 0.3662614
2 0.1786448 0.09856263 0.2859343
3 0.1752049 0.10553279 0.2940574
4 0.2517659 0.02976791 0.3309788
5 0.2027721 0.07032854 0.2818275
6 0.1846324 0.04919845 0.2675511</code></pre>
<pre class="r"><code>fishEcology &lt;- read.csv(here(data_dir, &quot;Full_fish_ecology_table_2025.csv&quot;))
head(fishEcology)</code></pre>
<pre><code>                  Species     Wmax      Lmax     tmax
1     Abbottina_rivularis       NA   9.10000  2.00000
2           Abramis_brama 4112.167  46.91111 14.41667
3    Abudefduf_vaigiensis       NA  18.96667       NA
4 Acanthemblemaria_aspera       NA   3.00000  4.00000
5  Acanthocybium_solandri 7200.000 177.88000  9.30000
6     Acanthopagrus_berda  660.000  56.00000 12.50000
                    FeedingType FoodTroph    tm       Lm Temp_min Temp_max
1                          &lt;NA&gt;        NA 1.000  4.50000       NA       NA
2 hunting macrofauna (predator)      3.15 4.125 28.75000       10       24
3     grazing on aquatic plants      3.39    NA 12.00000       NA       NA
4 hunting macrofauna (predator)      3.12    NA       NA       NA       NA
5 hunting macrofauna (predator)      4.45 2.000 99.33333       NA       NA
6 hunting macrofauna (predator)      3.50    NA 21.00000       NA       NA
  Climate_zone Temp_avg
1  subtropical       NA
2    temperate       17
3     tropical       NA
4         &lt;NA&gt;       NA
5  subtropical       NA
6     tropical       NA</code></pre>
</div>
<div id="studies-of-neutral-nucleotide-content-with-temperature"
class="section level1">
<h1>Studies of neutral nucleotide content with temperature</h1>
<p>##Correlating neutral nucleotide content and nucleotide skews with
temperature</p>
<pre class="r"><code>merging &lt;- list(mtDNAcodonUsage, fishEcology)
NucAndEco &lt;- merging %&gt;% reduce(left_join, by= &quot;Species&quot;)
taxonomy &lt;- rfishbase::species_names()
taxonomy$Species &lt;- gsub(&quot; &quot;, &quot;_&quot;, taxonomy$Species)
NucAndEco &lt;- merge(NucAndEco, taxonomy, by=&quot;Species&quot;, all.x = TRUE)
spN &lt;- nrow(NucAndEco[!is.na(NucAndEco$Temp_avg),])
spN</code></pre>
<pre><code>[1] 251</code></pre>
<pre class="r"><code>tidy(cor.test(NucAndEco$FrA.H, NucAndEco$Temp_avg, method = &#39;spearman&#39;))</code></pre>
<pre><code># A tibble: 1 × 5
  estimate statistic p.value method                          alternative
     &lt;dbl&gt;     &lt;dbl&gt;   &lt;dbl&gt; &lt;chr&gt;                           &lt;chr&gt;      
1  -0.0911  2875509.   0.150 Spearman&#39;s rank correlation rho two.sided  </code></pre>
<pre class="r"><code>tidy(cor.test(NucAndEco$FrG.H, NucAndEco$Temp_avg, method = &#39;spearman&#39;))</code></pre>
<pre><code># A tibble: 1 × 5
  estimate statistic p.value method                          alternative
     &lt;dbl&gt;     &lt;dbl&gt;   &lt;dbl&gt; &lt;chr&gt;                           &lt;chr&gt;      
1    0.188  2138741. 0.00272 Spearman&#39;s rank correlation rho two.sided  </code></pre>
<pre class="r"><code>tidy(cor.test(NucAndEco$FrT.H, NucAndEco$Temp_avg, method = &#39;spearman&#39;))</code></pre>
<pre><code># A tibble: 1 × 5
  estimate statistic p.value method                          alternative
     &lt;dbl&gt;     &lt;dbl&gt;   &lt;dbl&gt; &lt;chr&gt;                           &lt;chr&gt;      
1 -0.00681  2653443.   0.915 Spearman&#39;s rank correlation rho two.sided  </code></pre>
<pre class="r"><code>tidy(cor.test(NucAndEco$FrC.H, NucAndEco$Temp_avg, method = &#39;spearman&#39;)) #tree?</code></pre>
<pre><code># A tibble: 1 × 5
  estimate statistic     p.value method                          alternative
     &lt;dbl&gt;     &lt;dbl&gt;       &lt;dbl&gt; &lt;chr&gt;                           &lt;chr&gt;      
1   -0.326  3494698. 0.000000126 Spearman&#39;s rank correlation rho two.sided  </code></pre>
<pre class="r"><code>NucAndEco$AGskew &lt;- (NucAndEco$FrA.H - NucAndEco$FrG.H)/(NucAndEco$FrG.H + NucAndEco$FrA.H)
NucAndEco$CTskew &lt;- (NucAndEco$FrC.H - NucAndEco$FrT.H)/(NucAndEco$FrT.H + NucAndEco$FrC.H)

tidy(cor.test(NucAndEco$AGskew, NucAndEco$Temp_avg, method = &#39;spearman&#39;))</code></pre>
<pre><code># A tibble: 1 × 5
  estimate statistic p.value method                          alternative
     &lt;dbl&gt;     &lt;dbl&gt;   &lt;dbl&gt; &lt;chr&gt;                           &lt;chr&gt;      
1   -0.197  3153914. 0.00174 Spearman&#39;s rank correlation rho two.sided  </code></pre>
<pre class="r"><code>tidy(cor.test(NucAndEco$CTskew, NucAndEco$Temp_avg, method = &#39;spearman&#39;))</code></pre>
<pre><code># A tibble: 1 × 5
  estimate statistic   p.value method                          alternative
     &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt; &lt;chr&gt;                           &lt;chr&gt;      
1   -0.257  3312111. 0.0000385 Spearman&#39;s rank correlation rho two.sided  </code></pre>
<p>##Correlating neutral nucleotide content and nucleotide skews with
time of maturation</p>
<pre class="r"><code>nrow(NucAndEco[!is.na(NucAndEco$tm),])</code></pre>
<pre><code>[1] 99</code></pre>
<pre class="r"><code>tidy(cor.test(NucAndEco$FrA.H, NucAndEco$tm, method = &#39;spearman&#39;))</code></pre>
<pre><code># A tibble: 1 × 5
  estimate statistic p.value method                          alternative
     &lt;dbl&gt;     &lt;dbl&gt;   &lt;dbl&gt; &lt;chr&gt;                           &lt;chr&gt;      
1 -0.00942   163223.   0.926 Spearman&#39;s rank correlation rho two.sided  </code></pre>
<pre class="r"><code>tidy(cor.test(NucAndEco$FrG.H, NucAndEco$tm, method = &#39;spearman&#39;))</code></pre>
<pre><code># A tibble: 1 × 5
  estimate statistic p.value method                          alternative
     &lt;dbl&gt;     &lt;dbl&gt;   &lt;dbl&gt; &lt;chr&gt;                           &lt;chr&gt;      
1  -0.0200   164929.   0.844 Spearman&#39;s rank correlation rho two.sided  </code></pre>
<pre class="r"><code>tidy(cor.test(NucAndEco$FrT.H, NucAndEco$tm, method = &#39;spearman&#39;))</code></pre>
<pre><code># A tibble: 1 × 5
  estimate statistic p.value method                          alternative
     &lt;dbl&gt;     &lt;dbl&gt;   &lt;dbl&gt; &lt;chr&gt;                           &lt;chr&gt;      
1   0.0946   146403.   0.352 Spearman&#39;s rank correlation rho two.sided  </code></pre>
<pre class="r"><code>tidy(cor.test(NucAndEco$FrC.H, NucAndEco$tm, method = &#39;spearman&#39;))</code></pre>
<pre><code># A tibble: 1 × 5
  estimate statistic p.value method                          alternative
     &lt;dbl&gt;     &lt;dbl&gt;   &lt;dbl&gt; &lt;chr&gt;                           &lt;chr&gt;      
1   -0.142   184716.   0.160 Spearman&#39;s rank correlation rho two.sided  </code></pre>
<div id="figures" class="section level2">
<h2>Figures</h2>
<pre class="r"><code>sf4 &lt;- ggplot(NucAndEco, aes(x=Temp_avg, y=AGskew)) + 
    geom_point(color=&quot;#c99bc9&quot;) +                  # Scatter plot
    geom_smooth(method = &quot;lm&quot;, color = &quot;black&quot;,        # Regression line + CI
                fill = &quot;lightgray&quot;, se = TRUE) +     # Set se=TRUE for confidence interval
    stat_cor(method = &quot;spearman&quot;,                    # Spearman correlation stats
            aes(label = paste(..r.label.., ..p.label.., &quot;N&quot;, spN, sep = &quot;~&quot;))) +  # Include n
    labs(x = &quot;Median annual water temperature, ?C&quot;, y = &quot;AGskew&quot;) +   # Axis labels
    theme_minimal()                                  # Theme customization
 
#ggsave(here(plots_dir,&quot;.pdf&quot;),
#  plot = f1b,
#  width = 6,
#  height = 6
#) 
sf4</code></pre>
<p><img src="figure/II_Whole_Genomes_Data_Analysis%C2%A0_Simulation_Mutagenesis.Rmd/unnamed-chunk-1-1.png" width="1344" style="display: block; margin: auto;" /></p>
<pre class="r"><code>MaturationMedian &lt;- median(NucAndEco$tm, na.rm = TRUE)
TemperatureMedian &lt;- median(NucAndEco$Temp_avg, na.rm = TRUE)
NucAndEco$TemperatureGroup &lt;- ifelse(NucAndEco$Temp_avg &gt;= TemperatureMedian, &quot;Warm&quot;, &quot;Cold&quot;)
NucAndEco$LongevityGroup &lt;- ifelse(NucAndEco$tm &gt;= MaturationMedian, &quot;LongLived&quot;, &quot;ShortLived&quot;)
NucAndEco &lt;- NucAndEco[!is.na(NucAndEco$TemperatureGroup) &amp; !is.na(NucAndEco$LongevityGroup),]

sf6 &lt;- ggplot(data = NucAndEco, aes(x = TemperatureGroup, y = AGskew))+
  geom_boxplot()+
  facet_wrap(~LongevityGroup)+
  theme_test()+
  labs(y = &#39;AGskew&#39;)
sf6</code></pre>
<p><img src="figure/II_Whole_Genomes_Data_Analysis%C2%A0_Simulation_Mutagenesis.Rmd/unnamed-chunk-2-1.png" width="1344" style="display: block; margin: auto;" />
##LM</p>
<pre class="r"><code>summary(lm(AGskew ~ Temp_avg + tm, data = NucAndEco))</code></pre>
<pre><code>
Call:
lm(formula = AGskew ~ Temp_avg + tm, data = NucAndEco)

Residuals:
     Min       1Q   Median       3Q      Max 
-0.30407 -0.07081 -0.01040  0.06959  0.38941 

Coefficients:
             Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)  0.067218   0.063662   1.056   0.2948    
Temp_avg    -0.014028   0.002961  -4.738 1.17e-05 ***
tm          -0.015330   0.006500  -2.358   0.0213 *  
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 0.1218 on 67 degrees of freedom
Multiple R-squared:  0.256, Adjusted R-squared:  0.2338 
F-statistic: 11.53 on 2 and 67 DF,  p-value: 4.98e-05</code></pre>
<pre class="r"><code>summary(lm(CTskew ~ Temp_avg + tm, data = NucAndEco))</code></pre>
<pre><code>
Call:
lm(formula = CTskew ~ Temp_avg + tm, data = NucAndEco)

Residuals:
      Min        1Q    Median        3Q       Max 
-0.257588 -0.067603 -0.002239  0.068047  0.275417 

Coefficients:
             Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept) -0.535470   0.054464  -9.832 1.27e-14 ***
Temp_avg    -0.005579   0.002533  -2.203   0.0311 *  
tm          -0.009763   0.005561  -1.756   0.0837 .  
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 0.1042 on 67 degrees of freedom
Multiple R-squared:  0.082, Adjusted R-squared:  0.0546 
F-statistic: 2.992 on 2 and 67 DF,  p-value: 0.05691</code></pre>
<pre class="r"><code>summary(lm(FrA.H ~ Temp_avg + tm, data = NucAndEco))</code></pre>
<pre><code>
Call:
lm(formula = FrA.H ~ Temp_avg + tm, data = NucAndEco)

Residuals:
      Min        1Q    Median        3Q       Max 
-0.068232 -0.030820 -0.002251  0.027608  0.093488 

Coefficients:
              Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)  0.2993358  0.0207182  14.448  &lt; 2e-16 ***
Temp_avg    -0.0041873  0.0009636  -4.346 4.83e-05 ***
tm          -0.0057121  0.0021153  -2.700  0.00876 ** 
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 0.03964 on 67 degrees of freedom
Multiple R-squared:  0.2347,    Adjusted R-squared:  0.2118 
F-statistic: 10.27 on 2 and 67 DF,  p-value: 0.0001286</code></pre>
<pre class="r"><code>summary(lm(FrG.H ~ Temp_avg + tm, data = NucAndEco))</code></pre>
<pre><code>
Call:
lm(formula = FrG.H ~ Temp_avg + tm, data = NucAndEco)

Residuals:
      Min        1Q    Median        3Q       Max 
-0.163169 -0.044326  0.000264  0.045528  0.120750 

Coefficients:
             Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept) 0.2690186  0.0295255   9.111 2.42e-13 ***
Temp_avg    0.0034362  0.0013732   2.502   0.0148 *  
tm          0.0007556  0.0030145   0.251   0.8028    
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 0.0565 on 67 degrees of freedom
Multiple R-squared:  0.09066,   Adjusted R-squared:  0.06352 
F-statistic:  3.34 on 2 and 67 DF,  p-value: 0.04143</code></pre>
</div>
<div id="pgls" class="section level2">
<h2>PGLS</h2>
<pre class="r"><code>tree = read.tree(file = here(data_dir, &quot;chordates_species.nwk&quot;))
tree &lt;- ape::drop.tip(tree, &quot;Tenebrosternarchus_preto&quot;)
NucAndEcoPGLS &lt;- NucAndEco %&gt;% dplyr::select(Species, FrA.H, FrG.H, FrT.H, FrC.H, AGskew, CTskew, Temp_avg)
NucAndEcoPGLS &lt;- NucAndEcoPGLS[!is.na(NucAndEcoPGLS$Temp_avg),]

row.names(NucAndEcoPGLS) = NucAndEcoPGLS$Species
tree_pruned = treedata(tree, NucAndEcoPGLS, sort=T, warnings=T)$phy
data&lt;-as.data.frame(treedata(tree_pruned, NucAndEcoPGLS, sort=T, warnings=T)$data)
data$Species = as.character(data$Species)
data$FrA.H = as.numeric(data$FrA.H)
data$FrG.H = as.numeric(data$FrG.H)
data$FrT.H = as.numeric(data$FrT.H)
data$FrC.H = as.numeric(data$FrC.H)
data$AGskew = as.numeric(data$AGskew)
data$CTskew = as.numeric(data$CTskew)
data$Temp_avg = as.numeric(data$Temp_avg)
data_comp &lt;- comparative.data(tree_pruned, data, Species, vcv=TRUE)

if (exists(&quot;tree_pruned&quot;) &amp;&amp; inherits(tree_pruned, &quot;phylo&quot;)) {
  # Check for zero-length branches
  num_zero_branches &lt;- sum(tree_pruned$edge.length == 0)
  cat(&quot;Number of zero-length branches in tree_pruned:&quot;, num_zero_branches, &quot;\n&quot;)
  if (num_zero_branches &gt; 0) {
    cat(&quot;Some branch lengths are exactly zero.\n&quot;)
  }

  # Check for very small branch lengths (can also cause numerical instability)
  min_branch_length &lt;- min(tree_pruned$edge.length[tree_pruned$edge.length &gt; 0]) # Smallest non-zero
  cat(&quot;Smallest non-zero branch length:&quot;, min_branch_length, &quot;\n&quot;)
  cat(&quot;Number of branches &lt; 1e-8 (a very small number):&quot;, sum(tree_pruned$edge.length &lt; 1e-8), &quot;\n&quot;)

  # Check if the tree is binary (no polytomies)
  is_binary_tree &lt;- is.binary(tree_pruned)
  cat(&quot;Is tree_pruned binary (no polytomies)?&quot;, is_binary_tree, &quot;\n&quot;)
  if (!is_binary_tree) {
    cat(&quot;Tree_pruned contains polytomies.\n&quot;)
  }
  
  # Basic tree summary
  # summary(tree_pruned) # Provides a general overview
  
} else {
  cat(&quot;The &#39;tree_pruned&#39; object doesn&#39;t exist or is not a valid phylo object.\n&quot;)
}</code></pre>
<pre><code>Number of zero-length branches in tree_pruned: 2 
Some branch lengths are exactly zero.
Smallest non-zero branch length: 0.34885 
Number of branches &lt; 1e-8 (a very small number): 2 
Is tree_pruned binary (no polytomies)? TRUE </code></pre>
<pre class="r"><code>if (exists(&quot;tree_pruned&quot;) &amp;&amp; inherits(tree_pruned, &quot;phylo&quot;)) {
    if(any(tree_pruned$edge.length == 0)) {
    cat(&quot;Adding a tiny value to zero-length branches...\n&quot;)
    tree_pruned$edge.length[tree_pruned$edge.length == 0] &lt;- 1e-8 # Or a similar small value
     # After this, you would need to recreate data_comp with the modified tree_pruned
    data_comp &lt;- comparative.data(tree_pruned, data, Species, vcv=TRUE)
   }
 }</code></pre>
<pre><code>Adding a tiny value to zero-length branches...</code></pre>
<pre class="r"><code>summary(pgls(FrA.H ~ Temp_avg, data_comp, lambda=&quot;ML&quot;))</code></pre>
<pre><code>
Call:
pgls(formula = FrA.H ~ Temp_avg, data = data_comp, lambda = &quot;ML&quot;)

Residuals:
       Min         1Q     Median         3Q        Max 
-0.0058727 -0.0020066  0.0003919  0.0024126  0.0088301 

Branch length transformations:

kappa  [Fix]  : 1.000
lambda [ ML]  : 1.000
   lower bound : 0.000, p = 1.979e-12
   upper bound : 1.000, p = 1    
   95.0% CI   : (0.980, NA)
delta  [Fix]  : 1.000

Coefficients:
               Estimate  Std. Error t value  Pr(&gt;|t|)    
(Intercept)  0.18296519  0.03604914  5.0754 3.567e-06 ***
Temp_avg    -0.00094915  0.00064487 -1.4719     0.146    
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 0.003447 on 64 degrees of freedom
Multiple R-squared: 0.03274,    Adjusted R-squared: 0.01763 
F-statistic: 2.166 on 1 and 64 DF,  p-value: 0.146 </code></pre>
<pre class="r"><code>summary(pgls(AGskew ~ Temp_avg, data_comp, lambda=&quot;ML&quot;))</code></pre>
<pre><code>
Call:
pgls(formula = AGskew ~ Temp_avg, data = data_comp, lambda = &quot;ML&quot;)

Residuals:
      Min        1Q    Median        3Q       Max 
-0.021054 -0.007901  0.000493  0.007265  0.052588 

Branch length transformations:

kappa  [Fix]  : 1.000
lambda [ ML]  : 1.000
   lower bound : 0.000, p = 2.9916e-05
   upper bound : 1.000, p = 1    
   95.0% CI   : (0.980, NA)
delta  [Fix]  : 1.000

Coefficients:
              Estimate Std. Error t value Pr(&gt;|t|)  
(Intercept) -0.3089393  0.1394460 -2.2155  0.03029 *
Temp_avg    -0.0029978  0.0024945 -1.2018  0.23388  
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 0.01333 on 64 degrees of freedom
Multiple R-squared: 0.02207,    Adjusted R-squared: 0.006789 
F-statistic: 1.444 on 1 and 64 DF,  p-value: 0.2339 </code></pre>
<pre class="r"><code>summary(pgls(FrG.H ~ Temp_avg, data_comp, lambda=&quot;ML&quot;))</code></pre>
<pre><code>
Call:
pgls(formula = FrG.H ~ Temp_avg, data = data_comp, lambda = &quot;ML&quot;)

Residuals:
       Min         1Q     Median         3Q        Max 
-0.0150385 -0.0026115  0.0002822  0.0026428  0.0079230 

Branch length transformations:

kappa  [Fix]  : 1.000
lambda [ ML]  : 1.000
   lower bound : 0.000, p = 2.4425e-15
   upper bound : 1.000, p = 1    
   95.0% CI   : (0.987, NA)
delta  [Fix]  : 1.000

Coefficients:
              Estimate Std. Error t value  Pr(&gt;|t|)    
(Intercept) 0.34156454 0.04482866  7.6193 1.512e-10 ***
Temp_avg    0.00075506 0.00080192  0.9416      0.35    
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 0.004286 on 64 degrees of freedom
Multiple R-squared: 0.01366,    Adjusted R-squared: -0.001749 
F-statistic: 0.8865 on 1 and 64 DF,  p-value: 0.35 </code></pre>
<pre class="r"><code>#summary(pgls(FrT.H ~ Temp_avg, data_comp, lambda=&quot;ML&quot;))
summary(pgls(FrC.H ~ Temp_avg, data_comp, lambda=&quot;ML&quot;))</code></pre>
<pre><code>
Call:
pgls(formula = FrC.H ~ Temp_avg, data = data_comp, lambda = &quot;ML&quot;)

Residuals:
       Min         1Q     Median         3Q        Max 
-0.0018535 -0.0007831 -0.0001233  0.0006779  0.0035278 

Branch length transformations:

kappa  [Fix]  : 1.000
lambda [ ML]  : 0.000
   lower bound : 0.000, p = 1    
   upper bound : 1.000, p = 0.001973
   95.0% CI   : (NA, 0.909)
delta  [Fix]  : 1.000

Coefficients:
               Estimate  Std. Error t value Pr(&gt;|t|)    
(Intercept)  0.09092808  0.00873541 10.4091 2.22e-15 ***
Temp_avg    -0.00086470  0.00049721 -1.7391  0.08683 .  
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 0.001145 on 64 degrees of freedom
Multiple R-squared: 0.04512,    Adjusted R-squared: 0.0302 
F-statistic: 3.024 on 1 and 64 DF,  p-value: 0.08683 </code></pre>
<pre class="r"><code>summary(pgls(CTskew ~ Temp_avg, data_comp, lambda=&quot;ML&quot;))</code></pre>
<pre><code>
Call:
pgls(formula = CTskew ~ Temp_avg, data = data_comp, lambda = &quot;ML&quot;)

Residuals:
      Min        1Q    Median        3Q       Max 
-0.017516 -0.006179 -0.002185  0.005706  0.023230 

Branch length transformations:

kappa  [Fix]  : 1.000
lambda [ ML]  : 0.817
   lower bound : 0.000, p = 0.17567
   upper bound : 1.000, p = 0.065069
   95.0% CI   : (NA, NA)
delta  [Fix]  : 1.000

Coefficients:
              Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept) -0.5926456  0.0880094 -6.7339 5.44e-09 ***
Temp_avg    -0.0043311  0.0024717 -1.7523  0.08451 .  
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 0.008231 on 64 degrees of freedom
Multiple R-squared: 0.04578,    Adjusted R-squared: 0.03087 
F-statistic: 3.071 on 1 and 64 DF,  p-value: 0.08451 </code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.5.0 (2025-04-11)
Platform: aarch64-apple-darwin20
Running under: macOS Sequoia 15.5

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.5-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.5-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.1

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Vienna
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] rfishbase_5.0.1    ggstatsplot_0.13.1 broom_1.0.8        ggpubr_0.6.0      
 [5] ggExtra_0.10.1     geiger_2.0.11      phytools_2.4-4     maps_3.4.3        
 [9] caper_1.0.3        mvtnorm_1.3-3      MASS_7.3-65        ape_5.8-1         
[13] magrittr_2.0.3     lubridate_1.9.4    forcats_1.0.0      stringr_1.5.1     
[17] dplyr_1.1.4        purrr_1.0.4        readr_2.1.5        tidyr_1.3.1       
[21] tibble_3.2.1       ggplot2_3.5.2      tidyverse_2.0.0    knitr_1.50        
[25] here_1.0.1         workflowr_1.7.1   

loaded via a namespace (and not attached):
  [1] RColorBrewer_1.1-3      rstudioapi_0.17.1       jsonlite_2.0.0         
  [4] datawizard_1.1.0        correlation_0.8.7       subplex_1.9            
  [7] estimability_1.5.1      farver_2.1.2            rmarkdown_2.29         
 [10] fs_1.6.6                vctrs_0.6.5             memoise_2.0.1          
 [13] paletteer_1.6.0         effectsize_1.0.1        rstatix_0.7.2          
 [16] htmltools_0.5.8.1       curl_6.2.3              DEoptim_2.2-8          
 [19] deSolve_1.40            Formula_1.2-5           sass_0.4.10            
 [22] bslib_0.9.0             emmeans_1.11.1          cachem_1.1.0           
 [25] whisker_0.4.1           igraph_2.1.4            mime_0.13              
 [28] lifecycle_1.0.4         iterators_1.0.14        pkgconfig_2.0.3        
 [31] Matrix_1.7-3            R6_2.6.1                fastmap_1.2.0          
 [34] shiny_1.10.0            digest_0.6.37           numDeriv_2016.8-1.1    
 [37] rematch2_2.1.2          patchwork_1.3.0         ps_1.9.1               
 [40] rprojroot_2.0.4         labeling_0.4.3          clusterGeneration_1.3.8
 [43] timechange_0.3.0        httr_1.4.7              abind_1.4-8            
 [46] mgcv_1.9-3              compiler_4.5.0          withr_3.0.2            
 [49] doParallel_1.0.17       duckdbfs_0.1.0          backports_1.5.0        
 [52] optimParallel_1.0-2     carData_3.0-5           DBI_1.2.3              
 [55] duckdb_1.3.0            ggsignif_0.6.4          scatterplot3d_0.3-44   
 [58] tools_4.5.0             httpuv_1.6.16           statsExpressions_1.7.0 
 [61] glue_1.8.0              quadprog_1.5-8          callr_3.7.6            
 [64] nlme_3.1-168            promises_1.3.2          grid_4.5.0             
 [67] getPass_0.2-4           generics_0.1.4          gtable_0.3.6           
 [70] tzdb_0.5.0              hms_1.1.3               car_3.1-3              
 [73] utf8_1.2.5              foreach_1.5.2           pillar_1.10.2          
 [76] later_1.4.2             splines_4.5.0           lattice_0.22-7         
 [79] tidyselect_1.2.1        miniUI_0.1.2            git2r_0.36.2           
 [82] xfun_0.52               expm_1.0-0              stringi_1.8.7          
 [85] yaml_2.3.10             evaluate_1.0.3          codetools_0.2-20       
 [88] cli_3.6.5               RcppParallel_5.1.10     xtable_1.8-4           
 [91] parameters_0.26.0       processx_3.8.6          jquerylib_0.1.4        
 [94] Rcpp_1.0.14             zeallot_0.2.0           dbplyr_2.5.0           
 [97] coda_0.19-4.1           parallel_4.5.0          rstantools_2.4.0       
[100] blob_1.2.4              bayestestR_0.16.0       phangorn_2.12.1        
[103] scales_1.4.0            insight_1.3.0           combinat_0.0-8         
[106] rlang_1.1.6             fastmatch_1.1-6         mnormt_2.1.1           </code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
