<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Alina G. Mikhailova" />

<meta name="date" content="2025-06-02" />

<title>Analysis of the mitochondrial mutational spectra between different classes</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">TemperatureEffectsOnMtDNAspectra</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="I_Polymorphic_Data_Analysis.html">Analysis of the polymorphic data in Fishes obtained by NeMu pipeline</a>
    </li>
  </ul>
</li>
<li>
  <a href="methods.html">Methods</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="license.html">License</a>
</li>
<li>
  <a href="https://github.com/mitoclub/TemperatureEffectsOnMtDNAspectra">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Analysis of the mitochondrial mutational
spectra between different classes</h1>
<h4 class="author">Alina G. Mikhailova</h4>
<h4 class="date">2025-06-02</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-06-02
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong>
<code>TemperatureEffectsOnMtDNAspectra/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240719code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240719)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240719code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240719)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcommitoclubTemperatureEffectsOnMtDNAspectratree530b2a62fc373b56fd5e6ef0b3970c3a6d4e274ctargetblank530b2a6a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/mitoclub/TemperatureEffectsOnMtDNAspectra/tree/530b2a62fc373b56fd5e6ef0b3970c3a6d4e274c" target="_blank">530b2a6</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcommitoclubTemperatureEffectsOnMtDNAspectratree530b2a62fc373b56fd5e6ef0b3970c3a6d4e274ctargetblank530b2a6a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/mitoclub/TemperatureEffectsOnMtDNAspectra/tree/530b2a62fc373b56fd5e6ef0b3970c3a6d4e274c" target="_blank">530b2a6</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/.RData
    Ignored:    analysis/.Rhistory
    Ignored:    code/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    data/AllGenesCodonUsageNoOverlap.txt
    Ignored:    data/ColdAndWarmFishesSpectra.txt
    Ignored:    data/Dataset_S1.csv
    Ignored:    data/ExpectedFractions.txt
    Ignored:    data/Extra_climate_data_2025.csv
    Ignored:    data/Extra_temperature_grow_data_2025.csv
    Ignored:    data/Extra_temperature_popQB_data_2025.csv
    Ignored:    data/Full_fish_ecology_table_2025.csv
    Ignored:    data/MIDORI2_LONGEST_NUC_GB265_Cytb_BLAST.fasta
    Ignored:    data/MIDORI2_LONGEST_NUC_GB265_Cytb_BLAST.fasta.fai
    Ignored:    data/MIDORI_WG_CYTB_FISHES.csv
    Ignored:    data/SimulationData.csv
    Ignored:    output/.DS_Store
    Ignored:    output/tables/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/III_Comparison_of_all_Classes.Rmd</code>) and HTML
(<code>docs/III_Comparison_of_all_Classes.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mitoclub/TemperatureEffectsOnMtDNAspectra/blob/530b2a62fc373b56fd5e6ef0b3970c3a6d4e274c/analysis/III_Comparison_of_all_Classes.Rmd" target="_blank">530b2a6</a>
</td>
<td>
Alya Mikhailova
</td>
<td>
2025-06-02
</td>
<td>
fix chunk name after failed render
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mitoclub/TemperatureEffectsOnMtDNAspectra/530b2a62fc373b56fd5e6ef0b3970c3a6d4e274c/docs/III_Comparison_of_all_Classes.html" target="_blank">530b2a6</a>
</td>
<td>
Alya Mikhailova
</td>
<td>
2025-06-02
</td>
<td>
fix chunk name after failed render
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mitoclub/TemperatureEffectsOnMtDNAspectra/blob/ad28257961a99798122a3b968b1be129307f6f34/analysis/III_Comparison_of_all_Classes.Rmd" target="_blank">ad28257</a>
</td>
<td>
Alya Mikhailova
</td>
<td>
2025-06-02
</td>
<td>
Violin plots for tempersture and A&gt;G
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div
id="reading-mutation-spectra-database-l-strand-and-ecological-databases"
class="section level1">
<h1>Reading mutation spectra database (L-strand) and ecological
databases</h1>
<pre class="r"><code>mutSpec12all &lt;- read_csv(here(data_dir, &quot;MutSpecVertebrates12all.csv.gz&quot;))
mutSpec12allCb &lt;- dplyr::filter(mutSpec12all, Gene == &quot;Cytb&quot;)

MutSpecForAnalysisCbAll &lt;- mutSpec12allCb %&gt;% dplyr::select(Class, Species, Mut, MutSpec)
MutSpecForAnalysisCbAll$Mut &lt;- gsub(&quot;&gt;&quot;, &quot;_&quot;, MutSpecForAnalysisCbAll$Mut)
MutSpecForAnalysisCbAll &lt;- MutSpecForAnalysisCbAll %&gt;% pivot_wider(names_from = Mut, values_from = MutSpec, , values_fn = median) 
colnames(MutSpecForAnalysisCbAll) &lt;- c(&quot;Class&quot;,
                                  &quot;Species&quot;, 
                                  &quot;T_G.H&quot;, 
                                  &quot;T_C.H&quot;, 
                                  &quot;T_A.H&quot;, 
                                  &quot;G_T.H&quot;, 
                                  &quot;G_C.H&quot;, 
                                  &quot;G_A.H&quot;, 
                                  &quot;C_T.H&quot;, 
                                  &quot;C_G.H&quot;, 
                                  &quot;C_A.H&quot;, 
                                  &quot;A_T.H&quot;, 
                                  &quot;A_G.H&quot;, 
                                  &quot;A_C.H&quot;)</code></pre>
<pre class="r"><code>fishEcology &lt;- read.csv(here(data_dir, &quot;Full_fish_ecology_table_2025.csv&quot;))
head(fishEcology)</code></pre>
<pre><code>                  Species     Wmax      Lmax     tmax
1     Abbottina_rivularis       NA   9.10000  2.00000
2           Abramis_brama 4112.167  46.91111 14.41667
3    Abudefduf_vaigiensis       NA  18.96667       NA
4 Acanthemblemaria_aspera       NA   3.00000  4.00000
5  Acanthocybium_solandri 7200.000 177.88000  9.30000
6     Acanthopagrus_berda  660.000  56.00000 12.50000
                    FeedingType FoodTroph    tm       Lm Temp_min Temp_max
1                          &lt;NA&gt;        NA 1.000  4.50000       NA       NA
2 hunting macrofauna (predator)      3.15 4.125 28.75000       10       24
3     grazing on aquatic plants      3.39    NA 12.00000       NA       NA
4 hunting macrofauna (predator)      3.12    NA       NA       NA       NA
5 hunting macrofauna (predator)      4.45 2.000 99.33333       NA       NA
6 hunting macrofauna (predator)      3.50    NA 21.00000       NA       NA
  Climate_zone Temp_avg
1  subtropical       NA
2    temperate       17
3     tropical       NA
4         &lt;NA&gt;       NA
5  subtropical       NA
6     tropical       NA</code></pre>
<pre class="r"><code>fishEcologyTemp &lt;- fishEcology %&gt;% dplyr::select(Species, Temp_avg)
fishEcologyTemp &lt;- na.omit(fishEcologyTemp)
fishEcologyTemp$Class &lt;- &quot;Fishes&quot;
allEcology &lt;-read.csv(here(data_dir, &quot;Dataset_S1.csv&quot;), sep=&quot;;&quot;)
allEcology$Temp_avg &lt;- as.numeric(gsub(&quot;,&quot;, &quot;.&quot;, allEcology$Tb))
allEcology$Tb &lt;- NULL
allEcology &lt;- rbind(allEcology, fishEcologyTemp)
todelete &lt;- c(
    &quot;Testudines&quot;,
    &quot;Crocodylia&quot;)
allEcology &lt;- allEcology[!allEcology$Class %in% todelete,]
allEcology &lt;- allEcology %&gt;% mutate(
  Class = fct_relevel(
    Class,
    &quot;Fishes&quot;,
    &quot;Amphibia&quot;,
    &quot;Lepidosauria&quot;,
    &quot;Mammalia&quot;, 
    &quot;Aves&quot;
  )
)

allEcologyNOCLASS &lt;- allEcology 
allEcologyNOCLASS$Class &lt;- NULL</code></pre>
</div>
<div id="comparison-temperatures-of-all-classes" class="section level1">
<h1>Comparison temperatures of all Classes</h1>
<pre class="r"><code>plt &lt;- ggbetweenstats(data = allEcology, x = Class, y = Temp_avg) +
  labs(x = &quot;Classes&quot;, y = &quot;Temp_avg&quot;) +
  scale_color_manual(values = c(
      &quot;#6760db&quot;,
      &quot;#7849bf&quot;,
      &quot;#9145c4&quot;,
      &quot;#c73a69&quot;,
      &quot;#c2464c&quot;))

plt</code></pre>
<p><img src="figure/III_Comparison_of_all_Classes.Rmd/violin%20temperature-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-violin-temperature-1">
Past versions of "violin temperature-1.png"
</button>
</p>
<div id="fig-no-spaces-violin-temperature-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mitoclub/TemperatureEffectsOnMtDNAspectra/blob/530b2a62fc373b56fd5e6ef0b3970c3a6d4e274c/docs/figure/III_Comparison_of_all_Classes.Rmd/violin temperature-1.png" target="_blank">530b2a6</a>
</td>
<td>
Alya Mikhailova
</td>
<td>
2025-06-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="comparison-ag-of-all-classes" class="section level1">
<h1>Comparison A&gt;G of all Classes</h1>
<pre class="r"><code>merging &lt;- list(MutSpecForAnalysisCbAll, allEcologyNOCLASS)
MutSpecForAnalysisCbAll &lt;- merging %&gt;% purrr::reduce(left_join, by= &quot;Species&quot;)
MutSpecForAnalysisCbAll[MutSpecForAnalysisCbAll$Class == &quot;Actinopteri&quot;,]$Class &lt;- &quot;Fishes&quot;
MutSpecForAnalysisCbAll &lt;- MutSpecForAnalysisCbAll %&gt;% mutate(Class = fct_relevel(
  Class,
  &quot;Fishes&quot;,
  &quot;Amphibia&quot;,
  &quot;Lepidosauria&quot;,
  &quot;Mammalia&quot;,
  &quot;Aves&quot;
))

str(MutSpecForAnalysisCbAll$Class)</code></pre>
<pre><code> Factor w/ 5 levels &quot;Fishes&quot;,&quot;Amphibia&quot;,..: 4 3 4 1 4 4 4 4 3 4 ...</code></pre>
<pre class="r"><code>plt2 &lt;- ggbetweenstats(data = MutSpecForAnalysisCbAll, x = Class, y = A_G.H) +
  labs(x = &quot;Classes&quot;, y = &quot;A&gt;G.heavy&quot;) +
  scale_color_manual(values = c(
      &quot;#6760db&quot;,
      &quot;#7849bf&quot;,
      &quot;#9145c4&quot;,
      &quot;#c73a69&quot;,
      &quot;red&quot;))

plt2</code></pre>
<p><img src="figure/III_Comparison_of_all_Classes.Rmd/violin-A2G-1.png" width="1344" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.5.0 (2025-04-11)
Platform: aarch64-apple-darwin20
Running under: macOS Sequoia 15.5

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.5-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.5-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.1

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Vienna
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] skimr_2.1.5        ggstatsplot_0.13.1 broom_1.0.8        ggpubr_0.6.0      
 [5] ggExtra_0.10.1     geiger_2.0.11      phytools_2.4-4     maps_3.4.3        
 [9] caper_1.0.3        mvtnorm_1.3-3      MASS_7.3-65        ape_5.8-1         
[13] magrittr_2.0.3     lubridate_1.9.4    forcats_1.0.0      stringr_1.5.1     
[17] dplyr_1.1.4        purrr_1.0.4        readr_2.1.5        tidyr_1.3.1       
[21] tibble_3.2.1       ggplot2_3.5.2      tidyverse_2.0.0    knitr_1.50        
[25] here_1.0.1         workflowr_1.7.1   

loaded via a namespace (and not attached):
  [1] RColorBrewer_1.1-3      rstudioapi_0.17.1       jsonlite_2.0.0         
  [4] datawizard_1.1.0        correlation_0.8.7       subplex_1.9            
  [7] SuppDists_1.1-9.9       estimability_1.5.1      farver_2.1.2           
 [10] rmarkdown_2.29          fs_1.6.6                vctrs_0.6.5            
 [13] memoise_2.0.1           paletteer_1.6.0         base64enc_0.1-3        
 [16] effectsize_1.0.1        rstatix_0.7.2           htmltools_0.5.8.1      
 [19] DEoptim_2.2-8           deSolve_1.40            BWStest_0.2.3          
 [22] Formula_1.2-5           sass_0.4.10             bslib_0.9.0            
 [25] emmeans_1.11.1          cachem_1.1.0            whisker_0.4.1          
 [28] igraph_2.1.4            mime_0.13               lifecycle_1.0.4        
 [31] iterators_1.0.14        pkgconfig_2.0.3         Matrix_1.7-3           
 [34] R6_2.6.1                fastmap_1.2.0           PMCMRplus_1.9.12       
 [37] shiny_1.10.0            BayesFactor_0.9.12-4.7  digest_0.6.37          
 [40] numDeriv_2016.8-1.1     rematch2_2.1.2          patchwork_1.3.0        
 [43] ps_1.9.1                rprojroot_2.0.4         labeling_0.4.3         
 [46] clusterGeneration_1.3.8 timechange_0.3.0        httr_1.4.7             
 [49] abind_1.4-8             compiler_4.5.0          bit64_4.6.0-1          
 [52] withr_3.0.2             doParallel_1.0.17       backports_1.5.0        
 [55] optimParallel_1.0-2     carData_3.0-5           performance_0.14.0     
 [58] ggsignif_0.6.4          scatterplot3d_0.3-44    tools_4.5.0            
 [61] httpuv_1.6.16           statsExpressions_1.7.0  glue_1.8.0             
 [64] quadprog_1.5-8          callr_3.7.6             nlme_3.1-168           
 [67] promises_1.3.2          grid_4.5.0              getPass_0.2-4          
 [70] generics_0.1.4          gtable_0.3.6            tzdb_0.5.0             
 [73] hms_1.1.3               car_3.1-3               ggrepel_0.9.6          
 [76] foreach_1.5.2           pillar_1.10.2           vroom_1.6.5            
 [79] later_1.4.2             lattice_0.22-7          gmp_0.7-5              
 [82] bit_4.6.0               tidyselect_1.2.1        miniUI_0.1.2           
 [85] pbapply_1.7-2           git2r_0.36.2            xfun_0.52              
 [88] expm_1.0-0              stringi_1.8.7           boot_1.3-31            
 [91] yaml_2.3.10             kSamples_1.2-10         evaluate_1.0.3         
 [94] codetools_0.2-20        multcompView_0.1-10     cli_3.6.5              
 [97] RcppParallel_5.1.10     xtable_1.8-4            parameters_0.26.0      
[100] repr_1.1.7              processx_3.8.6          jquerylib_0.1.4        
[103] Rcpp_1.0.14             zeallot_0.2.0           coda_0.19-4.1          
[106] parallel_4.5.0          rstantools_2.4.0        MatrixModels_0.5-4     
[109] bayestestR_0.16.0       Rmpfr_1.1-0             phangorn_2.12.1        
[112] scales_1.4.0            insight_1.3.0           crayon_1.5.3           
[115] combinat_0.0-8          rlang_1.1.6             fastmatch_1.1-6        
[118] mnormt_2.1.1           </code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
