---
title: "Analysis of the polymorphic data in Fishes obtained by NeMu pipeline"
author: "Alina G. Mikhailova"
date: "`r Sys.Date()`"
output:
    workflowr::wflow_html:
    toc: true
knitr:
  opts_chunk:
    echo: false
---

```{r setup, echo=FALSE, include=FALSE}
docname <- "I_Polymorphic_Data_Analysis"
now <- Sys.time()
# Time chunks during knitting
knitr::knit_hooks$set(timeit = function(before) {
  if (before) {
    print(paste("Start:", Sys.time()))
    now <<- Sys.time()
  } else {
    print(paste("Stop:", Sys.time()))
    print(Sys.time() - now)
  }
})

knitr::opts_chunk$set(
  autodep        = TRUE,
  cache          = FALSE,
  cache.path     = paste0("cache/", docname, "/"),
  cache.comments = FALSE,
  cache.lazy     = FALSE,
  dev            = c("png", "pdf"),
  echo           = TRUE,
  error          = FALSE,
  fig.align      = "center",
  fig.width      = 14,
  fig.height     = 12,
  message        = FALSE,
  warning        = FALSE,
  timeit         = TRUE
)
# Load packages
suppressPackageStartupMessages({
  library(here)
  library(knitr)
  library(tidyverse)
  library(magrittr)
  library(caper)
  library(geiger)
  library(ggExtra)
  library(ggpubr)
  library(broom)
  library(phytools)
  library(ggstatsplot)
  library(rfishbase)
})

# Set paths
src_dir <- here("code")
data_dir <- here("data")
output_dir <- here("output")
plots_dir <- here(output_dir, "figures")
tables_dir <- here(output_dir, "tables")

# parallelisation
n_cores <- 4

# set seed
reseed <- 13
set.seed(seed = reseed)
```

# Reading mutation spectra database (L-strand) and ecological databases

```{r spectra, message=FALSE}
mutSpec12 <- read_csv(here(data_dir, "MutSpecVertebrates12fish.csv"))
mutSpec12Cb <- dplyr::filter(mutSpec12, Gene == "Cytb")
cat("N species =", length(unique(mutSpec12Cb$Species)))
head(mutSpec12Cb)

MutSpecForAnalysisCb <- mutSpec12Cb %>% dplyr::select(Class, Species, Mut, MutSpec)
MutSpecForAnalysisCb$Mut <- gsub(">", "_", MutSpecForAnalysisCb$Mut)
MutSpecForAnalysisCb <- MutSpecForAnalysisCb %>% pivot_wider(names_from = Mut, values_from = MutSpec, , values_fn = median) 
colnames(MutSpecForAnalysisCb) <- c("Class",
                                  "Species", 
                                  "T_G.H", 
                                  "T_C.H", 
                                  "T_A.H", 
                                  "G_T.H", 
                                  "G_C.H", 
                                  "G_A.H", 
                                  "C_T.H", 
                                  "C_G.H", 
                                  "C_A.H", 
                                  "A_T.H", 
                                  "A_G.H", 
                                  "A_C.H")

MutSpecForAnalysis <- mutSpec12 %>% dplyr::select(Class, Species, Mut, MutSpec)
MutSpecForAnalysis <- mutSpec12 %>% dplyr::select(Class, Species, Mut, MutSpec)
MutSpecForAnalysis$Mut <- gsub(">", "_", MutSpecForAnalysis$Mut)
MutSpecForAnalysis <- MutSpecForAnalysis %>% pivot_wider(names_from = Mut, values_from = MutSpec, , values_fn = median) 
colnames(MutSpecForAnalysis) <- c("Class",
                                  "Species", 
                                  "T_G.H", 
                                  "T_C.H", 
                                  "T_A.H", 
                                  "G_T.H", 
                                  "G_C.H", 
                                  "G_A.H", 
                                  "C_T.H", 
                                  "C_G.H", 
                                  "C_A.H", 
                                  "A_T.H", 
                                  "A_G.H", 
                                  "A_C.H")
```

```{r ecology}
fishEcology <- read.csv(here(data_dir, "Full_fish_ecology_table_2025.csv"))
head(fishEcology)

cor.test(fishEcology$tm, fishEcology$Temp_avg)
```

# Calculating of average MutSpec for Fisches
## Cytb

```{r plots average cytb, message=FALSE}
AverageMutSpecCb <- mutSpec12Cb %>% group_by(Mut) %>% summarise(meanmutspec = mean(MutSpec, na.rm = TRUE))
AverageMutSpecCb$Mut.H <- c("T_G.H", 
                          "T_C.H", 
                          "T_A.H", 
                          "G_T.H", 
                          "G_C.H", 
                          "G_A.H", 
                          "C_T.H", 
                          "C_G.H", 
                          "C_A.H", 
                          "A_T.H", 
                          "A_G.H", 
                          "A_C.H")

f1a = ggplot(AverageMutSpecCb,
  aes(
    x = AverageMutSpecCb$Mut.H,
    y = AverageMutSpecCb$meanmutspec,
    fill = AverageMutSpecCb$Mut.H
    )
  ) +
  geom_bar(stat = "identity") +
  scale_fill_manual(
    values = c(
      "#bdbdbd",
      "#00a426",
      "#bdbdbd",
      "#bdbdbd",
      "#bdbdbd",
      "#ff0000",
      "#900000",
      "#bdbdbd",
      "#bdbdbd",
      "#bdbdbd",
      "#005213",
      "#bdbdbd"
    )
  ) +
  labs(title = "Average mitochondrial mutational spectrum in Fishes", x = "Substitution types", y = "Normalised Frequencies") + theme(
    legend.position = "none",
    # Remove legend
    panel.background = element_blank(),
    axis.line = element_line(color = "black")# Remove background
  )

ggsave(here(plots_dir,"PolymorphicData.Actinopterygii.AverageMutSpecCytB.pdf"),
  plot = f1a,
  width = 6,
  height = 6
)
f1a
```
## Cytb+COX1

```{r plots average all, message=FALSE}
AverageMutSpec <- mutSpec12 %>% group_by(Mut) %>% summarise(meanmutspec = mean(MutSpec, na.rm = TRUE))
AverageMutSpec$Mut.H <- c("T_G.H", 
                          "T_C.H", 
                          "T_A.H", 
                          "G_T.H", 
                          "G_C.H", 
                          "G_A.H", 
                          "C_T.H", 
                          "C_G.H", 
                          "C_A.H", 
                          "A_T.H", 
                          "A_G.H", 
                          "A_C.H")

sf1 = ggplot(AverageMutSpec,
  aes(
    x = AverageMutSpec$Mut.H,
    y = AverageMutSpec$meanmutspec,
    fill = AverageMutSpec$Mut.H
    )
  ) +
  geom_bar(stat = "identity") +
  scale_fill_manual(
    values = c(
      "#bdbdbd",
      "#00a426",
      "#bdbdbd",
      "#bdbdbd",
      "#bdbdbd",
      "#ff0000",
      "#900000",
      "#bdbdbd",
      "#bdbdbd",
      "#bdbdbd",
      "#005213",
      "#bdbdbd"
    )
  ) +
  labs(title = "Average mitochondrial mutational spectrum in Fishes", x = "Substitution types", y = "Normalised Frequencies") + theme(
    legend.position = "none",
    # Remove legend
    panel.background = element_blank(),
    axis.line = element_line(color = "black")# Remove background
  )

#ggsave(here(plots_dir,"PolymorphicData.Actinopterygii.AverageMutSpec.pdf"),
#  plot = sf1,
#  width = 6,
#  height = 6
#)
sf1
```

# Comparison of average MutSpec between cold and warm fisches
##Cytb

```{r barplots cold warm cytb, message=FALSE, warning= FALSE}
distances <- read.table(here(data_dir, "Cytb.nm.Mafft_L-INS-i.pr.Bs.mtVer-F-I-G8.treefile.cut-c0.0001-rr.dist"), header = FALSE)
colnames(distances) <- c("Species", "Rates")

merging <- list(MutSpecForAnalysisCb, fishEcology, distances)
MutAndEcoCb <- merging %>% reduce(left_join, by= "Species")
temperaturestat <- summary(MutAndEcoCb$Temp_avg)

WarmFishes <- filter(MutAndEcoCb, Temp_avg > as.numeric(temperaturestat["3rd Qu."]))
WarmFishes <- pivot_longer(WarmFishes, cols = c("T_G.H", 
                                  "T_C.H", 
                                  "T_A.H", 
                                  "G_T.H", 
                                  "G_C.H", 
                                  "G_A.H", 
                                  "C_T.H", 
                                  "C_G.H", 
                                  "C_A.H", 
                                  "A_T.H", 
                                  "A_G.H", 
                                  "A_C.H"), names_to = "Mut", values_to = "MutSpec" )
AverageMutSpecWarm <- WarmFishes %>% group_by(Mut) %>% summarise(meanmutspec = mean(MutSpec, na.rm = TRUE), error = 
                                                                 sd(MutSpec, na.rm = TRUE))
Nwarm <- length(unique(WarmFishes$Species))

ColdFishes <- filter(MutAndEcoCb, Temp_avg < as.numeric(temperaturestat["1st Qu."]))
ColdFishes <- pivot_longer(ColdFishes, cols = c("T_G.H", 
                                  "T_C.H", 
                                  "T_A.H", 
                                  "G_T.H", 
                                  "G_C.H", 
                                  "G_A.H", 
                                  "C_T.H", 
                                  "C_G.H", 
                                  "C_A.H", 
                                  "A_T.H", 
                                  "A_G.H", 
                                  "A_C.H"), names_to = "Mut", values_to = "MutSpec" )
AverageMutSpecCold <- ColdFishes %>% group_by(Mut) %>% summarise(meanmutspec = mean(MutSpec, na.rm = TRUE), error =
                                                                sd(MutSpec, na.rm = TRUE))
Ncold <- length(unique(ColdFishes$Species))

f1bbw = ggplot(AverageMutSpecWarm,
  aes(
    x = AverageMutSpecWarm$Mut,
    y = AverageMutSpecWarm$meanmutspec,
    fill = AverageMutSpecWarm$Mut
    )
  ) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values = c(
                    "#bdbdbd",
                    "#00a426",
                    "#bdbdbd",
                    "#bdbdbd",
                    "#bdbdbd",
                    "#ff0000",
                    "#900000",
                    "#bdbdbd",
                    "#bdbdbd",
                    "#bdbdbd",
                    "#005213",
                    "#bdbdbd")) +
  labs(title = "Average mitochondrial mutational spectrum in Warm Fishes, <23C", x = "Substitution types", y = "Normalised Frequencies") + 
  theme(legend.position = "none",                        # Remove legend
        panel.background = element_blank(),
        axis.line = element_line(color = "black")) +     # Remove background
  geom_text(x=2, y=0.3, label=paste("N warm =", Nwarm), size=7) +
  geom_errorbar(aes(x=AverageMutSpecWarm$Mut, ymin=AverageMutSpecWarm$meanmutspec-AverageMutSpecWarm$error,
                    ymax=AverageMutSpecWarm$meanmutspec+AverageMutSpecWarm$error), width=0.4, colour="navy", alpha=0.9,
                    size=1.3)


f1bbc = ggplot(AverageMutSpecCold,
  aes(
    x = AverageMutSpecCold$Mut,
    y = AverageMutSpecCold$meanmutspec,
    fill = AverageMutSpecCold$Mut
    )
  ) +
  geom_bar(stat = "identity") +
  scale_fill_manual(
    values = c(
      "#bdbdbd",
      "#00a426",
      "#bdbdbd",
      "#bdbdbd",
      "#bdbdbd",
      "#ff0000",
      "#900000",
      "#bdbdbd",
      "#bdbdbd",
      "#bdbdbd",
      "#005213",
      "#bdbdbd"
    )
  ) +
  labs(title = "Average mitochondrial mutational spectrum in Cold Fishes, > 13C ", x = "Substitution types", y = "Normalised Frequencies") + theme(
    legend.position = "none",
    # Remove legend
    panel.background = element_blank(),
    axis.line = element_line(color = "black")# Remove background
  )+ 
  geom_text(x=2, y=0.3, label=paste("N cold =", Ncold), size = 7)+
  geom_errorbar(aes(x=AverageMutSpecCold$Mut, ymin=AverageMutSpecCold$meanmutspec-AverageMutSpecCold$error,
                    ymax=AverageMutSpecCold$meanmutspec+AverageMutSpecCold$error), width=0.4, colour="navy", alpha=0.9,
                    size=1.3)

both <- ggarrange(f1bbc, f1bbw, ncol = 2)
ggsave(here(plots_dir,"PolymorphicData.Actinopterygii.AverageMutSpecColdAndWarmCytB.pdf"),
  plot = both,
  width = 12,
  height = 6
)
both
```

## Cytb+COX1

```{r barplots cold warm, message=FALSE, warning= FALSE}
merging <- list(MutSpecForAnalysis, fishEcology)
MutAndEco <- merging %>% reduce(left_join, by= "Species")
temperaturestat <- summary (MutAndEco$Temp_avg)

WarmFishes <- filter(MutAndEco, Temp_avg > as.numeric(temperaturestat["3rd Qu."]))
WarmFishes <- pivot_longer(WarmFishes, cols = c("T_G.H", 
                                  "T_C.H", 
                                  "T_A.H", 
                                  "G_T.H", 
                                  "G_C.H", 
                                  "G_A.H", 
                                  "C_T.H", 
                                  "C_G.H", 
                                  "C_A.H", 
                                  "A_T.H", 
                                  "A_G.H", 
                                  "A_C.H"), names_to = "Mut", values_to = "MutSpec" )
AverageMutSpecWarm <- WarmFishes %>% group_by(Mut) %>% summarise(meanmutspec = mean(MutSpec, na.rm = TRUE), error =
                                                                sd(MutSpec, na.rm = TRUE))
Nwarm <- length(unique(WarmFishes$Species))

ColdFishes <- filter(MutAndEco, Temp_avg < as.numeric(temperaturestat["1st Qu."]))
ColdFishes <- pivot_longer(ColdFishes, cols = c("T_G.H", 
                                  "T_C.H", 
                                  "T_A.H", 
                                  "G_T.H", 
                                  "G_C.H", 
                                  "G_A.H", 
                                  "C_T.H", 
                                  "C_G.H", 
                                  "C_A.H", 
                                  "A_T.H", 
                                  "A_G.H", 
                                  "A_C.H"), names_to = "Mut", values_to = "MutSpec" )
AverageMutSpecCold <- ColdFishes %>% group_by(Mut) %>% summarise(meanmutspec = mean(MutSpec, na.rm = TRUE), error =
                                                                sd(MutSpec, na.rm = TRUE))
Ncold <- length(unique(ColdFishes$Species))

sf2w = ggplot(AverageMutSpecWarm,
  aes(
    x = AverageMutSpecWarm$Mut,
    y = AverageMutSpecWarm$meanmutspec,
    fill = AverageMutSpecWarm$Mut
    )
  ) +
  geom_bar(stat = "identity") +
  scale_fill_manual(
    values = c(
      "#bdbdbd",
      "#00a426",
      "#bdbdbd",
      "#bdbdbd",
      "#bdbdbd",
      "#ff0000",
      "#900000",
      "#bdbdbd",
      "#bdbdbd",
      "#bdbdbd",
      "#005213",
      "#bdbdbd"
    )
  ) +
  labs(title = "Average mitochondrial mutational spectrum in Warm Fishes, <23C", x = "Substitution types", y = "Normalised Frequencies") + theme(
    legend.position = "none",
    # Remove legend
    panel.background = element_blank(),
    axis.line = element_line(color = "black")# Remove background
  ) + 
  geom_text(x=2, y=0.3, label=paste("N warm =", Nwarm), size = 7)


sf2c = ggplot(AverageMutSpecCold,
  aes(
    x = AverageMutSpecCold$Mut,
    y = AverageMutSpecCold$meanmutspec,
    fill = AverageMutSpecCold$Mut
    )
  ) +
  geom_bar(stat = "identity") +
  scale_fill_manual(
    values = c(
      "#bdbdbd",
      "#00a426",
      "#bdbdbd",
      "#bdbdbd",
      "#bdbdbd",
      "#ff0000",
      "#900000",
      "#bdbdbd",
      "#bdbdbd",
      "#bdbdbd",
      "#005213",
      "#bdbdbd"
    )
  ) +
  labs(title = "Average mitochondrial mutational spectrum in Cold Fishes, > 13C ", x = "Substitution types", y = "Normalised Frequencies") + theme(
    legend.position = "none",
    # Remove legend
    panel.background = element_blank(),
    axis.line = element_line(color = "black")# Remove background
  )+ 
  geom_text(x=2, y=0.3, label=paste("N cold =", Ncold), size = 7)

both <- ggarrange(sf2c, sf2w, ncol = 2)
#ggsave(here(plots_dir,"PolymorphicData.Actinopterygii.AverageMutSpecColdAndWarm.pdf"),
#  plot = both,
#  width = 12,
#  height = 6
#)
both
```

# Studies of A\>G spectra with temperature and other parameters in fishes

## A\>G freqs dependence on climate zones
###CytB

```{r piecharts}
ClimateZonetest <- na.omit(data.frame(subset(MutAndEcoCb, select = c(A_G.H, Climate_zone))))
ClimateZonetest$Climate_zone <- as.factor(ClimateZonetest$Climate_zone)
AGmedian <- median(ClimateZonetest$A_G.H)
ClimateZonetest$A_G.Freq <- "low"
ClimateZonetest[ClimateZonetest$A_G.H >= AGmedian,]$A_G.Freq <- "high"
ClimateZonetest$A_G.Freq <- as.factor(ClimateZonetest$A_G.Freq)
ClimateZonetest <- ClimateZonetest[,2-3]
sf3 <- ggpiestats(data = ClimateZonetest, x = Climate_zone, y = A_G.Freq)
sf3
```

```{r statistics climate zones}
ClimateZonetest <- na.omit(data.frame(subset(MutAndEcoCb, select = c(A_G.H, Climate_zone))))
cold <- c("boreal", "polar", "temperate")
warm <- c("subtropical", "tropical")
wilcox.test(ClimateZonetest[ClimateZonetest$Climate_zone %in% cold,]$A_G.H, 
            ClimateZonetest[ClimateZonetest$Climate_zone %in% warm,]$A_G.H)
ClimateZoneSP <- na.omit(data.frame(subset(MutAndEcoCb, select = c(Species, Climate_zone))))
ClimateZoneSP$Group <- ifelse(ClimateZonetest$Climate_zone %in% cold, "Cold", "Warm")

#write.table(ClimateZoneSP, here(data_dir, "Species_by_climate_zones.txt"), row.names = F)
```

## A\>G freqs dependence on temperature

```{r statistics with temp cytb, message = FALSE, warning = FALSE}
tidy(cor.test(MutAndEcoCb$A_G.H, MutAndEcoCb$Temp_avg, method = 'spearman'))
tidy(cor.test(MutAndEcoCb$A_T.H, MutAndEcoCb$Temp_avg, method = 'spearman'))
tidy(cor.test(MutAndEcoCb$A_C.H, MutAndEcoCb$Temp_avg, method = 'spearman'))
tidy(cor.test(MutAndEcoCb$T_A.H, MutAndEcoCb$Temp_avg, method = 'spearman'))
tidy(cor.test(MutAndEcoCb$T_G.H, MutAndEcoCb$Temp_avg, method = 'spearman'))
tidy(cor.test(MutAndEcoCb$T_C.H, MutAndEcoCb$Temp_avg, method = 'spearman'))
tidy(cor.test(MutAndEcoCb$C_A.H, MutAndEcoCb$Temp_avg, method = 'spearman'))
tidy(cor.test(MutAndEcoCb$C_T.H, MutAndEcoCb$Temp_avg, method = 'spearman'))
tidy(cor.test(MutAndEcoCb$C_G.H, MutAndEcoCb$Temp_avg, method = 'spearman'))
tidy(cor.test(MutAndEcoCb$G_T.H, MutAndEcoCb$Temp_avg, method = 'spearman'))
tidy(cor.test(MutAndEcoCb$G_A.H, MutAndEcoCb$Temp_avg, method = 'spearman'))
tidy(cor.test(MutAndEcoCb$G_C.H, MutAndEcoCb$Temp_avg, method = 'spearman'))

cor.test(MutAndEcoCb$Temp_avg, MutAndEcoCb$Rates, method="spearman")


SubsetClass<- unique(MutAndEcoCb$Class)
spN <- nrow(MutAndEcoCb[!is.na(MutAndEcoCb$Temp_avg),])
cat("N species =", spN, "Taxa =", SubsetClass)

f1b <- ggplot(MutAndEcoCb, aes(x=Temp_avg, y=A_G.H)) + 
    geom_point(color="#73514f") +                  # Scatter plot
    geom_smooth(method = "lm", color = "black",        # Regression line + CI
                fill = "lightgray", se = TRUE) +     # Set se=TRUE for confidence interval
    stat_cor(method = "spearman",                    # Spearman correlation stats
            aes(label = paste(..r.label.., ..p.label.., "N", spN, sep = "~"))) +  # Include n
    labs(x = "Median annual water temperature, ?C", y = "AH>GH") +   # Axis labels
    theme_minimal()                                  # Theme customization
 
ggsave(here(plots_dir,"PolymorphicData.Actinopterygii.A_GwitnTemperatureCytB.pdf"),
  plot = f1b,
  width = 6,
  height = 6
) 
f1b

```

## A\>G freqs dependence on time of maturation
```{r statistics with maturation cytb, message = FALSE, warning = FALSE}
tidy(cor.test(MutAndEcoCb$A_G.H, MutAndEcoCb$tm, method = 'spearman'))
tidy(cor.test(MutAndEcoCb$A_T.H, MutAndEcoCb$tm, method = 'spearman'))
tidy(cor.test(MutAndEcoCb$A_C.H, MutAndEcoCb$tm, method = 'spearman'))
tidy(cor.test(MutAndEcoCb$T_A.H, MutAndEcoCb$tm, method = 'spearman'))
tidy(cor.test(MutAndEcoCb$T_G.H, MutAndEcoCb$tm, method = 'spearman'))
tidy(cor.test(MutAndEcoCb$T_C.H, MutAndEcoCb$tm, method = 'spearman'))
tidy(cor.test(MutAndEcoCb$C_A.H, MutAndEcoCb$tm, method = 'spearman'))
tidy(cor.test(MutAndEcoCb$C_T.H, MutAndEcoCb$tm, method = 'spearman'))
tidy(cor.test(MutAndEcoCb$C_G.H, MutAndEcoCb$tm, method = 'spearman'))
tidy(cor.test(MutAndEcoCb$G_T.H, MutAndEcoCb$tm, method = 'spearman'))
tidy(cor.test(MutAndEcoCb$G_A.H, MutAndEcoCb$tm, method = 'spearman'))
tidy(cor.test(MutAndEcoCb$G_C.H, MutAndEcoCb$tm, method = 'spearman'))
```

```{r statistics with temp orders, message = FALSE, warning = FALSE, echo=FALSE}
taxonomy <- rfishbase::species_names()
taxonomy$Species <- gsub(" ", "_", taxonomy$Species)
MutAndEco <- merge(MutAndEco, taxonomy, by="Species", all.x = TRUE)
temp <- MutAndEco[!is.na(MutAndEco$Temp_avg),]

my <- c("Acanthuriformes", "Anabantiformes", "Centrarchiformes", "Cypriniformes", "Cyprinodontiformes", "Gobiiformes", 
        "Eupercaria/misc", "Galaxiiformes", "Perciformes/Percoidei", "Salmoniformes", "Scombriformes", "Siluriformes")
temp <- temp[temp$Order %in% my,]
table(temp$Order)

MediansByOrder <- tibble()

for (i in my) {
  imediant = median(temp[temp$Order == i,]$Temp_avg, na.rm=TRUE)
  imedianm = median(temp[temp$Order == i,]$A_G.H, na.rm=TRUE)
  iorder <- c(i, imediant, imedianm)
  MediansByOrder <- rbind(MediansByOrder, iorder)
}
colnames(MediansByOrder) <- c("Order", "Median_temperature", "Median_A_G")
MediansByOrder$Median_temperature <- as.numeric(MediansByOrder$Median_temperature)
MediansByOrder$Median_A_G <- as.numeric(MediansByOrder$Median_A_G)
cor.test(MediansByOrder$Median_temperature, MediansByOrder$Median_A_G, method = 'spearman')

sf5 <- ggplot(MediansByOrder, aes(x=Median_temperature, y=Median_A_G, label = Order, color = Order)) + 
    geom_point() + 
    geom_smooth(method = "lm", color = "red", se=FALSE) +
    labs(x = "Median annual water temperature, ?C", y = "Median_A.H>G.H") +
    geom_text(
      label=MediansByOrder$Order, 
      nudge_x = 0.0025, nudge_y = 0.0025, 
      check_overlap = T) +
    theme_bw()
sf5 + theme(legend.position = "none")

```

## LM

```{r message=FALSE, warning=FALSE}
MutAndEcoCb$DeltaT <- MutAndEcoCb$Temp_max - MutAndEcoCb$Temp_min
summary(lm(A_G.H ~ Temp_avg + DeltaT + Wmax + tmax + tm + FoodTroph, data = MutAndEcoCb))
summary(lm(A_G.H ~ Temp_avg + FoodTroph + Wmax, data = MutAndEcoCb))
summary(lm(A_G.H ~ Temp_avg + tmax + tm, data = MutAndEcoCb))
summary(lm(A_G.H ~ Temp_avg + DeltaT, data = MutAndEcoCb))

```

## PGLS

```{r PGLS_rates, message=FALSE, warning=FALSE}
tree = read.tree(file = here(data_dir, "Cytb.nm.Mafft_L-INS-i.pr.Bs.mtVer-F-I-G8.treefile.cut-c0.0001-rr"))
MutAndEcoPGLS <- MutAndEcoCb %>% dplyr::select(Species, A_G.H, C_T.H, G_A.H, T_C.H, Rates)

row.names(MutAndEcoPGLS) = MutAndEcoPGLS$Species
tree_pruned = treedata(tree, MutAndEcoPGLS, sort=T, warnings=T)$phy
data<-as.data.frame(treedata(tree_pruned, MutAndEcoPGLS, sort=T, warnings=T)$data)
data$Species = as.character(data$Species)
data$A_G.H = as.numeric(data$A_G.H)
data$C_T.H = as.numeric(data$C_T.H)
data$G_A.H = as.numeric(data$G_A.H)
data$T_C.H = as.numeric(data$T_C.H)
data$Rates = as.numeric(data$Rates)
data$Tv <- 1-(data$A_G.H + data$G_A.H + data$T_C.H + data$C_T.H)
data_comp <- comparative.data(tree_pruned, data, Species, vcv=TRUE)


summary(pgls(A_G.H ~ Rates, data_comp, lambda="ML"))
summary(pgls(C_T.H ~ Rates, data_comp, lambda="ML"))
summary(pgls(G_A.H ~ Rates, data_comp, lambda="ML"))
summary(pgls(T_C.H ~ Rates, data_comp, lambda="ML"))

#la.A_G <- phylosig(tree_pruned, data$A_G.H, method = "lambda", test = TRUE) #Phylogenetic signal lambda : 0.280122 
#la.C_T <- phylosig(tree_pruned, data$C_T.H, method = "lambda", test = TRUE) #Phylogenetic signal lambda : 0.119451 
#la.G_A <- phylosig(tree_pruned, data$G_A.H, method = "lambda", test = TRUE) #Phylogenetic signal lambda : 0.719915 
#la.T_C <- phylosig(tree_pruned, data$T_C.H, method = "lambda", test = TRUE) #Phylogenetic signal lambda : 0.813846 
#la.Tv <- phylosig(tree_pruned, data$Tv, method = "lambda", test = TRUE) #Phylogenetic signal lambda : 0.113722
#la.temp <- phylosig(tree_pruned, data$Temp_avg, method = "lambda", test = TRUE) #Phylogenetic signal lambda : 0.92032
```
```{r PGLS_temperature, message=FALSE, warning=FALSE}
tree = read.tree(file = here(data_dir, "Cytb.nm.Mafft_L-INS-i.pr.Bs.mtVer-F-I-G8.treefile.cut-c0.0001-rr"))
MutAndEcoPGLS <- MutAndEcoCb %>% dplyr::select(Species, A_G.H, C_T.H, G_A.H, T_C.H, Temp_avg)

row.names(MutAndEcoPGLS) = MutAndEcoPGLS$Species
tree_pruned = treedata(tree, MutAndEcoPGLS, sort=T, warnings=T)$phy
data<-as.data.frame(treedata(tree_pruned, MutAndEcoPGLS, sort=T, warnings=T)$data)
data$Species = as.character(data$Species)
data$A_G.H = as.numeric(data$A_G.H)
data$C_T.H = as.numeric(data$C_T.H)
data$G_A.H = as.numeric(data$G_A.H)
data$T_C.H = as.numeric(data$T_C.H)
data$Temp_avg = as.numeric(data$Temp_avg)
data$Tv <- 1-(data$A_G.H + data$G_A.H + data$T_C.H + data$C_T.H)
data_comp <- comparative.data(tree_pruned, data, Species, vcv=TRUE)


summary(pgls(A_G.H ~ Temp_avg, data_comp, lambda="ML"))
summary(pgls(C_T.H ~ Temp_avg, data_comp, lambda="ML"))
summary(pgls(G_A.H ~ Temp_avg, data_comp, lambda="ML"))
summary(pgls(T_C.H ~ Temp_avg, data_comp, lambda="ML"))

```

